{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "032d8a5e-66e3-4513-9d5f-caa69a38a4c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "aff4d015-d7ab-485e-9707-b88f3b582c75",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>p_value</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>hm_coordinate_conversion</th>\n",
       "      <th>hm_code</th>\n",
       "      <th>rsid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>13668</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.4138</td>\n",
       "      <td>0.2733</td>\n",
       "      <td>0.005716</td>\n",
       "      <td>0.1300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>10</td>\n",
       "      <td>rs2691328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>14506</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.4047</td>\n",
       "      <td>0.2571</td>\n",
       "      <td>0.005682</td>\n",
       "      <td>0.1155</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>10</td>\n",
       "      <td>rs1240557819</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>14717</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>-0.5758</td>\n",
       "      <td>0.5729</td>\n",
       "      <td>0.000775</td>\n",
       "      <td>0.3149</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>10</td>\n",
       "      <td>rs377122907</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>14773</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.1723</td>\n",
       "      <td>0.1617</td>\n",
       "      <td>0.017180</td>\n",
       "      <td>0.2866</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>10</td>\n",
       "      <td>rs878915777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>14860</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.1691</td>\n",
       "      <td>0.1915</td>\n",
       "      <td>0.012010</td>\n",
       "      <td>0.3772</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>10</td>\n",
       "      <td>rs533499096</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chromosome  base_pair_location effect_allele other_allele    beta  \\\n",
       "0           1               13668             A            G -0.4138   \n",
       "1           1               14506             A            G -0.4047   \n",
       "2           1               14717             A            G -0.5758   \n",
       "3           1               14773             T            C -0.1723   \n",
       "4           1               14860             T            G  0.1691   \n",
       "\n",
       "   standard_error  effect_allele_frequency  p_value  variant_id  \\\n",
       "0          0.2733                 0.005716   0.1300         NaN   \n",
       "1          0.2571                 0.005682   0.1155         NaN   \n",
       "2          0.5729                 0.000775   0.3149         NaN   \n",
       "3          0.1617                 0.017180   0.2866         NaN   \n",
       "4          0.1915                 0.012010   0.3772         NaN   \n",
       "\n",
       "  hm_coordinate_conversion  hm_code          rsid  \n",
       "0                       lo       10     rs2691328  \n",
       "1                       lo       10  rs1240557819  \n",
       "2                       lo       10   rs377122907  \n",
       "3                       lo       10   rs878915777  \n",
       "4                       lo       10   rs533499096  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# The following GWAS Summary Statistics is from GWAS Catalog\n",
    "df1 = pd.read_csv(\n",
    "    \"GCST90018799.h.tsv\",\n",
    "    sep=\"\\t\"\n",
    ")\n",
    "df1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "7c6454a7-6689-4d71-b831-25cdd319d36b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['chromosome', 'base_pair_location', 'effect_allele', 'other_allele', 'beta', 'standard_error', 'effect_allele_frequency', 'p_value', 'variant_id', 'hm_coordinate_conversion', 'hm_code', 'rsid']\n"
     ]
    }
   ],
   "source": [
    "print(df1.columns.tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "0b343f7f-ec0c-438c-a9f5-623be3218a62",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>p_value</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>hm_coordinate_conversion</th>\n",
       "      <th>hm_code</th>\n",
       "      <th>rsid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>720930</th>\n",
       "      <td>1</td>\n",
       "      <td>87980553</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.1193</td>\n",
       "      <td>0.0216</td>\n",
       "      <td>0.054206</td>\n",
       "      <td>3.313000e-08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>10</td>\n",
       "      <td>rs17130230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>989397</th>\n",
       "      <td>1</td>\n",
       "      <td>121471494</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0841</td>\n",
       "      <td>0.0125</td>\n",
       "      <td>0.392685</td>\n",
       "      <td>1.840000e-11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>10</td>\n",
       "      <td>rs12134101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>989678</th>\n",
       "      <td>1</td>\n",
       "      <td>121538815</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0825</td>\n",
       "      <td>0.0115</td>\n",
       "      <td>0.307711</td>\n",
       "      <td>9.287000e-13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>10</td>\n",
       "      <td>rs11249433</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>990472</th>\n",
       "      <td>1</td>\n",
       "      <td>143279994</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.1863</td>\n",
       "      <td>0.0312</td>\n",
       "      <td>0.263500</td>\n",
       "      <td>2.356000e-09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>10</td>\n",
       "      <td>rs1437461076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>990506</th>\n",
       "      <td>1</td>\n",
       "      <td>143284560</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.1893</td>\n",
       "      <td>0.0296</td>\n",
       "      <td>0.319100</td>\n",
       "      <td>1.603000e-10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>10</td>\n",
       "      <td>rs1457477682</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23349347</th>\n",
       "      <td>22</td>\n",
       "      <td>40658455</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.0707</td>\n",
       "      <td>0.0116</td>\n",
       "      <td>0.260785</td>\n",
       "      <td>1.059000e-09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>10</td>\n",
       "      <td>rs56169543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23349367</th>\n",
       "      <td>22</td>\n",
       "      <td>40661666</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0710</td>\n",
       "      <td>0.0116</td>\n",
       "      <td>0.260724</td>\n",
       "      <td>9.625000e-10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>10</td>\n",
       "      <td>rs113860965</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23349370</th>\n",
       "      <td>22</td>\n",
       "      <td>40661973</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.0709</td>\n",
       "      <td>0.0116</td>\n",
       "      <td>0.260781</td>\n",
       "      <td>1.026000e-09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>10</td>\n",
       "      <td>rs7284506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23349376</th>\n",
       "      <td>22</td>\n",
       "      <td>40663167</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0697</td>\n",
       "      <td>0.0116</td>\n",
       "      <td>0.264747</td>\n",
       "      <td>1.585000e-09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>10</td>\n",
       "      <td>rs79898470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23349384</th>\n",
       "      <td>22</td>\n",
       "      <td>40664849</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.0708</td>\n",
       "      <td>0.0116</td>\n",
       "      <td>0.261468</td>\n",
       "      <td>1.065000e-09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>5</td>\n",
       "      <td>rs73171017</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3457 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          chromosome  base_pair_location effect_allele other_allele    beta  \\\n",
       "720930             1            87980553             A            G  0.1193   \n",
       "989397             1           121471494             G            A  0.0841   \n",
       "989678             1           121538815             G            A  0.0825   \n",
       "990472             1           143279994             T            C  0.1863   \n",
       "990506             1           143284560             C            T  0.1893   \n",
       "...              ...                 ...           ...          ...     ...   \n",
       "23349347          22            40658455             A            G  0.0707   \n",
       "23349367          22            40661666             T            C  0.0710   \n",
       "23349370          22            40661973             T            C  0.0709   \n",
       "23349376          22            40663167             G            A  0.0697   \n",
       "23349384          22            40664849             T            A  0.0708   \n",
       "\n",
       "          standard_error  effect_allele_frequency       p_value  variant_id  \\\n",
       "720930            0.0216                 0.054206  3.313000e-08         NaN   \n",
       "989397            0.0125                 0.392685  1.840000e-11         NaN   \n",
       "989678            0.0115                 0.307711  9.287000e-13         NaN   \n",
       "990472            0.0312                 0.263500  2.356000e-09         NaN   \n",
       "990506            0.0296                 0.319100  1.603000e-10         NaN   \n",
       "...                  ...                      ...           ...         ...   \n",
       "23349347          0.0116                 0.260785  1.059000e-09         NaN   \n",
       "23349367          0.0116                 0.260724  9.625000e-10         NaN   \n",
       "23349370          0.0116                 0.260781  1.026000e-09         NaN   \n",
       "23349376          0.0116                 0.264747  1.585000e-09         NaN   \n",
       "23349384          0.0116                 0.261468  1.065000e-09         NaN   \n",
       "\n",
       "         hm_coordinate_conversion  hm_code          rsid  \n",
       "720930                         lo       10    rs17130230  \n",
       "989397                         lo       10    rs12134101  \n",
       "989678                         lo       10    rs11249433  \n",
       "990472                         lo       10  rs1437461076  \n",
       "990506                         lo       10  rs1457477682  \n",
       "...                           ...      ...           ...  \n",
       "23349347                       lo       10    rs56169543  \n",
       "23349367                       lo       10   rs113860965  \n",
       "23349370                       lo       10     rs7284506  \n",
       "23349376                       lo       10    rs79898470  \n",
       "23349384                       lo        5    rs73171017  \n",
       "\n",
       "[3457 rows x 12 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1_pval_sig = df1[df1[\"p_value\"] <= 5e-8]\n",
    "df1_pval_sig"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "1d5d343a-60d9-41e6-a48c-1df29014f641",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "720930        rs17130230\n",
       "989397        rs12134101\n",
       "989678        rs11249433\n",
       "990472      rs1437461076\n",
       "990506      rs1457477682\n",
       "                ...     \n",
       "23349347      rs56169543\n",
       "23349367     rs113860965\n",
       "23349370       rs7284506\n",
       "23349376      rs79898470\n",
       "23349384      rs73171017\n",
       "Name: rsid, Length: 3457, dtype: object"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1_pval_sig_rsid = df1_pval_sig[\"rsid\"]\n",
    "df1_pval_sig_rsid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "278a7f2d-b8dd-4fd5-be79-66b393e8ce5a",
   "metadata": {},
   "outputs": [],
   "source": [
    "df1_pval_sig_rsid.to_csv(\n",
    "    \"gwas_rsid_5e8.txt\",\n",
    "    index = False,\n",
    "    header = False\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "a42d808d-d98c-4343-b3de-7e1775e8225d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>›#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>SYMBOL.1</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>rsid</th>\n",
       "      <th>clinvar</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1_1986896_A/G</td>\n",
       "      <td>1:1986896-1986896</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1_1986896_A/G</td>\n",
       "      <td>1:1986896-1986896</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1_1986896_A/G</td>\n",
       "      <td>1:1986896-1986896</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1_1986896_A/G</td>\n",
       "      <td>1:1986896-1986896</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1_1986896_A/G</td>\n",
       "      <td>1:1986896-1986896</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101496</th>\n",
       "      <td>X_153138145_C/G</td>\n",
       "      <td>X:153138145-153138145</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>ENSG00000198910</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101497</th>\n",
       "      <td>X_153138145_C/G</td>\n",
       "      <td>X:153138145-153138145</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>ENSG00000198910</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101498</th>\n",
       "      <td>X_153138145_C/G</td>\n",
       "      <td>X:153138145-153138145</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>ENSG00000198910</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101499</th>\n",
       "      <td>X_153138145_C/G</td>\n",
       "      <td>X:153138145-153138145</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>ENSG00000198910</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101500</th>\n",
       "      <td>X_153138145_C/G</td>\n",
       "      <td>X:153138145-153138145</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>ENSG00000198910</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>101501 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       ›#Uploaded_variation               Location Existing_variation SYMBOL  \\\n",
       "0             1_1986896_A/G      1:1986896-1986896        rs765901754  PRKCZ   \n",
       "1             1_1986896_A/G      1:1986896-1986896        rs765901754  PRKCZ   \n",
       "2             1_1986896_A/G      1:1986896-1986896        rs765901754  PRKCZ   \n",
       "3             1_1986896_A/G      1:1986896-1986896        rs765901754  PRKCZ   \n",
       "4             1_1986896_A/G      1:1986896-1986896        rs765901754  PRKCZ   \n",
       "...                     ...                    ...                ...    ...   \n",
       "101496      X_153138145_C/G  X:153138145-153138145        rs201990980  L1CAM   \n",
       "101497      X_153138145_C/G  X:153138145-153138145        rs201990980  L1CAM   \n",
       "101498      X_153138145_C/G  X:153138145-153138145        rs201990980  L1CAM   \n",
       "101499      X_153138145_C/G  X:153138145-153138145        rs201990980  L1CAM   \n",
       "101500      X_153138145_C/G  X:153138145-153138145        rs201990980  L1CAM   \n",
       "\n",
       "                   Gene SYMBOL.1                         Consequence  \\\n",
       "0       ENSG00000067606    PRKCZ                    missense_variant   \n",
       "1       ENSG00000067606    PRKCZ               upstream_gene_variant   \n",
       "2       ENSG00000067606    PRKCZ                    missense_variant   \n",
       "3       ENSG00000067606    PRKCZ  non_coding_transcript_exon_variant   \n",
       "4       ENSG00000067606    PRKCZ  non_coding_transcript_exon_variant   \n",
       "...                 ...      ...                                 ...   \n",
       "101496  ENSG00000198910    L1CAM  non_coding_transcript_exon_variant   \n",
       "101497  ENSG00000198910    L1CAM               upstream_gene_variant   \n",
       "101498  ENSG00000198910    L1CAM               upstream_gene_variant   \n",
       "101499  ENSG00000198910    L1CAM                    missense_variant   \n",
       "101500  ENSG00000198910    L1CAM                    missense_variant   \n",
       "\n",
       "          IMPACT              CLIN_SIG         rsid clinvar  \n",
       "0       MODERATE                     -  rs765901754     NaN  \n",
       "1       MODIFIER                     -  rs765901754     NaN  \n",
       "2       MODERATE                     -  rs765901754     NaN  \n",
       "3       MODIFIER                     -  rs765901754     NaN  \n",
       "4       MODIFIER                     -  rs765901754     NaN  \n",
       "...          ...                   ...          ...     ...  \n",
       "101496  MODIFIER  benign,likely_benign  rs201990980     NaN  \n",
       "101497  MODIFIER  benign,likely_benign  rs201990980     NaN  \n",
       "101498  MODIFIER  benign,likely_benign  rs201990980     NaN  \n",
       "101499  MODERATE  benign,likely_benign  rs201990980     NaN  \n",
       "101500  MODERATE  benign,likely_benign  rs201990980     NaN  \n",
       "\n",
       "[101501 rows x 11 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filepath = \"annotated_df5.csv\"\n",
    "df2 = pd.read_csv(\n",
    "   filepath,\n",
    "    sep=\"\\t\"\n",
    ")\n",
    "\n",
    "df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "dc838afd-e4f1-4e67-ba51-5e4df25e6f3e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['›#Uploaded_variation', 'Location', 'Existing_variation', 'SYMBOL', 'Gene', 'SYMBOL.1', 'Consequence', 'IMPACT', 'CLIN_SIG', 'rsid', 'clinvar']\n"
     ]
    }
   ],
   "source": [
    "print(df2.columns.tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "8a106098-b74a-42d6-a6bf-4c89db1f995b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "rsid\n",
       "rs121434592     948\n",
       "rs121913279     932\n",
       "rs28934578      663\n",
       "rs121912651     527\n",
       "rs11540652      476\n",
       "               ... \n",
       "rs1004071220      1\n",
       "rs913362638       1\n",
       "rs1237767299      1\n",
       "rs148279547       1\n",
       "rs752960380       1\n",
       "Name: count, Length: 8939, dtype: int64"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2[\"rsid\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "a3f1fee9-04db-40d7-8a4b-23e65963ca35",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "rsid\n",
       "rs72758040     2\n",
       "rs4752570      2\n",
       "rs35301881     1\n",
       "rs12050723     1\n",
       "rs12050726     1\n",
       "              ..\n",
       "rs377455592    1\n",
       "rs546810768    1\n",
       "rs78059834     1\n",
       "rs76077130     1\n",
       "rs34315830     1\n",
       "Name: count, Length: 13765, dtype: int64"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1_pval_sig[\"rsid\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "eb49486d-4117-464f-a1ad-f3803215cc43",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>p_value</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>hm_coordinate_conversion</th>\n",
       "      <th>...</th>\n",
       "      <th>›#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>SYMBOL.1</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>clinvar</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>2144214</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.2885</td>\n",
       "      <td>0.3585</td>\n",
       "      <td>0.000361</td>\n",
       "      <td>0.4210</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>1_2075653_C/T</td>\n",
       "      <td>1:2075653-2075653</td>\n",
       "      <td>rs143527868,COSV66070211</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>2144214</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.2885</td>\n",
       "      <td>0.3585</td>\n",
       "      <td>0.000361</td>\n",
       "      <td>0.4210</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>1_2075653_C/T</td>\n",
       "      <td>1:2075653-2075653</td>\n",
       "      <td>rs143527868,COSV66070211</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>2144214</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.2885</td>\n",
       "      <td>0.3585</td>\n",
       "      <td>0.000361</td>\n",
       "      <td>0.4210</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>1_2075653_C/T</td>\n",
       "      <td>1:2075653-2075653</td>\n",
       "      <td>rs143527868,COSV66070211</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>2144214</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.2885</td>\n",
       "      <td>0.3585</td>\n",
       "      <td>0.000361</td>\n",
       "      <td>0.4210</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>1_2075653_C/T</td>\n",
       "      <td>1:2075653-2075653</td>\n",
       "      <td>rs143527868,COSV66070211</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>2144214</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.2885</td>\n",
       "      <td>0.3585</td>\n",
       "      <td>0.000361</td>\n",
       "      <td>0.4210</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>1_2075653_C/T</td>\n",
       "      <td>1:2075653-2075653</td>\n",
       "      <td>rs143527868,COSV66070211</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5665</th>\n",
       "      <td>22</td>\n",
       "      <td>41178680</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.8445</td>\n",
       "      <td>0.5394</td>\n",
       "      <td>0.000536</td>\n",
       "      <td>0.1174</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>22_41574684_C/G</td>\n",
       "      <td>22:41574684-41574684</td>\n",
       "      <td>rs111489369</td>\n",
       "      <td>RP1-85F18.6</td>\n",
       "      <td>ENSG00000232754</td>\n",
       "      <td>RP1-85F18.6</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5666</th>\n",
       "      <td>22</td>\n",
       "      <td>41178680</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.8445</td>\n",
       "      <td>0.5394</td>\n",
       "      <td>0.000536</td>\n",
       "      <td>0.1174</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>22_41574684_C/G</td>\n",
       "      <td>22:41574684-41574684</td>\n",
       "      <td>rs111489369</td>\n",
       "      <td>RP1-85F18.5</td>\n",
       "      <td>ENSG00000231993</td>\n",
       "      <td>RP1-85F18.5</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5667</th>\n",
       "      <td>22</td>\n",
       "      <td>41178680</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.8445</td>\n",
       "      <td>0.5394</td>\n",
       "      <td>0.000536</td>\n",
       "      <td>0.1174</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>22_41574684_C/G</td>\n",
       "      <td>22:41574684-41574684</td>\n",
       "      <td>rs111489369</td>\n",
       "      <td>EP300</td>\n",
       "      <td>ENSG00000100393</td>\n",
       "      <td>EP300</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5668</th>\n",
       "      <td>22</td>\n",
       "      <td>41178680</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.8445</td>\n",
       "      <td>0.5394</td>\n",
       "      <td>0.000536</td>\n",
       "      <td>0.1174</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>22_41574684_C/G</td>\n",
       "      <td>22:41574684-41574684</td>\n",
       "      <td>rs111489369</td>\n",
       "      <td>RP1-85F18.6</td>\n",
       "      <td>ENSG00000232754</td>\n",
       "      <td>RP1-85F18.6</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5669</th>\n",
       "      <td>22</td>\n",
       "      <td>41178680</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.8445</td>\n",
       "      <td>0.5394</td>\n",
       "      <td>0.000536</td>\n",
       "      <td>0.1174</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>22_41574684_C/G</td>\n",
       "      <td>22:41574684-41574684</td>\n",
       "      <td>rs111489369</td>\n",
       "      <td>RP1-85F18.5</td>\n",
       "      <td>ENSG00000231993</td>\n",
       "      <td>RP1-85F18.5</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5670 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chromosome  base_pair_location effect_allele other_allele    beta  \\\n",
       "0              1             2144214             T            C  0.2885   \n",
       "1              1             2144214             T            C  0.2885   \n",
       "2              1             2144214             T            C  0.2885   \n",
       "3              1             2144214             T            C  0.2885   \n",
       "4              1             2144214             T            C  0.2885   \n",
       "...          ...                 ...           ...          ...     ...   \n",
       "5665          22            41178680             G            C -0.8445   \n",
       "5666          22            41178680             G            C -0.8445   \n",
       "5667          22            41178680             G            C -0.8445   \n",
       "5668          22            41178680             G            C -0.8445   \n",
       "5669          22            41178680             G            C -0.8445   \n",
       "\n",
       "      standard_error  effect_allele_frequency  p_value  variant_id  \\\n",
       "0             0.3585                 0.000361   0.4210         NaN   \n",
       "1             0.3585                 0.000361   0.4210         NaN   \n",
       "2             0.3585                 0.000361   0.4210         NaN   \n",
       "3             0.3585                 0.000361   0.4210         NaN   \n",
       "4             0.3585                 0.000361   0.4210         NaN   \n",
       "...              ...                      ...      ...         ...   \n",
       "5665          0.5394                 0.000536   0.1174         NaN   \n",
       "5666          0.5394                 0.000536   0.1174         NaN   \n",
       "5667          0.5394                 0.000536   0.1174         NaN   \n",
       "5668          0.5394                 0.000536   0.1174         NaN   \n",
       "5669          0.5394                 0.000536   0.1174         NaN   \n",
       "\n",
       "     hm_coordinate_conversion  ...  ›#Uploaded_variation  \\\n",
       "0                          lo  ...         1_2075653_C/T   \n",
       "1                          lo  ...         1_2075653_C/T   \n",
       "2                          lo  ...         1_2075653_C/T   \n",
       "3                          lo  ...         1_2075653_C/T   \n",
       "4                          lo  ...         1_2075653_C/T   \n",
       "...                       ...  ...                   ...   \n",
       "5665                       lo  ...       22_41574684_C/G   \n",
       "5666                       lo  ...       22_41574684_C/G   \n",
       "5667                       lo  ...       22_41574684_C/G   \n",
       "5668                       lo  ...       22_41574684_C/G   \n",
       "5669                       lo  ...       22_41574684_C/G   \n",
       "\n",
       "                  Location        Existing_variation       SYMBOL  \\\n",
       "0        1:2075653-2075653  rs143527868,COSV66070211        PRKCZ   \n",
       "1        1:2075653-2075653  rs143527868,COSV66070211        PRKCZ   \n",
       "2        1:2075653-2075653  rs143527868,COSV66070211        PRKCZ   \n",
       "3        1:2075653-2075653  rs143527868,COSV66070211        PRKCZ   \n",
       "4        1:2075653-2075653  rs143527868,COSV66070211        PRKCZ   \n",
       "...                    ...                       ...          ...   \n",
       "5665  22:41574684-41574684               rs111489369  RP1-85F18.6   \n",
       "5666  22:41574684-41574684               rs111489369  RP1-85F18.5   \n",
       "5667  22:41574684-41574684               rs111489369        EP300   \n",
       "5668  22:41574684-41574684               rs111489369  RP1-85F18.6   \n",
       "5669  22:41574684-41574684               rs111489369  RP1-85F18.5   \n",
       "\n",
       "                 Gene     SYMBOL.1  \\\n",
       "0     ENSG00000067606        PRKCZ   \n",
       "1     ENSG00000067606        PRKCZ   \n",
       "2     ENSG00000067606        PRKCZ   \n",
       "3     ENSG00000067606        PRKCZ   \n",
       "4     ENSG00000067606        PRKCZ   \n",
       "...               ...          ...   \n",
       "5665  ENSG00000232754  RP1-85F18.6   \n",
       "5666  ENSG00000231993  RP1-85F18.5   \n",
       "5667  ENSG00000100393        EP300   \n",
       "5668  ENSG00000232754  RP1-85F18.6   \n",
       "5669  ENSG00000231993  RP1-85F18.5   \n",
       "\n",
       "                                       Consequence    IMPACT  \\\n",
       "0                                 missense_variant  MODERATE   \n",
       "1                              5_prime_UTR_variant  MODIFIER   \n",
       "2                              5_prime_UTR_variant  MODIFIER   \n",
       "3               non_coding_transcript_exon_variant  MODIFIER   \n",
       "4                                 missense_variant  MODERATE   \n",
       "...                                            ...       ...   \n",
       "5665  intron_variant,non_coding_transcript_variant  MODIFIER   \n",
       "5666  intron_variant,non_coding_transcript_variant  MODIFIER   \n",
       "5667                            synonymous_variant       LOW   \n",
       "5668  intron_variant,non_coding_transcript_variant  MODIFIER   \n",
       "5669  intron_variant,non_coding_transcript_variant  MODIFIER   \n",
       "\n",
       "                  CLIN_SIG clinvar  \n",
       "0                        -     NaN  \n",
       "1                        -     NaN  \n",
       "2                        -     NaN  \n",
       "3                        -     NaN  \n",
       "4                        -     NaN  \n",
       "...                    ...     ...  \n",
       "5665  benign,likely_benign     NaN  \n",
       "5666  benign,likely_benign     NaN  \n",
       "5667  benign,likely_benign     NaN  \n",
       "5668  benign,likely_benign     NaN  \n",
       "5669  benign,likely_benign     NaN  \n",
       "\n",
       "[5670 rows x 22 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_inner_rsid = pd.merge(\n",
    "    df1,\n",
    "    df2,\n",
    "    on=\"rsid\",\n",
    "    how=\"inner\"\n",
    ")\n",
    "df_inner_rsid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "0bb1ebc6-7c54-4243-a024-86612f81212e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAjsAAAHFCAYAAAAUpjivAAAAOnRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjEwLjgsIGh0dHBzOi8vbWF0cGxvdGxpYi5vcmcvwVt1zgAAAAlwSFlzAAAPYQAAD2EBqD+naQAAM6xJREFUeJzt3QucTfX+//HPMGZcYsRgZhiMwhFChHRxiXGLXIrSBUl1xEljjkxymI5jop/boXTqiArxqJBflKZyDR2X5FZSDcYxmnKZcWuI9Xt8vo//3v/Zc6MxM3vv77yej8dqZq393Wt/19o76z3fy9oBjuM4AgAAYKkS3q4AAABAYSLsAAAAqxF2AACA1Qg7AADAaoQdAABgNcIOAACwGmEHAABYjbADAACsRtgBAABWI+wAXjJ//nwJCAiQbdu25fj4PffcI7Vr1/bYpuuDBg36Q6+zadMmmTBhgpw6deqa6lucLFmyRBo2bChlypQx79HOnTvFX+TnMwLYjrAD+JFly5bJuHHj/nDYiY+PJ+xcpV9++UUeeeQRueGGG+STTz6RzZs3S7169fLzdgHwEYHergCAq9esWTO/O10XL140rSOBgf7xz833339v6vzwww9L27ZtvV0dAAWAlh3Aj2Ttorh8+bJMnDhR6tevb7pcKlasKDfffLPMnDnTPK7dV3/961/N71FRUSZ06LJ27Vr386dMmSJ/+tOfJDg4WKpWrSqPPvqoHDlyxON19fuCJ02aJLVq1ZLSpUtLixYtJDExUdq1a2cWF92v7v+dd96RUaNGSfXq1c1+f/jhB9NiMmzYMLnpppvkuuuuM6/VoUMH2bBhg8drHTx40Ozj5ZdflsmTJ5tj1mPT13EFkTFjxkhERISEhIRI7969JTU19arO34oVK+S2226TsmXLSvny5aVTp06m5cZFz+0dd9xhfu/fv7+pR+bjy60rUs/F4MGDpVKlSlKuXDnp0aOH/PTTT3nWRY9Dz4G2ImWlXY56zDExMWb9t99+M+ezadOm5pj1dfQ4Pvzwwyses6uOel4zc71Xrs+Cy2effSZ33323VKhQwZyn22+/XT7//HOPMvpePvHEExIZGWne3ypVqphy+lzAF/nHn1qAxS5duiS///57tu0aMK5Eg4oGmhdeeEHuuusucwH97rvv3F1Wjz/+uJw4cUJmzZolS5culfDwcLNdA4f685//LK+//roMHz7cjBHSC6J2k+kFcMeOHRIaGmrKjR07VhISEswFrk+fPpKcnGz2ra+XUxdPXFycuRi/9tprUqJECXNR1wukGj9+vISFhcmZM2dMt5yGCb2YZg0Vr7zyiglu+lOPRy/2GiJatWolpUqVkjfffFMOHToksbGxpi4aZPKyaNEieeihhyQ6OlreffddycjIMOfP9foacvTYW7ZsKU8//bQJd+3btzcX/SsZMmSICU76Gnpu9P3Q/e7atcsE0JzoMWjrkZ4jPcbMr6P104CjAUppXfV91GPVAHnhwgUTLPS9mDdvngmoBWHBggVmX/fee6+89dZbpo7/+te/pHPnzrJ69WoTgpQGNP18/OMf/zDvv74/un78+PECqQdQ4BwAXjFv3jxNM3kutWrV8niOrg8cONC9fs899zhNmzbN83Vefvlls6+kpCSP7d9++63ZPmzYMI/tX331ldn+/PPPm/UTJ044wcHBTv/+/T3Kbd682ZRr27ate9uaNWvMtrvuuuuKx//77787Fy9edO6++26nd+/e7u1aT91HkyZNnEuXLrm3z5gxw2zv2bOnx35GjhxptqelpeX6WrqfiIgIp3Hjxh77PH36tFO1alWnTZs22Y7hvffeu+r3MHP91Zdffmm2T5w4Mc/n79q1y5R7/fXXPba3bNnSad68+RXP3ZAhQ5xmzZrl+Rlx1THr++86Tv2pzp4961SqVMnp0aOHRzk9X/peaJ1crrvuOnPeAX9BNxbgZW+//bZs3bo12+LqTsmLtkJ88803pntI//JOT0+/6tdds2aN+Zl15o7us0GDBu6uiy1btpiWhX79+nmUa926dbbZYi59+/bNcbu2Ytxyyy2mK0zH8GjLgb7Ot99+m61st27dTKuQi9ZJde/e3aOca/vhw4dzPdb9+/fL0aNHTYtE5n1qd5rWVY/x3Llzkl/aYpRZmzZtTJef6xxrK5223mVeVOPGjaV58+amdcZFz8V//vMfeeyxxzz2+d5775muIq2z69zNnTs3x3OXHzqQXVuPBg4c6FFP7ers0qWL+UyePXvW/RnR7jHtQtVzpy18gC8j7ABephdrHQOTddGxGVei3UX/8z//Yy44Xbt2lcqVK5uuhtyms2fm6nJwdW1lpuNhXI+7flarVi1buZy25bbPadOmmW4z7Yb64IMPTJ31AqoX0vPnz2crr+NSMgsKCspzu3b75PdY9YJ+8uRJyS/tlstpm+t1XV1CmRcXDTU6bki7H5UGHx0H8+CDD7rLaBekhk3twtKuJi2v506fm9dx/xE///yz+Xnfffdlq6uOndLApmHINTVfQ9G///1v012p74l2fx07dqxA6gIUNMbsAH5M/8LXQay66LgJHcfx/PPPmzEWOnZEB5jmRoORSklJkRo1ang8pq0grvE6rnKui2FmenHLqXVHB75mpRdpHccyZ84cj+2nT5+Wwpb5WLPSY9XWnuuvvz7f+8/pIq/bbrzxRvO7jjXScJITDTX6/mlLiY6B0cHdvXr18qiPnjsdYK4hI/O51Ra3K9FWtJzK/vrrrx7rrvdbx3dpq11OXOFWy86YMcMs2qKm46V00LgOFNfp+oCvoWUHsIQOhNW/ynVwrf4F7pp9o60EKmvric6Ecl1IM9OLsnaNuAajakuM7kMvtJlpy4wOEL5aepF21cVFB/Bmng1VWHS2mraK6ADizAO/tVtGW5lcM7Tya+HChdm6hPTcuAZda9jK2nLnoqFGw412Z3700UcmJGXtwtJzpy1YmYOOlrua2ViuMKrnOrOsA7q1i0w/Q/v27cuxpVEXVytaZjVr1jQD3HWAtg5SBnwRLTuAH9MWg0aNGpkLkU7/1Qus/rWt40Xq1q3rHheidDq6dj1ot4Re/HXR2VX6l7y2bGg3mGs2lk4pfvbZZ83ztItCWx50NpZemHWqt05N1xsVardQ5jEwedHZXn//+9/NbCy9f42Oo3nxxRdNi0VOs9EKktZRZ17p2Bqtx5NPPmlaOnR6u7aIvfTSS9e0f+021Blh999/v2lR09lrGq50LNXV0HCjYVJDg7aydezY0eNxrbN2Zen+NNDqa+i51PN/4MCBPPd96623mvdaZ3Lpedb3UGfBbdy40aOcjgXSz4J+RjQs6+u4ZtHpuDD9qa1yaWlpZpbagAEDzC0LdAq/BmRt0dHZYYBP8vYIaaC4cs2S2bp1a46Pd+/e/YqzsaZOnWpmEoWGhjpBQUFOzZo1zQydgwcPejwvLi7OzEYqUaKExwwcnWkzefJkp169ek6pUqXMfh5++GEnOTnZ4/mXL182M4tq1KhhXufmm292PvroIzNLJ/NMpLxmMmVkZDixsbFO9erVndKlSzu33HKLs3z5cnM8mY/TNRtLZ5Flltu+r3QeM9PXa9WqlXn9cuXKmZlgOnPqal4nJ67X/vTTT51HHnnEqVixolOmTBmnW7duzoEDB5yrpe9DZGSk2dfYsWNzLPPSSy85tWvXNjPjGjRo4LzxxhvO+PHjzXPy+oyo77//3omOjnYqVKjgVKlSxRkxYoSzcuVKj8+Cy7p168xnT2dm6WdC3y9dd52P3377zXnqqafMZ0D3p8dbv359Uxed0QX4ogD9j7cDFwD/k5SUZP6y15YaHSdUHOk4G70XjrZsZO6aAuBb6MYCcEXajaE3utMp1XrzO+2C0m4h/V1vqAcAvoywA+CK9CsQdFyK3tdFx7jotHgdfKuzh3Kbfg4AvoJuLAAAYDWmngMAAKsRdgAAgNUIOwAAwGoMUBYx34ujt4zXm2PldJt7AADge/TuOfqVM/odd3nd4JSw8/++G0fvGAsAAPyP3lU863f8ZUbYETEtOq6TpfcNAQAAvi89Pd00Vriu47kh7GT6hmYNOoQdAAD8y5WGoDBAGQAAWI2wAwAArEbYAQAAViPsAAAAqxF2AACA1Qg7AADAaoQdAABgNcIOAACwGmEHAABYjbADAACsRtgBAABWI+wAAACrEXYAAIDVCDsAAMBqhB0AAGC1QG9XAChKtcesvGKZgy91L5K6AACKBi07AADAal4NOwkJCXLrrbdK+fLlpWrVqtKrVy/Zv3+/RxnHcWTChAkSEREhZcqUkXbt2snevXs9ymRkZMiIESMkNDRUypUrJz179pQjR44U8dEAAABf5NWws27dOnn66adly5YtkpiYKL///rtER0fL2bNn3WWmTJki06ZNk9mzZ8vWrVslLCxMOnXqJKdPn3aXGTlypCxbtkwWL14sGzdulDNnzsg999wjly5d8tKRAQAAXxHgaNOJj/jll19MC4+GoLvuusu06miLjoaZ5557zt2KU61aNZk8ebI8+eSTkpaWJlWqVJF33nlH+vfvb8ocPXpUIiMjZdWqVdK5c+crvm56erqEhISYfVWoUKHQjxPew5gdALDH1V6/fWrMjlZWVapUyfxMSkqSY8eOmdYel+DgYGnbtq1s2rTJrG/fvl0uXrzoUUYDUqNGjdxlAABA8eUzs7G0FScmJkbuuOMOE1SUBh2lLTmZ6fqhQ4fcZYKCguT666/PVsb1/Ky0dUiXzMkQAADYyWdadoYPHy67du2Sd999N9tjAQEB2YJR1m1Z5VVGB0Zrs5dr0S4vAABgJ58IOzqTasWKFbJmzRqpUaOGe7sORlZZW2hSU1PdrT1a5sKFC3Ly5Mlcy2QVFxdnusxcS3JyciEcFQAAkOIedrT1RVt0li5dKl988YVERUV5PK7rGmZ0ppaLBhsdwNymTRuz3rx5cylVqpRHmZSUFNmzZ4+7TFY67kcHMmVeAACAnbw6ZkennS9atEg+/PBDc68dVwuOdi3pPXW0G0pnYk2aNEnq1q1rFv29bNmyMmDAAHfZIUOGyKhRo6Ry5cpmcHNsbKw0btxYOnbs6M3DQzHHzC8A8A1eDTtz5swxP/VGgZnNmzdPBg0aZH4fPXq0nD9/XoYNG2a6qlq1aiWffvqpCUcu06dPl8DAQOnXr58pe/fdd8v8+fOlZMmSRXxEAADA1/jUfXa8hfvsFB9F2dpCyw7nEEDh8sv77AAAABQ0wg4AALAaYQcAAFiNsAMAAKxG2AEAAFYj7AAAAKsRdgAAgNUIOwAAwGqEHQAAYDXCDgAAsBphBwAAWI2wAwAArEbYAQAAViPsAAAAqxF2AACA1Qg7AADAaoQdAABgNcIOAACwGmEHAABYjbADAACsRtgBAABWI+wAAACrEXYAAIDVCDsAAMBqhB0AAGC1QG9XAACAwlB7zMorljn4UndOfjFAyw4AALAaYQcAAFiNsAMAAKxG2AEAAFYj7AAAAKsRdgAAgNW8GnbWr18vPXr0kIiICAkICJDly5d7PK7bclpefvlld5l27dple/yBBx7wwtEAAABf5NX77Jw9e1aaNGkigwcPlr59+2Z7PCUlxWP9448/liFDhmQrO3ToUHnxxRfd62XKlCnEWgMA8Mdx359iGna6du1qltyEhYV5rH/44YfSvn17qVOnjsf2smXLZisLAADgV2N2fv75Z1m5cqVp2clq4cKFEhoaKg0bNpTY2Fg5ffq0V+oIAAB8j998XcRbb70l5cuXlz59+nhsf+ihhyQqKsq07OzZs0fi4uLkm2++kcTExFz3lZGRYRaX9PT0Qq07AADwHr8JO2+++aYJNqVLl842XselUaNGUrduXWnRooXs2LFDbrnllhz3lZCQIPHx8YVeZwB2YcwF4J/8ohtrw4YNsn//fnn88cevWFYDTqlSpeTAgQO5ltHWn7S0NPeSnJxcwDUGAAC+wi9adubOnSvNmzc3M7euZO/evXLx4kUJDw/PtUxwcLBZAACA/bwads6cOSM//PCDez0pKUl27twplSpVkpo1a7rH07z33nsyderUbM//8ccfzeDkbt26mQHK+/btk1GjRkmzZs3k9ttvL9JjAeC76H4Cijevhp1t27aZqeQuMTEx5ufAgQNl/vz55vfFixeL4zjy4IMPZnt+UFCQfP755zJz5kwTnCIjI6V79+4yfvx4KVmyZBEeCQAA8FVeDTt692MNMnl54oknzJITDTfr1q0rpNoBAAAb+MUAZQAAgPwi7AAAAKsRdgAAgNUIOwAAwGqEHQAAYDXCDgAAsBphBwAAWI2wAwAArEbYAQAAViPsAAAAqxF2AACA1bz63ViwH982DQDwNlp2AACA1Qg7AADAanRjAQDgI+j6Lxy07AAAAKsRdgAAgNUIOwAAwGqEHQAAYDXCDgAAsBqzsXwAo+8BACg8hB0AwBXxRxn8Gd1YAADAarTsWIa/vgAA8ETLDgAAsBphBwAAWI2wAwAArEbYAQAAViPsAAAAqxF2AACA1Qg7AADAaoQdAABgNW4qCAB+ipuIAn7QsrN+/Xrp0aOHRERESEBAgCxfvtzj8UGDBpntmZfWrVt7lMnIyJARI0ZIaGiolCtXTnr27ClHjhwp4iMBAAC+yqth5+zZs9KkSROZPXt2rmW6dOkiKSkp7mXVqlUej48cOVKWLVsmixcvlo0bN8qZM2fknnvukUuXLhXBEQAAAF/n1W6srl27miUvwcHBEhYWluNjaWlpMnfuXHnnnXekY8eOZtuCBQskMjJSPvvsM+ncuXOh1BsAAPgPnx+gvHbtWqlatarUq1dPhg4dKqmpqe7Htm/fLhcvXpTo6Gj3Nu0Sa9SokWzatCnXfWrXV3p6uscCAADs5NNhR1t9Fi5cKF988YVMnTpVtm7dKh06dDBhRR07dkyCgoLk+uuv93hetWrVzGO5SUhIkJCQEPeiLUEAAMBOPj0bq3///u7ftbWmRYsWUqtWLVm5cqX06dMn1+c5jmMGM+cmLi5OYmJi3OvaskPgAQDATj4ddrIKDw83YefAgQNmXcfyXLhwQU6ePOnRuqNdXW3atMlzHJAuADwxlRmAjfwq7Bw/flySk5NN6FHNmzeXUqVKSWJiovTr189s0xlbe/bskSlTpni5tgCA4uJq/lBAMQ07Ok38hx9+cK8nJSXJzp07pVKlSmaZMGGC9O3b14SbgwcPyvPPP2/up9O7d29TXsfbDBkyREaNGiWVK1c2z4mNjZXGjRu7Z2cBAIDizathZ9u2bdK+fXv3umsczcCBA2XOnDmye/duefvtt+XUqVMm8GjZJUuWSPny5d3PmT59ugQGBpqWnfPnz8vdd98t8+fPl5IlS4ovIO0DAFCMw067du3MYOLcrF69+or7KF26tMyaNcssAAAAfj1mB0DOGFgMAH56nx0AAIBrRdgBAABWoxsLgF9jEgCAK6FlBwAAWI2wAwAArEbYAQAAViPsAAAAqxF2AACA1Qg7AADAaoQdAABgNcIOAACwGmEHAABYjbADAACsRtgBAABWI+wAAACrEXYAAIDVCDsAAMBqhB0AAGA1wg4AALAaYQcAAFgt0NsVAABfUHvMSimuivOxo3igZQcAAFiNsAMAAKxGNxby3ax98KXunD0AgM+jZQcAAFiNsAMAAKxG2AEAAFYj7AAAAKsRdgAAgNUIOwAAwGqEHQAAYDWvhp3169dLjx49JCIiQgICAmT58uXuxy5evCjPPfecNG7cWMqVK2fKPProo3L06FGPfbRr1848N/PywAMPeOFoAACAL/Jq2Dl79qw0adJEZs+ene2xc+fOyY4dO2TcuHHm59KlS+X777+Xnj17Zis7dOhQSUlJcS//+te/iugIAACAr/PqHZS7du1qlpyEhIRIYmKix7ZZs2ZJy5Yt5fDhw1KzZk339rJly0pYWFih1xcAUPi4gzuK9ZidtLQ0001VsWJFj+0LFy6U0NBQadiwocTGxsrp06fz3E9GRoakp6d7LAAAwE5+891Yv/32m4wZM0YGDBggFSpUcG9/6KGHJCoqyrTs7NmzR+Li4uSbb77J1iqUWUJCgsTHxxdRzQEAgDf5RdjRwco66Pjy5cvy6quvZhuv49KoUSOpW7eutGjRwozzueWWW3LcnwaimJgY97q27ERGRhbiEQAAAG8J9Ieg069fP0lKSpIvvvjCo1UnJxpwSpUqJQcOHMg17AQHB5sFAAB/w5gmy8KOK+hocFmzZo1Urlz5is/Zu3eveV54eHiR1BEAAPg2r4adM2fOyA8//OBe19abnTt3SqVKlcx9de677z7THfXRRx/JpUuX5NixY6acPh4UFCQ//vijGZzcrVs3M0B53759MmrUKGnWrJncfvvtXjwyAADgK7wadrZt2ybt27d3r7vG0QwcOFAmTJggK1asMOtNmzb1eJ628ujNBDXwfP755zJz5kwTnHTcTffu3WX8+PFSsmTJIj4aAADgi7wadjSwOI6T6+N5PaY03Kxbt64QagYAAGzhV/fZAQAA+KMIOwAAwGqEHQAAYDWfnnoOwPdwjw8A/oaWHQAAYDXCDgAAsFq+wk6dOnXk+PHj2bafOnXKPAYAAODXYefgwYPmjsZZZWRkyH//+9+CqBcAAEDRD1B23dFYrV69WkJCQtzrGn70bsa1a9cumJoBAAAUddjp1auX+RkQEGC+0iEz/aZxDTpTp04tiHrBklk5AAD4Vdi5fPmy+RkVFSVbt241X74JAABg3X129NvJAQAArL6poI7P0SU1NdXd4uPy5ptvFkTdgD+EbjUAQIGFnfj4eHnxxRelRYsWEh4ebsbwAAAAWBN2XnvtNZk/f7488sgjBV8jAAAAb4edCxcuSJs2bQqyHgCAYoDuZvjNTQUff/xxWbRoUcHXBgAAwBdadn777Td5/fXX5bPPPpObb77Z3GMns2nTphVU/QBYir/wAfh02Nm1a5c0bdrU/L5nzx6PxxisDAAA/D7srFmzpuBrAgAA4CtjdgAAAKxu2Wnfvn2e3VVffPHFtdQJAADAu2HHNV7H5eLFi7Jz504zfifrF4QCAP744OyDL3XntAHeDDvTp0/PcfuECRPkzJkz11onAACAAlOgY3YefvhhvhcLAADYG3Y2b94spUuXLshdAgAAFH03Vp8+fTzWHceRlJQU2bZtm4wbN+7aagQAAODtsBMSEuKxXqJECalfv775JvTo6OiCqhsAAIB3ws68efOu/ZUBAAB8Ney4bN++Xb799ltzz52bbrpJmjVrVnA1AwAA8FbYSU1NlQceeEDWrl0rFStWNGN20tLSzM0GFy9eLFWqVCmIugEAAHhnNtaIESMkPT1d9u7dKydOnJCTJ0+aGwrqtr/85S/XXisAAABvhp1PPvlE5syZIw0aNHBv026sV155RT7++OOr3s/69eulR48eEhERYbrCli9f7vG4thjpjQr18TJlyki7du1MwMosIyPDhK/Q0FApV66c9OzZU44cOZKfwwIAABbKVzfW5cuXpVSpUtm26zZ97GqdPXtWmjRpIoMHD5a+fftme3zKlCkybdo0mT9/vtSrV08mTpwonTp1kv3790v58uVNmZEjR8r//u//mu6zypUry6hRo+See+4x44lKliyZn8MDigxfGwAAPtqy06FDB3nmmWfk6NGj7m3//e9/5dlnn5W77777qvfTtWtXE2Cy3rfH1aozY8YMGTt2rHm8UaNG8tZbb8m5c+dk0aJFpoyOE5o7d65MnTpVOnbsaAZIL1iwQHbv3i2fffZZfg4NAABYJl9hZ/bs2XL69GmpXbu23HDDDXLjjTdKVFSU2TZr1qwCqVhSUpIcO3bM4749wcHB0rZtW9m0aZNZ19Yb/RLSzGW0y0uDkatMTrTrS8cXZV4AAICd8tWNFRkZKTt27JDExET57rvvTCuMjtnR1pWCokFHVatWzWO7rh86dMhdJigoSK6//vpsZVzPz0lCQoLEx8cXWF0BAIAlLTtffPGFCTWulhAdP6ODg3UG1q233ioNGzaUDRs2FGgFdeByZhqssm7L6kpl4uLiTBeYa0lOTi6w+gIAAD9u2dExNEOHDpUKFSrk+BUSTz75pBlQfOedd15zxcLCwsxPbaEJDw/3uMePq7VHy1y4cMFMfc/cuqNl2rRpk+u+tTtMFwAAYP9Eiz/UsvPNN99Ily5dcn1cx87oOJqCoGOANMxoV5mLBpt169a5g0zz5s3NDLDMZfQLSfWeP3mFHQAAUHz8oZadn3/+Occp5+6dBQbKL7/8ctX7O3PmjPzwww8eg5J37twplSpVkpo1a5pp5ZMmTZK6deuaRX8vW7asDBgwwN2aNGTIEDPdXKed6/NiY2OlcePGBTp+CAAAFJOwU716dTOtW2df5WTXrl0eXU5Xsm3bNvMVEy4xMTHm58CBA829dUaPHi3nz5+XYcOGma6qVq1ayaeffuq+x46aPn26CVn9+vUzZXXquz6Xe+wAAHBtrqaLyrqw061bN/nb3/5m7o9TunRpj8c0aIwfP97c0O9q6R2RdTBxbnSQsd5BWZfcaD10untBTXkHAAB2+UNh54UXXpClS5eauxkPHz5c6tevbwKJfvO5flXEpUuXzE0AAQAA/DLs6CwovVnfn//8ZzN929Uqo4Gnc+fO8uqrr2a7Lw4AAIBf3VSwVq1asmrVKjOGRgcXa+DRwcNZb+wHAADgt3dQVhpu9EaCAAAA1n03FgAAgL8g7AAAAKsRdgAAgNUIOwAAwGqEHQAAYDXCDgAAsBphBwAAWI2wAwAArJbvmwoCKBq2fOsw/hjed6Dg0LIDAACsRtgBAABWI+wAAACrEXYAAIDVGKAM5AODRwHAf9CyAwAArEbYAQAAViPsAAAAqxF2AACA1Qg7AADAaoQdAABgNcIOAACwGmEHAABYjbADAACsRtgBAABW4+si/ARfTwAAQP7QsgMAAKxG2AEAAFYj7AAAAKv5fNipXbu2BAQEZFuefvpp8/igQYOyPda6dWtvVxsAAPgInx+gvHXrVrl06ZJ7fc+ePdKpUye5//773du6dOki8+bNc68HBQUVeT0BAIBv8vmwU6VKFY/1l156SW644QZp27ate1twcLCEhYV5oXYAAMDX+Xw3VmYXLlyQBQsWyGOPPWa6q1zWrl0rVatWlXr16snQoUMlNTU1z/1kZGRIenq6xwIAAOzk8y07mS1fvlxOnTplxum4dO3a1XRp1apVS5KSkmTcuHHSoUMH2b59u2nxyUlCQoLEx8cXYc1xrbjPEPwFn1XA9/hV2Jk7d64JNxEREe5t/fv3d//eqFEjadGihQk+K1eulD59+uS4n7i4OImJiXGva8tOZGRkIdceAAB4g9+EnUOHDslnn30mS5cuzbNceHi4CTsHDhzItYy2+OTW6gMAAOziN2N2dLaVjsvp3r17nuWOHz8uycnJJvQAAAD4RcvO5cuXTdgZOHCgBAb+/yqfOXNGJkyYIH379jXh5uDBg/L8889LaGio9O7d26t1BgAUHsZGwbqwo91Xhw8fNrOwMitZsqTs3r1b3n77bTNwWQNP+/btZcmSJVK+fHmv1RcAAPgOvwg70dHR4jhOtu1lypSR1atXe6VOAADAP/jNmB0AAID8IOwAAACrEXYAAIDVCDsAAMBqfjFAGShKTGkFALvQsgMAAKxG2AEAAFYj7AAAAKsRdgAAgNUIOwAAwGqEHQAAYDXCDgAAsBphBwAAWI2wAwAArEbYAQAAViPsAAAAq/HdWAAAFEO1x6yU4oKWHQAAYDXCDgAAsBphBwAAWI2wAwAArMYAZQAALFOcBh9fDVp2AACA1Qg7AADAaoQdAABgNcIOAACwGmEHAABYjbADAACsRtgBAABWI+wAAACrEXYAAIDVCDsAAMBqPh12JkyYIAEBAR5LWFiY+3HHcUyZiIgIKVOmjLRr10727t3r1ToDAADf4tNhRzVs2FBSUlLcy+7du92PTZkyRaZNmyazZ8+WrVu3miDUqVMnOX36tFfrDAAAfIfPh53AwEATYlxLlSpV3K06M2bMkLFjx0qfPn2kUaNG8tZbb8m5c+dk0aJF3q42AADwET7/recHDhww3VTBwcHSqlUrmTRpktSpU0eSkpLk2LFjEh0d7S6rZdq2bSubNm2SJ598Mtd9ZmRkmMUlPT290I8DKE74xmUAvsSnW3Y03Lz99tuyevVqeeONN0y4adOmjRw/ftz8rqpVq+bxHF13PZabhIQECQkJcS+RkZGFehwAAMB7fDrsdO3aVfr27SuNGzeWjh07ysqVK8127a5y0UHLmWn3VtZtWcXFxUlaWpp7SU5OLqQjAAAA3ubTYSercuXKmeCjXVuuWVlZW3FSU1OztfZkpd1dFSpU8FgAAICd/Crs6Dibb7/9VsLDwyUqKsoEnsTERPfjFy5ckHXr1pmuLgAAAJ8foBwbGys9evSQmjVrmhabiRMnmsHEAwcONF1VI0eONAOW69ataxb9vWzZsjJgwABvVx0AAPgInw47R44ckQcffFB+/fVXM+W8devWsmXLFqlVq5Z5fPTo0XL+/HkZNmyYnDx50gxo/vTTT6V8+fLerjoAAPARPh12Fi9enOfj2rqjd1DWBQAAwO/H7AAAAPxRhB0AAGA1n+7GAgBcG+5mDdCyAwAALEc3FgAAsBphBwAAWI2wAwAArEbYAQAAViPsAAAAqzH1HABQIJjmDl9Fyw4AALAaLTsAAOSBFiv/R8sOAACwGmEHAABYjbADAACsRtgBAABWI+wAAACrEXYAAIDVmHoOACi2mFZePBB24HX8YwMAKEx0YwEAAKsRdgAAgNUIOwAAwGqEHQAAYDXCDgAAsBphBwAAWI2wAwAArEbYAQAAViPsAAAAqxF2AACA1Qg7AADAaoQdAABgNZ8OOwkJCXLrrbdK+fLlpWrVqtKrVy/Zv3+/R5lBgwZJQECAx9K6dWuv1RkAAPgWnw4769atk6efflq2bNkiiYmJ8vvvv0t0dLScPXvWo1yXLl0kJSXFvaxatcprdQYAAL4lUHzYJ5984rE+b94808Kzfft2ueuuu9zbg4ODJSwszAs1BAAAvs6nW3aySktLMz8rVarksX3t2rUmBNWrV0+GDh0qqampXqohAADwNT7dspOZ4zgSExMjd9xxhzRq1Mi9vWvXrnL//fdLrVq1JCkpScaNGycdOnQwrT/a4pOTjIwMs7ikp6cXyTEAAICi5zdhZ/jw4bJr1y7ZuHGjx/b+/fu7f9cQ1KJFCxN8Vq5cKX369Ml14HN8fHyh1xkAAHifX3RjjRgxQlasWCFr1qyRGjVq5Fk2PDzchJ0DBw7kWiYuLs50ibmW5OTkQqg1AADwBYG+3nWlQWfZsmVmXE5UVNQVn3P8+HETXjT05Ea7t3Lr4gIAAHbx6ZYdnXa+YMECWbRokbnXzrFjx8xy/vx58/iZM2ckNjZWNm/eLAcPHjSBqEePHhIaGiq9e/f2dvUBAIAP8OmWnTlz5pif7dq1yzYFXW8mWLJkSdm9e7e8/fbbcurUKdOa0759e1myZIkJRwAAAD7fjZWXMmXKyOrVq4usPraoPWalt6sAAECR8eluLAAAgGtF2AEAAFYj7AAAAKsRdgAAgNUIOwAAwGqEHQAAYDXCDgAAsJpP32cHQMHh/koAiitadgAAgNUIOwAAwGqEHQAAYDXCDgAAsBphBwAAWI2wAwAArEbYAQAAViPsAAAAqxF2AACA1Qg7AADAaoQdAABgNcIOAACwGmEHAABYjbADAACsRtgBAABWI+wAAACrEXYAAIDVCDsAAMBqhB0AAGA1wg4AALAaYQcAAFiNsAMAAKxG2AEAAFYj7AAAAKtZE3ZeffVViYqKktKlS0vz5s1lw4YN3q4SAADwAVaEnSVLlsjIkSNl7Nix8vXXX8udd94pXbt2lcOHD3u7agAAwMusCDvTpk2TIUOGyOOPPy4NGjSQGTNmSGRkpMyZM8fbVQMAAF7m92HnwoULsn37domOjvbYruubNm3yWr0AAIBvCBQ/9+uvv8qlS5ekWrVqHtt1/dixYzk+JyMjwywuaWlp5md6enqB1+9yxrkC3ycAAP4kvRCur5n36ziO3WHHJSAgwGNdDzzrNpeEhASJj4/Ptl27vgAAQMEKmSGF6vTp0xISEmJv2AkNDZWSJUtma8VJTU3N1trjEhcXJzExMe71y5cvy4kTJ6Ry5cq5BqT8Jk4NUMnJyVKhQoUC2y84197CZ5rzbBM+z/5/nrVhQ4NOREREnuX8PuwEBQWZqeaJiYnSu3dv93Zdv/fee3N8TnBwsFkyq1ixYqHVUd9cwk7R4Fxznm3C55nzbJMKhXQtzKtFx5qwo7SV5pFHHpEWLVrIbbfdJq+//rqZdv7UU095u2oAAMDLrAg7/fv3l+PHj8uLL74oKSkp0qhRI1m1apXUqlXL21UDAABeZkXYUcOGDTOLL9GusvHjx2frMgPn2l/xmeY824TPc/E5zwHOleZrAQAA+DG/v6kgAABAXgg7AADAaoQdAABgNcIOAACwGmHnGr366qsSFRUlpUuXNjc33LBhQ57l161bZ8pp+Tp16shrr712rVUoFv7IeV66dKl06tRJqlSpYm5gpfdeWr16dZHWtzh9pl2+/PJLCQwMlKZNmxZ6HYvjedbv8xs7dqy5pYbOarnhhhvkzTffLLL6FpfzvHDhQmnSpImULVtWwsPDZfDgwebWJsjd+vXrpUePHuYuxvotBMuXL5crKfJroc7GQv4sXrzYKVWqlPPGG284+/btc5555hmnXLlyzqFDh3Is/9NPPzlly5Y15bS8Pk+f//777/MWFOB51scnT57s/Oc//3G+//57Jy4uzjx/x44dnOcC/ky7nDp1yqlTp44THR3tNGnShPNcCOe5Z8+eTqtWrZzExEQnKSnJ+eqrr5wvv/ySc12A53nDhg1OiRIlnJkzZ5p/r3W9YcOGTq9evTjPeVi1apUzduxY54MPPtDZ3c6yZcvyKu6VayFh5xq0bNnSeeqppzy2/elPf3LGjBmTY/nRo0ebxzN78sknndatW19LNaz3R89zTm666SYnPj6+EGpnl/ye6/79+zsvvPCCM378eMJOIZznjz/+2AkJCXGOHz9+NbtHPs/zyy+/bEJ7Zv/85z+dGjVqcE6v0tWEHW9cC+nGyqcLFy7I9u3bJTo62mO7rm/atCnH52zevDlb+c6dO8u2bdvk4sWL+a2K1fJznrPSL3rVL4qrVKlSIdWyeJ/refPmyY8//mhuGobCOc8rVqwwX4czZcoUqV69utSrV09iY2Pl/PnznPICPM9t2rSRI0eOmDvw63X7559/lvfff1+6d+/OeS5A3rgWWnMH5aL266+/yqVLl7J9s7quZ/0GdhfdnlP533//3exP+4dx7ec5q6lTp8rZs2elX79+nN4C/kwfOHBAxowZY8ZB6HgdFM55/umnn2Tjxo1mfMOyZcvMPvSO8SdOnGDcTgGeZw07OmZHv4Lot99+M/829+zZU2bNmsVHuwB541pIy8410sFYmelfA1m3Xal8TttxbefZ5d1335UJEybIkiVLpGrVqpzWAjzXeiEZMGCAxMfHm5YGFN5nWlsn9TG9ELds2VK6desm06ZNk/nz59O6U4Dned++ffKXv/xF/va3v5lWoU8++USSkpL4UulCUNTXQv4Uy6fQ0FApWbJktr8QUlNTsyVWl7CwsBzL61/ElStXzm9VrJaf8+yiAWfIkCHy3nvvSceOHQu5psXvXGvXoDY7f/311zJ8+HD3RVn/0dLP9KeffiodOnQosvrb/JnWv3S1+yokJMS9rUGDBuZca7dL3bp1C73exeE8JyQkyO233y5//etfzfrNN98s5cqVkzvvvFMmTpxI63sB8ca1kJadfAoKCjLT5hITEz2267o2heZEp0BnLa8XBO2LL1WqVH6rYrX8nGdXi86gQYNk0aJF9LcX0rnWaf27d++WnTt3upennnpK6tevb35v1arV1b50sZKfz7RegI8ePSpnzpxxb/v++++lRIkSUqNGjUKvc3E5z+fOnTPnNDMNTIqvkSw4XrkWFtrQ52I0rXHu3Llm+tzIkSPNtMaDBw+ax3XE/yOPPJJtut2zzz5ryuvzmHpe8Od50aJFTmBgoPPKK684KSkp7kWnR6Ngz3VWzMYqnPN8+vRpMyPovvvuc/bu3eusW7fOqVu3rvP444/zkS7A8zxv3jzzb8err77q/Pjjj87GjRudFi1amFldyJ1+Pr/++muzaKyYNm2a+d01xd8XroWEnWukF9RatWo5QUFBzi233GL+EXIZOHCg07ZtW4/ya9eudZo1a2bK165d25kzZ861VqFY+CPnWX/X/+GyLloOBXuusyLsFM5nWn377bdOx44dnTJlypjgExMT45w7d46PdAGfZ51qrreq0PMcHh7uPPTQQ86RI0c4z3lYs2ZNnv/m+sK1MED/UzhtRgAAAN7HmB0AAGA1wg4AALAaYQcAAFiNsAMAAKxG2AEAAFYj7AAAAKsRdgAAgNUIOwCKHf1y2KZNm3q7GgCKCGEHAABYjbADAACsRtgB4FXt2rWT4cOHm6VixYpSuXJleeGFF3L8lum0tDQpU6aMfPLJJx7bly5dKuXKlXN/K/hzzz0n9erVk7Jly0qdOnVk3LhxcvHixTzrMHLkSI9tvXr1kkGDBrnXL1y4IKNHj5bq1aub19JvdV+7dm0BnAEAhY2wA8Dr3nrrLQkMDJSvvvpK/vnPf8r06dPl3//+d7ZyISEh0r17d1m4cKHH9kWLFsm9994r1113nVkvX768zJ8/X/bt2yczZ86UN954w+zzWgwePFi+/PJLWbx4sezatUvuv/9+6dKlixw4cOCa9gug8AUWwWsAQJ4iIyNNGAkICJD69evL7t27zfrQoUOzlX3ooYfk0UcflXPnzpmWm/T0dFm5cqV88MEH7jLaMuRSu3ZtGTVqlCxZssS0zOTHjz/+KO+++64cOXJEIiIizLbY2FjTwjRv3jyZNGkS7zDgw2jZAeB1rVu3NkHH5bbbbjMtJv/4xz9Ma41rOXz4sGnZ0VagFStWmLIacrQlJzo62v38999/X+644w4JCwszz9NuLH1ufu3YscN0q2nXWOb6rFu3zgQhAL6Nlh0APuupp56S/v37u9e1VUWDzn333We6rh544AHzU8vodrVlyxazPT4+Xjp37my6vrTraerUqbm+TokSJbKNEco8xufy5ctSsmRJ2b59u/mZmavrDIDvIuwA8DoNKFnX69atawYr65JTV5a25Ozdu1fWrFkjf//7392P6biaWrVqydixY93bDh06lOfrV6lSRVJSUtzrly5dkj179kj79u3NerNmzcy21NRUufPOO6/pWAEUPbqxAHhdcnKyxMTEyP79+83YmFmzZskzzzyTa/m2bdtKtWrVTOjRMTnaDeZy4403mi4rbc3RLiYd8Lxs2bI8X79Dhw5m3I8u3333nQwbNkxOnTrlfly7r1xjhXTmV1JSkmzdulUmT54sq1atKqCzAKCwEHYAeJ2GiPPnz0vLli3l6aeflhEjRsgTTzyRa3kd3/Pggw/KN998Y0JIZjor69lnnzVT2fUuyZs2bTJjdvLy2GOPycCBA009NEhFRUW5W3VcdCCyPq6DnXUQdc+ePc3sMR1cDcC3BTg53cwCAIqI3uNGQ8mMGTM45wAKBS07AADAaoQdAABgNbqxAACA1WjZAQAAViPsAAAAqxF2AACA1Qg7AADAaoQdAABgNcIOAACwGmEHAABYjbADAACsRtgBAABis/8D75WFAPby2MEAAAAASUVORK5CYII=",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "\n",
    "plt.hist(df_inner_rsid[\"p_value\"], bins=50)\n",
    "plt.xlabel(\"p-value\")\n",
    "plt.ylabel(\"Count\")\n",
    "plt.title(\"Histogram of p-values\")\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "d2e2f327-cc6e-467d-8da4-0a7b63aba7c9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "np.float64(0.0003049)"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_inner_rsid[\"p_value\"].min()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "f49e1234-fb29-43e3-bbc1-8698048d78f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_final_selected = df_inner_rsid[df_inner_rsid[\"p_value\"] <= 0.001]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "d4a9061e-3eb7-49dc-ad32-3e833ddeb257",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>base_pair_location</th>\n",
       "      <th>effect_allele</th>\n",
       "      <th>other_allele</th>\n",
       "      <th>beta</th>\n",
       "      <th>standard_error</th>\n",
       "      <th>effect_allele_frequency</th>\n",
       "      <th>p_value</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>hm_coordinate_conversion</th>\n",
       "      <th>...</th>\n",
       "      <th>›#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>SYMBOL.1</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>clinvar</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>749</th>\n",
       "      <td>2</td>\n",
       "      <td>237340531</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.4805</td>\n",
       "      <td>0.1422</td>\n",
       "      <td>0.005393</td>\n",
       "      <td>0.000727</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238249174_G/A</td>\n",
       "      <td>2:238249174-238249174</td>\n",
       "      <td>rs201460305</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>750</th>\n",
       "      <td>2</td>\n",
       "      <td>237340531</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.4805</td>\n",
       "      <td>0.1422</td>\n",
       "      <td>0.005393</td>\n",
       "      <td>0.000727</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238249174_G/A</td>\n",
       "      <td>2:238249174-238249174</td>\n",
       "      <td>rs201460305</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>751</th>\n",
       "      <td>2</td>\n",
       "      <td>237340531</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.4805</td>\n",
       "      <td>0.1422</td>\n",
       "      <td>0.005393</td>\n",
       "      <td>0.000727</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238249174_G/A</td>\n",
       "      <td>2:238249174-238249174</td>\n",
       "      <td>rs201460305</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>752</th>\n",
       "      <td>2</td>\n",
       "      <td>237340531</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.4805</td>\n",
       "      <td>0.1422</td>\n",
       "      <td>0.005393</td>\n",
       "      <td>0.000727</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238249174_G/A</td>\n",
       "      <td>2:238249174-238249174</td>\n",
       "      <td>rs201460305</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>753</th>\n",
       "      <td>2</td>\n",
       "      <td>237340531</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.4805</td>\n",
       "      <td>0.1422</td>\n",
       "      <td>0.005393</td>\n",
       "      <td>0.000727</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238249174_G/A</td>\n",
       "      <td>2:238249174-238249174</td>\n",
       "      <td>rs201460305</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>754</th>\n",
       "      <td>2</td>\n",
       "      <td>237340531</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.4805</td>\n",
       "      <td>0.1422</td>\n",
       "      <td>0.005393</td>\n",
       "      <td>0.000727</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238249174_G/A</td>\n",
       "      <td>2:238249174-238249174</td>\n",
       "      <td>rs201460305</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>755</th>\n",
       "      <td>2</td>\n",
       "      <td>237340531</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.4805</td>\n",
       "      <td>0.1422</td>\n",
       "      <td>0.005393</td>\n",
       "      <td>0.000727</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238249174_G/A</td>\n",
       "      <td>2:238249174-238249174</td>\n",
       "      <td>rs201460305</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>756</th>\n",
       "      <td>2</td>\n",
       "      <td>237340531</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.4805</td>\n",
       "      <td>0.1422</td>\n",
       "      <td>0.005393</td>\n",
       "      <td>0.000727</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238249174_G/A</td>\n",
       "      <td>2:238249174-238249174</td>\n",
       "      <td>rs201460305</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>811</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>812</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>813</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>814</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>815</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>816</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>817</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>818</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>819</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>820</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>821</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>822</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>823</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>824</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>825</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>826</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>827</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>828</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>829</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>830</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>831</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>832</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>833</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>834</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>835</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>836</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>837</th>\n",
       "      <td>2</td>\n",
       "      <td>237379158</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.3086</td>\n",
       "      <td>0.3741</td>\n",
       "      <td>0.000345</td>\n",
       "      <td>0.000469</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>2_238287801_G/A</td>\n",
       "      <td>2:238287801-238287801</td>\n",
       "      <td>rs146291186</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>ENSG00000163359</td>\n",
       "      <td>COL6A3</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5626</th>\n",
       "      <td>22</td>\n",
       "      <td>36301574</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.2025</td>\n",
       "      <td>0.3330</td>\n",
       "      <td>0.001547</td>\n",
       "      <td>0.000305</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>22_36697620_G/A</td>\n",
       "      <td>22:36697620-36697620</td>\n",
       "      <td>rs142467329</td>\n",
       "      <td>MYH9</td>\n",
       "      <td>ENSG00000100345</td>\n",
       "      <td>MYH9</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5627</th>\n",
       "      <td>22</td>\n",
       "      <td>36301574</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.2025</td>\n",
       "      <td>0.3330</td>\n",
       "      <td>0.001547</td>\n",
       "      <td>0.000305</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>22_36697620_G/A</td>\n",
       "      <td>22:36697620-36697620</td>\n",
       "      <td>rs142467329</td>\n",
       "      <td>MYH9</td>\n",
       "      <td>ENSG00000100345</td>\n",
       "      <td>MYH9</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5628</th>\n",
       "      <td>22</td>\n",
       "      <td>36301574</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.2025</td>\n",
       "      <td>0.3330</td>\n",
       "      <td>0.001547</td>\n",
       "      <td>0.000305</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>22_36697620_G/A</td>\n",
       "      <td>22:36697620-36697620</td>\n",
       "      <td>rs142467329</td>\n",
       "      <td>MYH9</td>\n",
       "      <td>ENSG00000100345</td>\n",
       "      <td>MYH9</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5629</th>\n",
       "      <td>22</td>\n",
       "      <td>36301574</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.2025</td>\n",
       "      <td>0.3330</td>\n",
       "      <td>0.001547</td>\n",
       "      <td>0.000305</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>22_36697620_G/A</td>\n",
       "      <td>22:36697620-36697620</td>\n",
       "      <td>rs142467329</td>\n",
       "      <td>MYH9</td>\n",
       "      <td>ENSG00000100345</td>\n",
       "      <td>MYH9</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5630</th>\n",
       "      <td>22</td>\n",
       "      <td>36301574</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.2025</td>\n",
       "      <td>0.3330</td>\n",
       "      <td>0.001547</td>\n",
       "      <td>0.000305</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>22_36697620_G/A</td>\n",
       "      <td>22:36697620-36697620</td>\n",
       "      <td>rs142467329</td>\n",
       "      <td>MYH9</td>\n",
       "      <td>ENSG00000100345</td>\n",
       "      <td>MYH9</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5631</th>\n",
       "      <td>22</td>\n",
       "      <td>36301574</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.2025</td>\n",
       "      <td>0.3330</td>\n",
       "      <td>0.001547</td>\n",
       "      <td>0.000305</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>22_36697620_G/A</td>\n",
       "      <td>22:36697620-36697620</td>\n",
       "      <td>rs142467329</td>\n",
       "      <td>MYH9</td>\n",
       "      <td>ENSG00000100345</td>\n",
       "      <td>MYH9</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5632</th>\n",
       "      <td>22</td>\n",
       "      <td>36301574</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.2025</td>\n",
       "      <td>0.3330</td>\n",
       "      <td>0.001547</td>\n",
       "      <td>0.000305</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>22_36697620_G/A</td>\n",
       "      <td>22:36697620-36697620</td>\n",
       "      <td>rs142467329</td>\n",
       "      <td>MYH9</td>\n",
       "      <td>ENSG00000100345</td>\n",
       "      <td>MYH9</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5633</th>\n",
       "      <td>22</td>\n",
       "      <td>36301574</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.2025</td>\n",
       "      <td>0.3330</td>\n",
       "      <td>0.001547</td>\n",
       "      <td>0.000305</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lo</td>\n",
       "      <td>...</td>\n",
       "      <td>22_36697620_G/A</td>\n",
       "      <td>22:36697620-36697620</td>\n",
       "      <td>rs142467329</td>\n",
       "      <td>MYH9</td>\n",
       "      <td>ENSG00000100345</td>\n",
       "      <td>MYH9</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>43 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chromosome  base_pair_location effect_allele other_allele    beta  \\\n",
       "749            2           237340531             A            G  0.4805   \n",
       "750            2           237340531             A            G  0.4805   \n",
       "751            2           237340531             A            G  0.4805   \n",
       "752            2           237340531             A            G  0.4805   \n",
       "753            2           237340531             A            G  0.4805   \n",
       "754            2           237340531             A            G  0.4805   \n",
       "755            2           237340531             A            G  0.4805   \n",
       "756            2           237340531             A            G  0.4805   \n",
       "811            2           237379158             A            G  1.3086   \n",
       "812            2           237379158             A            G  1.3086   \n",
       "813            2           237379158             A            G  1.3086   \n",
       "814            2           237379158             A            G  1.3086   \n",
       "815            2           237379158             A            G  1.3086   \n",
       "816            2           237379158             A            G  1.3086   \n",
       "817            2           237379158             A            G  1.3086   \n",
       "818            2           237379158             A            G  1.3086   \n",
       "819            2           237379158             A            G  1.3086   \n",
       "820            2           237379158             A            G  1.3086   \n",
       "821            2           237379158             A            G  1.3086   \n",
       "822            2           237379158             A            G  1.3086   \n",
       "823            2           237379158             A            G  1.3086   \n",
       "824            2           237379158             A            G  1.3086   \n",
       "825            2           237379158             A            G  1.3086   \n",
       "826            2           237379158             A            G  1.3086   \n",
       "827            2           237379158             A            G  1.3086   \n",
       "828            2           237379158             A            G  1.3086   \n",
       "829            2           237379158             A            G  1.3086   \n",
       "830            2           237379158             A            G  1.3086   \n",
       "831            2           237379158             A            G  1.3086   \n",
       "832            2           237379158             A            G  1.3086   \n",
       "833            2           237379158             A            G  1.3086   \n",
       "834            2           237379158             A            G  1.3086   \n",
       "835            2           237379158             A            G  1.3086   \n",
       "836            2           237379158             A            G  1.3086   \n",
       "837            2           237379158             A            G  1.3086   \n",
       "5626          22            36301574             A            G  1.2025   \n",
       "5627          22            36301574             A            G  1.2025   \n",
       "5628          22            36301574             A            G  1.2025   \n",
       "5629          22            36301574             A            G  1.2025   \n",
       "5630          22            36301574             A            G  1.2025   \n",
       "5631          22            36301574             A            G  1.2025   \n",
       "5632          22            36301574             A            G  1.2025   \n",
       "5633          22            36301574             A            G  1.2025   \n",
       "\n",
       "      standard_error  effect_allele_frequency   p_value  variant_id  \\\n",
       "749           0.1422                 0.005393  0.000727         NaN   \n",
       "750           0.1422                 0.005393  0.000727         NaN   \n",
       "751           0.1422                 0.005393  0.000727         NaN   \n",
       "752           0.1422                 0.005393  0.000727         NaN   \n",
       "753           0.1422                 0.005393  0.000727         NaN   \n",
       "754           0.1422                 0.005393  0.000727         NaN   \n",
       "755           0.1422                 0.005393  0.000727         NaN   \n",
       "756           0.1422                 0.005393  0.000727         NaN   \n",
       "811           0.3741                 0.000345  0.000469         NaN   \n",
       "812           0.3741                 0.000345  0.000469         NaN   \n",
       "813           0.3741                 0.000345  0.000469         NaN   \n",
       "814           0.3741                 0.000345  0.000469         NaN   \n",
       "815           0.3741                 0.000345  0.000469         NaN   \n",
       "816           0.3741                 0.000345  0.000469         NaN   \n",
       "817           0.3741                 0.000345  0.000469         NaN   \n",
       "818           0.3741                 0.000345  0.000469         NaN   \n",
       "819           0.3741                 0.000345  0.000469         NaN   \n",
       "820           0.3741                 0.000345  0.000469         NaN   \n",
       "821           0.3741                 0.000345  0.000469         NaN   \n",
       "822           0.3741                 0.000345  0.000469         NaN   \n",
       "823           0.3741                 0.000345  0.000469         NaN   \n",
       "824           0.3741                 0.000345  0.000469         NaN   \n",
       "825           0.3741                 0.000345  0.000469         NaN   \n",
       "826           0.3741                 0.000345  0.000469         NaN   \n",
       "827           0.3741                 0.000345  0.000469         NaN   \n",
       "828           0.3741                 0.000345  0.000469         NaN   \n",
       "829           0.3741                 0.000345  0.000469         NaN   \n",
       "830           0.3741                 0.000345  0.000469         NaN   \n",
       "831           0.3741                 0.000345  0.000469         NaN   \n",
       "832           0.3741                 0.000345  0.000469         NaN   \n",
       "833           0.3741                 0.000345  0.000469         NaN   \n",
       "834           0.3741                 0.000345  0.000469         NaN   \n",
       "835           0.3741                 0.000345  0.000469         NaN   \n",
       "836           0.3741                 0.000345  0.000469         NaN   \n",
       "837           0.3741                 0.000345  0.000469         NaN   \n",
       "5626          0.3330                 0.001547  0.000305         NaN   \n",
       "5627          0.3330                 0.001547  0.000305         NaN   \n",
       "5628          0.3330                 0.001547  0.000305         NaN   \n",
       "5629          0.3330                 0.001547  0.000305         NaN   \n",
       "5630          0.3330                 0.001547  0.000305         NaN   \n",
       "5631          0.3330                 0.001547  0.000305         NaN   \n",
       "5632          0.3330                 0.001547  0.000305         NaN   \n",
       "5633          0.3330                 0.001547  0.000305         NaN   \n",
       "\n",
       "     hm_coordinate_conversion  ...  ›#Uploaded_variation  \\\n",
       "749                        lo  ...       2_238249174_G/A   \n",
       "750                        lo  ...       2_238249174_G/A   \n",
       "751                        lo  ...       2_238249174_G/A   \n",
       "752                        lo  ...       2_238249174_G/A   \n",
       "753                        lo  ...       2_238249174_G/A   \n",
       "754                        lo  ...       2_238249174_G/A   \n",
       "755                        lo  ...       2_238249174_G/A   \n",
       "756                        lo  ...       2_238249174_G/A   \n",
       "811                        lo  ...       2_238287801_G/A   \n",
       "812                        lo  ...       2_238287801_G/A   \n",
       "813                        lo  ...       2_238287801_G/A   \n",
       "814                        lo  ...       2_238287801_G/A   \n",
       "815                        lo  ...       2_238287801_G/A   \n",
       "816                        lo  ...       2_238287801_G/A   \n",
       "817                        lo  ...       2_238287801_G/A   \n",
       "818                        lo  ...       2_238287801_G/A   \n",
       "819                        lo  ...       2_238287801_G/A   \n",
       "820                        lo  ...       2_238287801_G/A   \n",
       "821                        lo  ...       2_238287801_G/A   \n",
       "822                        lo  ...       2_238287801_G/A   \n",
       "823                        lo  ...       2_238287801_G/A   \n",
       "824                        lo  ...       2_238287801_G/A   \n",
       "825                        lo  ...       2_238287801_G/A   \n",
       "826                        lo  ...       2_238287801_G/A   \n",
       "827                        lo  ...       2_238287801_G/A   \n",
       "828                        lo  ...       2_238287801_G/A   \n",
       "829                        lo  ...       2_238287801_G/A   \n",
       "830                        lo  ...       2_238287801_G/A   \n",
       "831                        lo  ...       2_238287801_G/A   \n",
       "832                        lo  ...       2_238287801_G/A   \n",
       "833                        lo  ...       2_238287801_G/A   \n",
       "834                        lo  ...       2_238287801_G/A   \n",
       "835                        lo  ...       2_238287801_G/A   \n",
       "836                        lo  ...       2_238287801_G/A   \n",
       "837                        lo  ...       2_238287801_G/A   \n",
       "5626                       lo  ...       22_36697620_G/A   \n",
       "5627                       lo  ...       22_36697620_G/A   \n",
       "5628                       lo  ...       22_36697620_G/A   \n",
       "5629                       lo  ...       22_36697620_G/A   \n",
       "5630                       lo  ...       22_36697620_G/A   \n",
       "5631                       lo  ...       22_36697620_G/A   \n",
       "5632                       lo  ...       22_36697620_G/A   \n",
       "5633                       lo  ...       22_36697620_G/A   \n",
       "\n",
       "                   Location Existing_variation  SYMBOL             Gene  \\\n",
       "749   2:238249174-238249174        rs201460305  COL6A3  ENSG00000163359   \n",
       "750   2:238249174-238249174        rs201460305  COL6A3  ENSG00000163359   \n",
       "751   2:238249174-238249174        rs201460305  COL6A3  ENSG00000163359   \n",
       "752   2:238249174-238249174        rs201460305  COL6A3  ENSG00000163359   \n",
       "753   2:238249174-238249174        rs201460305  COL6A3  ENSG00000163359   \n",
       "754   2:238249174-238249174        rs201460305  COL6A3  ENSG00000163359   \n",
       "755   2:238249174-238249174        rs201460305  COL6A3  ENSG00000163359   \n",
       "756   2:238249174-238249174        rs201460305  COL6A3  ENSG00000163359   \n",
       "811   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "812   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "813   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "814   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "815   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "816   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "817   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "818   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "819   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "820   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "821   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "822   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "823   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "824   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "825   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "826   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "827   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "828   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "829   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "830   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "831   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "832   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "833   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "834   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "835   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "836   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "837   2:238287801-238287801        rs146291186  COL6A3  ENSG00000163359   \n",
       "5626   22:36697620-36697620        rs142467329    MYH9  ENSG00000100345   \n",
       "5627   22:36697620-36697620        rs142467329    MYH9  ENSG00000100345   \n",
       "5628   22:36697620-36697620        rs142467329    MYH9  ENSG00000100345   \n",
       "5629   22:36697620-36697620        rs142467329    MYH9  ENSG00000100345   \n",
       "5630   22:36697620-36697620        rs142467329    MYH9  ENSG00000100345   \n",
       "5631   22:36697620-36697620        rs142467329    MYH9  ENSG00000100345   \n",
       "5632   22:36697620-36697620        rs142467329    MYH9  ENSG00000100345   \n",
       "5633   22:36697620-36697620        rs142467329    MYH9  ENSG00000100345   \n",
       "\n",
       "     SYMBOL.1                         Consequence    IMPACT  \\\n",
       "749    COL6A3                  synonymous_variant       LOW   \n",
       "750    COL6A3                  synonymous_variant       LOW   \n",
       "751    COL6A3                  synonymous_variant       LOW   \n",
       "752    COL6A3                  synonymous_variant       LOW   \n",
       "753    COL6A3                  synonymous_variant       LOW   \n",
       "754    COL6A3  non_coding_transcript_exon_variant  MODIFIER   \n",
       "755    COL6A3                  synonymous_variant       LOW   \n",
       "756    COL6A3  non_coding_transcript_exon_variant  MODIFIER   \n",
       "811    COL6A3                    missense_variant  MODERATE   \n",
       "812    COL6A3                      intron_variant  MODIFIER   \n",
       "813    COL6A3                    missense_variant  MODERATE   \n",
       "814    COL6A3                    missense_variant  MODERATE   \n",
       "815    COL6A3                    missense_variant  MODERATE   \n",
       "816    COL6A3                    missense_variant  MODERATE   \n",
       "817    COL6A3                    missense_variant  MODERATE   \n",
       "818    COL6A3                      intron_variant  MODIFIER   \n",
       "819    COL6A3                      intron_variant  MODIFIER   \n",
       "820    COL6A3                    missense_variant  MODERATE   \n",
       "821    COL6A3                      intron_variant  MODIFIER   \n",
       "822    COL6A3                    missense_variant  MODERATE   \n",
       "823    COL6A3                    missense_variant  MODERATE   \n",
       "824    COL6A3                    missense_variant  MODERATE   \n",
       "825    COL6A3                    missense_variant  MODERATE   \n",
       "826    COL6A3                    missense_variant  MODERATE   \n",
       "827    COL6A3                      intron_variant  MODIFIER   \n",
       "828    COL6A3                      intron_variant  MODIFIER   \n",
       "829    COL6A3                    missense_variant  MODERATE   \n",
       "830    COL6A3                      intron_variant  MODIFIER   \n",
       "831    COL6A3                    missense_variant  MODERATE   \n",
       "832    COL6A3                    missense_variant  MODERATE   \n",
       "833    COL6A3                    missense_variant  MODERATE   \n",
       "834    COL6A3                    missense_variant  MODERATE   \n",
       "835    COL6A3                    missense_variant  MODERATE   \n",
       "836    COL6A3                      intron_variant  MODIFIER   \n",
       "837    COL6A3                      intron_variant  MODIFIER   \n",
       "5626     MYH9                    missense_variant  MODERATE   \n",
       "5627     MYH9               upstream_gene_variant  MODIFIER   \n",
       "5628     MYH9             downstream_gene_variant  MODIFIER   \n",
       "5629     MYH9  non_coding_transcript_exon_variant  MODIFIER   \n",
       "5630     MYH9                    missense_variant  MODERATE   \n",
       "5631     MYH9               upstream_gene_variant  MODIFIER   \n",
       "5632     MYH9             downstream_gene_variant  MODIFIER   \n",
       "5633     MYH9  non_coding_transcript_exon_variant  MODIFIER   \n",
       "\n",
       "                                          CLIN_SIG clinvar  \n",
       "749                                         benign     NaN  \n",
       "750                                         benign     NaN  \n",
       "751                                         benign     NaN  \n",
       "752                                         benign     NaN  \n",
       "753                                         benign     NaN  \n",
       "754                                         benign     NaN  \n",
       "755                                         benign     NaN  \n",
       "756                                         benign     NaN  \n",
       "811    benign,uncertain_significance,likely_benign     NaN  \n",
       "812    benign,uncertain_significance,likely_benign     NaN  \n",
       "813    benign,uncertain_significance,likely_benign     NaN  \n",
       "814    benign,uncertain_significance,likely_benign     NaN  \n",
       "815    benign,uncertain_significance,likely_benign     NaN  \n",
       "816    benign,uncertain_significance,likely_benign     NaN  \n",
       "817    benign,uncertain_significance,likely_benign     NaN  \n",
       "818    benign,uncertain_significance,likely_benign     NaN  \n",
       "819    benign,uncertain_significance,likely_benign     NaN  \n",
       "820    benign,uncertain_significance,likely_benign     NaN  \n",
       "821    benign,uncertain_significance,likely_benign     NaN  \n",
       "822    benign,uncertain_significance,likely_benign     NaN  \n",
       "823    benign,uncertain_significance,likely_benign     NaN  \n",
       "824    benign,uncertain_significance,likely_benign     NaN  \n",
       "825    benign,uncertain_significance,likely_benign     NaN  \n",
       "826    benign,uncertain_significance,likely_benign     NaN  \n",
       "827    benign,uncertain_significance,likely_benign     NaN  \n",
       "828    benign,uncertain_significance,likely_benign     NaN  \n",
       "829    benign,uncertain_significance,likely_benign     NaN  \n",
       "830    benign,uncertain_significance,likely_benign     NaN  \n",
       "831    benign,uncertain_significance,likely_benign     NaN  \n",
       "832    benign,uncertain_significance,likely_benign     NaN  \n",
       "833    benign,uncertain_significance,likely_benign     NaN  \n",
       "834    benign,uncertain_significance,likely_benign     NaN  \n",
       "835    benign,uncertain_significance,likely_benign     NaN  \n",
       "836    benign,uncertain_significance,likely_benign     NaN  \n",
       "837    benign,uncertain_significance,likely_benign     NaN  \n",
       "5626  conflicting_interpretations_of_pathogenicity     NaN  \n",
       "5627  conflicting_interpretations_of_pathogenicity     NaN  \n",
       "5628  conflicting_interpretations_of_pathogenicity     NaN  \n",
       "5629  conflicting_interpretations_of_pathogenicity     NaN  \n",
       "5630  conflicting_interpretations_of_pathogenicity     NaN  \n",
       "5631  conflicting_interpretations_of_pathogenicity     NaN  \n",
       "5632  conflicting_interpretations_of_pathogenicity     NaN  \n",
       "5633  conflicting_interpretations_of_pathogenicity     NaN  \n",
       "\n",
       "[43 rows x 22 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_final_selected"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "b8e14862-fa0f-44e2-9738-31e548cb6679",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SYMBOL\n",
       "COL6A3    35\n",
       "MYH9       8\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_final_selected[\"SYMBOL\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "6ab44d28-549f-47d8-882e-d20fe0cffac8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>...</th>\n",
       "      <th>AF</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs17130230</td>\n",
       "      <td>1:88446236-88446236</td>\n",
       "      <td>A</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1256</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>34594039</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs17130230</td>\n",
       "      <td>1:88446236-88446236</td>\n",
       "      <td>C</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>34594039</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs12134101</td>\n",
       "      <td>1:121213348-121213348</td>\n",
       "      <td>G</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1943</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>27556229,30116257</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs11249433</td>\n",
       "      <td>1:121280613-121280613</td>\n",
       "      <td>C</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>EMBP1</td>\n",
       "      <td>ENSG00000231752</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000458200.1</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>21514219,23193118,22433456,21738711,23535729,2...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs11249433</td>\n",
       "      <td>1:121280613-121280613</td>\n",
       "      <td>G</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>EMBP1</td>\n",
       "      <td>ENSG00000231752</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000458200.1</td>\n",
       "      <td>unprocessed_pseudogene</td>\n",
       "      <td>...</td>\n",
       "      <td>0.1793</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>21514219,23193118,22433456,21738711,23535729,2...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27281</th>\n",
       "      <td>rs12628403</td>\n",
       "      <td>HSCHR22_1_CTG2:39358034-39358034</td>\n",
       "      <td>G</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>APOBEC3A</td>\n",
       "      <td>ENSG00000262156</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000570508.1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>24025454,26612672,27354352,27643540,34659891,3...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27282</th>\n",
       "      <td>rs12628403</td>\n",
       "      <td>HSCHR22_1_CTG2:39358034-39358034</td>\n",
       "      <td>A</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>APOBEC3B-AS1</td>\n",
       "      <td>ENSG00000262679</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000572413.1</td>\n",
       "      <td>antisense</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>24025454,26612672,27354352,27643540,34659891,3...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27283</th>\n",
       "      <td>rs12628403</td>\n",
       "      <td>HSCHR22_1_CTG2:39358034-39358034</td>\n",
       "      <td>G</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>APOBEC3B-AS1</td>\n",
       "      <td>ENSG00000262679</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000572413.1</td>\n",
       "      <td>antisense</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>24025454,26612672,27354352,27643540,34659891,3...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27284</th>\n",
       "      <td>rs12628403</td>\n",
       "      <td>HSCHR22_1_CTG2:39358034-39358034</td>\n",
       "      <td>A</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>APOBEC3A</td>\n",
       "      <td>ENSG00000262156</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000572661.1</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>24025454,26612672,27354352,27643540,34659891,3...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27285</th>\n",
       "      <td>rs12628403</td>\n",
       "      <td>HSCHR22_1_CTG2:39358034-39358034</td>\n",
       "      <td>G</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>APOBEC3A</td>\n",
       "      <td>ENSG00000262156</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000572661.1</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>24025454,26612672,27354352,27643540,34659891,3...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>27286 rows × 44 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Uploaded_variation                          Location Allele  \\\n",
       "0              rs17130230               1:88446236-88446236      A   \n",
       "1              rs17130230               1:88446236-88446236      C   \n",
       "2              rs12134101             1:121213348-121213348      G   \n",
       "3              rs11249433             1:121280613-121280613      C   \n",
       "4              rs11249433             1:121280613-121280613      G   \n",
       "...                   ...                               ...    ...   \n",
       "27281          rs12628403  HSCHR22_1_CTG2:39358034-39358034      G   \n",
       "27282          rs12628403  HSCHR22_1_CTG2:39358034-39358034      A   \n",
       "27283          rs12628403  HSCHR22_1_CTG2:39358034-39358034      G   \n",
       "27284          rs12628403  HSCHR22_1_CTG2:39358034-39358034      A   \n",
       "27285          rs12628403  HSCHR22_1_CTG2:39358034-39358034      G   \n",
       "\n",
       "                                        Consequence    IMPACT        SYMBOL  \\\n",
       "0                                intergenic_variant  MODIFIER             -   \n",
       "1                                intergenic_variant  MODIFIER             -   \n",
       "2                                intergenic_variant  MODIFIER             -   \n",
       "3      intron_variant,non_coding_transcript_variant  MODIFIER         EMBP1   \n",
       "4      intron_variant,non_coding_transcript_variant  MODIFIER         EMBP1   \n",
       "...                                             ...       ...           ...   \n",
       "27281                                intron_variant  MODIFIER      APOBEC3A   \n",
       "27282  intron_variant,non_coding_transcript_variant  MODIFIER  APOBEC3B-AS1   \n",
       "27283  intron_variant,non_coding_transcript_variant  MODIFIER  APOBEC3B-AS1   \n",
       "27284                       downstream_gene_variant  MODIFIER      APOBEC3A   \n",
       "27285                       downstream_gene_variant  MODIFIER      APOBEC3A   \n",
       "\n",
       "                  Gene Feature_type            Feature  \\\n",
       "0                    -            -                  -   \n",
       "1                    -            -                  -   \n",
       "2                    -            -                  -   \n",
       "3      ENSG00000231752   Transcript  ENST00000458200.1   \n",
       "4      ENSG00000231752   Transcript  ENST00000458200.1   \n",
       "...                ...          ...                ...   \n",
       "27281  ENSG00000262156   Transcript  ENST00000570508.1   \n",
       "27282  ENSG00000262679   Transcript  ENST00000572413.1   \n",
       "27283  ENSG00000262679   Transcript  ENST00000572413.1   \n",
       "27284  ENSG00000262156   Transcript  ENST00000572661.1   \n",
       "27285  ENSG00000262156   Transcript  ENST00000572661.1   \n",
       "\n",
       "                      BIOTYPE  ...      AF CLIN_SIG SOMATIC PHENO  \\\n",
       "0                           -  ...  0.1256        -       -     1   \n",
       "1                           -  ...       -        -       -     1   \n",
       "2                           -  ...  0.1943        -       -     -   \n",
       "3      unprocessed_pseudogene  ...       -        -       -     1   \n",
       "4      unprocessed_pseudogene  ...  0.1793        -       -     1   \n",
       "...                       ...  ...     ...      ...     ...   ...   \n",
       "27281          protein_coding  ...       -        -       -     1   \n",
       "27282               antisense  ...       -        -       -     1   \n",
       "27283               antisense  ...       -        -       -     1   \n",
       "27284         retained_intron  ...       -        -       -     1   \n",
       "27285         retained_intron  ...       -        -       -     1   \n",
       "\n",
       "                                                  PUBMED MOTIF_NAME MOTIF_POS  \\\n",
       "0                                               34594039          -         -   \n",
       "1                                               34594039          -         -   \n",
       "2                                      27556229,30116257          -         -   \n",
       "3      21514219,23193118,22433456,21738711,23535729,2...          -         -   \n",
       "4      21514219,23193118,22433456,21738711,23535729,2...          -         -   \n",
       "...                                                  ...        ...       ...   \n",
       "27281  24025454,26612672,27354352,27643540,34659891,3...          -         -   \n",
       "27282  24025454,26612672,27354352,27643540,34659891,3...          -         -   \n",
       "27283  24025454,26612672,27354352,27643540,34659891,3...          -         -   \n",
       "27284  24025454,26612672,27354352,27643540,34659891,3...          -         -   \n",
       "27285  24025454,26612672,27354352,27643540,34659891,3...          -         -   \n",
       "\n",
       "      HIGH_INF_POS MOTIF_SCORE_CHANGE TRANSCRIPTION_FACTORS  \n",
       "0                -                  -                     -  \n",
       "1                -                  -                     -  \n",
       "2                -                  -                     -  \n",
       "3                -                  -                     -  \n",
       "4                -                  -                     -  \n",
       "...            ...                ...                   ...  \n",
       "27281            -                  -                     -  \n",
       "27282            -                  -                     -  \n",
       "27283            -                  -                     -  \n",
       "27284            -                  -                     -  \n",
       "27285            -                  -                     -  \n",
       "\n",
       "[27286 rows x 44 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df3 = pd.read_csv(\n",
    "   \"YsgwttzfOa1RqZXO.txt\",\n",
    "    sep=\"\\t\"\n",
    ")\n",
    "\n",
    "df3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "3b5bc3ea-f180-4645-9a74-b715b16fc36c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['-', 'EMBP1', 'Y_RNA', 'U3', 'CASC10', 'MIR1915', 'SKIDA1', 'MLLT10', 'RNU6-306P', 'ZNF365', 'ZMIZ1', 'FGFR2', 'RP11-62L18.3', 'RPS15AP5', 'LSP1', 'MIR4298', 'AC051649.12', 'C11orf89', 'TNNT3', 'AC051649.1', 'AP000439.1', 'AP000439.5', 'AP000439.3', 'GNAI2P1', 'PTHLH', 'RP11-964E11.2', 'PAX9', 'RAD51B', 'CTD-2566J3.1', 'RN7SL706P', 'RN7SL108P', 'SCNN1B', 'PALB2', 'DCTN5', 'ERN2', 'TOX3', 'CASC16', 'RP11-297L17.2', 'RP11-297L17.3', 'RP11-297L17.4', 'FTO', 'MPHOSPH6', 'RP11-510J16.5', 'ATAD5', 'CTD-2349P21.11', 'TEFM', 'CHST9', 'AQP4-AS1', 'SLC14A2', 'CASP8', 'ALS2CR12', 'TRAK2', 'STRADB', 'AC007256.5', 'AC079610.1', 'AC007563.5', 'RP11-574O16.1', 'RN7SKP43', 'LINC00160', 'AP000330.8', 'PITPNB', 'TTC28-AS1', 'MIR3199-2', 'TTC28', 'CHEK2', 'CTA-292E10.6', 'ZNRF3', 'ZNRF3-IT1', 'ZNRF3-AS1', 'C22orf31', 'KREMEN1', 'EMID1', 'THOC5', 'NF2', 'SEC14L4', 'RP4-539M6.14', 'TCN2', 'APOBEC3A', 'ADSL', 'SGSM3', 'MKL1', 'RP5-1042K10.12', 'RP4-591N18.2', 'AL031594.1', 'COX6B1P3', 'RPL4P6', 'NEK10', 'MINOS1P3', 'RNU6-342P', 'AC099535.1', 'SLC4A7', 'RN7SL859P', 'RPS20P15', 'AC099535.4', 'RNU1-96P', 'FAM194A', 'RP11-103G8.2', 'RP11-103G8.1', 'SIAH2', 'SIAH2-AS1', 'ADAM29', 'RP13-577H12.3', 'AC105914.1', 'TERT', 'RP11-473L15.3', 'RP11-503D12.1', 'RN7SL383P', 'RP11-53O19.1', 'RP11-53O19.2', 'MRPS30', 'RP11-53O19.3', 'AC008940.1', 'MAP3K1', 'AC008937.2', 'snoU13', 'AC008937.3', 'SETD9', 'MIER3', 'CTD-2310F14.1', 'AC016644.1', 'AC114973.1', 'EBF1', 'RP1-111D6.2', 'RP1-111D6.3', 'TAB2', 'RN7SL234P', 'CCDC170', 'RNU6-813P', 'ESR1', 'RNU1-58P', 'KCNU1', 'AC090453.1', 'CASC8', 'RP11-382A18.2', 'RP11-438P9.2', 'APOBEC3B-AS1']\n"
     ]
    }
   ],
   "source": [
    "print(df3[\"SYMBOL\"].unique().tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "38565906-652d-4871-8e2b-30567e4c7c20",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SYMBOL\n",
       "NEK10         3333\n",
       "RAD51B        2903\n",
       "-             2549\n",
       "FGFR2         2547\n",
       "SLC4A7        2352\n",
       "              ... \n",
       "RN7SKP43         1\n",
       "AC079610.1       1\n",
       "AC007256.5       1\n",
       "AP000439.1       1\n",
       "ZNRF3-AS1        1\n",
       "Name: count, Length: 136, dtype: int64"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df3[\"SYMBOL\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "987574cf-16bd-464c-a414-3c5ce81df310",
   "metadata": {},
   "outputs": [],
   "source": [
    "gwas_hugo_symbol = ['EMBP1', 'Y_RNA', 'U3', 'CASC10', 'MIR1915', 'SKIDA1', 'MLLT10', 'RNU6-306P', 'ZNF365', 'ZMIZ1',\\\n",
    "                    'FGFR2', 'RP11-62L18.3', 'RPS15AP5', 'LSP1', 'MIR4298', 'AC051649.12', 'C11orf89', 'TNNT3', 'AC051649.1',\\\n",
    "                    'AP000439.1', 'AP000439.5', 'AP000439.3', 'GNAI2P1', 'PTHLH', 'RP11-964E11.2', 'PAX9', 'RAD51B', 'CTD-2566J3.1',\\\n",
    "                    'RN7SL706P', 'RN7SL108P', 'SCNN1B', 'PALB2', 'DCTN5', 'ERN2', 'TOX3', 'CASC16', 'RP11-297L17.2', 'RP11-297L17.3',\\\n",
    "                    'RP11-297L17.4', 'FTO', 'MPHOSPH6', 'RP11-510J16.5', 'ATAD5', 'CTD-2349P21.11', 'TEFM', 'CHST9', 'AQP4-AS1',\\\n",
    "                    'SLC14A2', 'CASP8', 'ALS2CR12', 'TRAK2', 'STRADB', 'AC007256.5', 'AC079610.1', 'AC007563.5', 'RP11-574O16.1',\\\n",
    "                    'RN7SKP43', 'LINC00160', 'AP000330.8', 'PITPNB', 'TTC28-AS1', 'MIR3199-2', 'TTC28', 'CHEK2', 'CTA-292E10.6',\\\n",
    "                    'ZNRF3', 'ZNRF3-IT1', 'ZNRF3-AS1', 'C22orf31', 'KREMEN1', 'EMID1', 'THOC5', 'NF2', 'SEC14L4', 'RP4-539M6.14',\\\n",
    "                    'TCN2', 'APOBEC3A', 'ADSL', 'SGSM3', 'MKL1', 'RP5-1042K10.12', 'RP4-591N18.2', 'AL031594.1', 'COX6B1P3',\\\n",
    "                    'RPL4P6', 'NEK10', 'MINOS1P3', 'RNU6-342P', 'AC099535.1', 'SLC4A7', 'RN7SL859P', 'RPS20P15', 'AC099535.4',\\\n",
    "                    'RNU1-96P', 'FAM194A', 'RP11-103G8.2', 'RP11-103G8.1', 'SIAH2', 'SIAH2-AS1', 'ADAM29', 'RP13-577H12.3',\\\n",
    "                    'AC105914.1', 'TERT', 'RP11-473L15.3', 'RP11-503D12.1', 'RN7SL383P', 'RP11-53O19.1', 'RP11-53O19.2', 'MRPS30',\\\n",
    "                    'RP11-53O19.3', 'AC008940.1', 'MAP3K1', 'AC008937.2', 'snoU13', 'AC008937.3', 'SETD9', 'MIER3', 'CTD-2310F14.1',\\\n",
    "                    'AC016644.1', 'AC114973.1', 'EBF1', 'RP1-111D6.2', 'RP1-111D6.3', 'TAB2', 'RN7SL234P', 'CCDC170', 'RNU6-813P',\\\n",
    "                    'ESR1', 'RNU1-58P', 'KCNU1', 'AC090453.1', 'CASC8', 'RP11-382A18.2', 'RP11-438P9.2', 'APOBEC3B-AS1']"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python (base2)",
   "language": "python",
   "name": "base2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.14.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
