{
 "cells": [
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-25T02:29:19.496211Z",
     "start_time": "2026-01-25T02:29:18.211896Z"
    }
   },
   "cell_type": "code",
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "from sklearn.impute import KNNImputer\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split"
   ],
   "id": "a05b393c220d6ffd",
   "outputs": [],
   "execution_count": 2
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-25T02:29:37.629103Z",
     "start_time": "2026-01-25T02:29:19.498408Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##upload the original mRNA and methylation\n",
    "#interated mRNA and methylation (n=1417) without any operation and exclusion of features\n",
    "methy_sub = pd.read_csv('/Users/mac1/Desktop/DATAFEST/methylation_sub.csv') #(1417, 13188)\n",
    "mRNA_sub = pd.read_csv('/Users/mac1/Desktop/DATAFEST/mRNA_sub.csv') #(1417, 20604)\n",
    "#overall mRNA (n=1980) without any operation and exclusion of features\n",
    "mRNA_genetic = pd.read_csv('/Users/mac1/Desktop/DATAFEST/mRNA_genetic.csv') #(1980, 20604)"
   ],
   "id": "92d7f60a08c32b03",
   "outputs": [],
   "execution_count": 3
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-25T02:29:37.743124Z",
     "start_time": "2026-01-25T02:29:37.717836Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##extract ID row index for cohort\n",
    "keep_ids_overall = mRNA_genetic['PATIENT_ID'] #overall\n",
    "keep_ids_integrate = methy_sub['PATIENT_ID'] #integrate\n",
    "##extract feature sets for methylation or mRNA\n",
    "methy_features = [c for c in methy_sub.columns if c not in [\"PATIENT_ID\"]]\n",
    "mRNA_features = [c for c in mRNA_sub.columns if c not in [\"PATIENT_ID\"]]"
   ],
   "id": "e2a781aa876b456",
   "outputs": [],
   "execution_count": 4
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-25T02:29:37.791779Z",
     "start_time": "2026-01-25T02:29:37.744393Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##clinical features （overall cohort)\n",
    "clinical_impute_overall = pd.read_csv(\"/Users/mac1/Desktop/DATAFEST/clinical_imputed_1.csv\")\n",
    "clinical_impute_overall.shape"
   ],
   "id": "844d0bc2bf2b920f",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1980, 29)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 5
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-25T02:29:37.823148Z",
     "start_time": "2026-01-25T02:29:37.794292Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##clinical feature (integrated cohort)\n",
    "clinical_impute_integrate = clinical_impute_overall[clinical_impute_overall['PATIENT_ID'].isin(keep_ids_integrate)].copy()\n",
    "clinical_impute_integrate.shape"
   ],
   "id": "97a95800c68c3c69",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1417, 29)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 6
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-25T02:29:40.134691Z",
     "start_time": "2026-01-25T02:29:40.085394Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##gwas feature (overall cohort)\n",
    "gwas=pd.read_csv('/Users/mac1/Desktop/DATAFEST/genomics_selected_gwas_ensembl.csv',index_col=0)\n",
    "gwas_overall = gwas[gwas['PATIENT_ID'].isin(keep_ids_overall)].copy()"
   ],
   "id": "ecea1239e13f7474",
   "outputs": [],
   "execution_count": 8
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-26T03:34:26.239546Z",
     "start_time": "2026-01-26T03:34:25.996061Z"
    }
   },
   "cell_type": "code",
   "source": "gwas_overall",
   "id": "5b847e3340179e8d",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "     PATIENT_ID  ACVRL1_methy  AHNAK_methy  AKT1_methy  ALK_methy  APC_methy  \\\n",
       "1       MB-0000           NaN          NaN         NaN        NaN        NaN   \n",
       "2       MB-0002           NaN          NaN         NaN        NaN        NaN   \n",
       "3       MB-0005           NaN          NaN         NaN        NaN        NaN   \n",
       "4       MB-0006      0.228235     0.039574    0.012346   0.032258   0.190789   \n",
       "5       MB-0008           NaN          NaN         NaN        NaN        NaN   \n",
       "...         ...           ...          ...         ...        ...        ...   \n",
       "1981    MB-7295           NaN          NaN         NaN        NaN        NaN   \n",
       "1982    MB-7296      0.491018     0.065934         NaN   0.000000   0.030303   \n",
       "1983    MB-7297           NaN          NaN         NaN        NaN        NaN   \n",
       "1984    MB-7298           NaN          NaN         NaN        NaN        NaN   \n",
       "1985    MB-7299           NaN          NaN         NaN        NaN        NaN   \n",
       "\n",
       "      ARID1A_methy  ARID1B_methy  ARID2_methy  ASXL1_methy  ...  EMID1_mrna  \\\n",
       "1              NaN           NaN          NaN          NaN  ...    5.825269   \n",
       "2              NaN           NaN          NaN          NaN  ...    5.698300   \n",
       "3              NaN           NaN          NaN          NaN  ...    5.713216   \n",
       "4         0.000000      0.000000     0.000000          0.0  ...    5.501439   \n",
       "5              NaN           NaN          NaN          NaN  ...    6.018440   \n",
       "...            ...           ...          ...          ...  ...         ...   \n",
       "1981           NaN           NaN          NaN          NaN  ...    5.598353   \n",
       "1982      0.007895      0.007519     0.003571          0.0  ...    6.195056   \n",
       "1983           NaN           NaN          NaN          NaN  ...    5.408007   \n",
       "1984           NaN           NaN          NaN          NaN  ...    5.629645   \n",
       "1985           NaN           NaN          NaN          NaN  ...    5.804815   \n",
       "\n",
       "      TTC28_mrna  ATAD5_mrna  PAX9_mrna  MLLT10_mrna  PTHLH_mrna  SKIDA1_mrna  \\\n",
       "1       5.798647    5.762455   5.469058     7.263043    6.013555     5.608817   \n",
       "2       5.609069    5.807519   6.411290     8.121593    5.764321     6.050104   \n",
       "3       5.485918    5.704256   5.817795     6.877284    9.808890     5.321103   \n",
       "4       5.356186    5.670296   6.377648     7.235799    6.089555     5.639904   \n",
       "5       5.738390    5.705560   5.596711     7.947678    6.121529     5.328361   \n",
       "...          ...         ...        ...          ...         ...          ...   \n",
       "1981    5.185025    5.718490   6.096267     7.250477    6.841198     5.444054   \n",
       "1982    5.512736    5.968755   5.843940     7.143147    5.865243     5.447092   \n",
       "1983    5.631185    6.096751   5.650109     7.113933    5.667898     5.381733   \n",
       "1984    5.543150    5.641894   5.715491     7.449389    6.630646     5.391292   \n",
       "1985    5.733710    5.717691   5.260670     7.913005    7.922645     5.583457   \n",
       "\n",
       "      MRPS30_mrna  APOBEC3A_mrna  SEC14L4_mrna  \n",
       "1        8.535214       5.336678      5.868042  \n",
       "2        9.628202       5.377759      5.747465  \n",
       "3        8.915721       5.338215      5.554798  \n",
       "4        9.058717       5.515105      5.881517  \n",
       "5        9.860345       5.375316      5.278561  \n",
       "...           ...            ...           ...  \n",
       "1981     9.119662       5.289611      5.461226  \n",
       "1982     9.195065       5.441914      5.688695  \n",
       "1983    10.738350       5.311344      5.461644  \n",
       "1984    12.146450       5.371765      5.449533  \n",
       "1985     8.909626       5.235019      5.607719  \n",
       "\n",
       "[1980 rows x 224 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PATIENT_ID</th>\n",
       "      <th>ACVRL1_methy</th>\n",
       "      <th>AHNAK_methy</th>\n",
       "      <th>AKT1_methy</th>\n",
       "      <th>ALK_methy</th>\n",
       "      <th>APC_methy</th>\n",
       "      <th>ARID1A_methy</th>\n",
       "      <th>ARID1B_methy</th>\n",
       "      <th>ARID2_methy</th>\n",
       "      <th>ASXL1_methy</th>\n",
       "      <th>...</th>\n",
       "      <th>EMID1_mrna</th>\n",
       "      <th>TTC28_mrna</th>\n",
       "      <th>ATAD5_mrna</th>\n",
       "      <th>PAX9_mrna</th>\n",
       "      <th>MLLT10_mrna</th>\n",
       "      <th>PTHLH_mrna</th>\n",
       "      <th>SKIDA1_mrna</th>\n",
       "      <th>MRPS30_mrna</th>\n",
       "      <th>APOBEC3A_mrna</th>\n",
       "      <th>SEC14L4_mrna</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MB-0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>5.825269</td>\n",
       "      <td>5.798647</td>\n",
       "      <td>5.762455</td>\n",
       "      <td>5.469058</td>\n",
       "      <td>7.263043</td>\n",
       "      <td>6.013555</td>\n",
       "      <td>5.608817</td>\n",
       "      <td>8.535214</td>\n",
       "      <td>5.336678</td>\n",
       "      <td>5.868042</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MB-0002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>5.698300</td>\n",
       "      <td>5.609069</td>\n",
       "      <td>5.807519</td>\n",
       "      <td>6.411290</td>\n",
       "      <td>8.121593</td>\n",
       "      <td>5.764321</td>\n",
       "      <td>6.050104</td>\n",
       "      <td>9.628202</td>\n",
       "      <td>5.377759</td>\n",
       "      <td>5.747465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MB-0005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>5.713216</td>\n",
       "      <td>5.485918</td>\n",
       "      <td>5.704256</td>\n",
       "      <td>5.817795</td>\n",
       "      <td>6.877284</td>\n",
       "      <td>9.808890</td>\n",
       "      <td>5.321103</td>\n",
       "      <td>8.915721</td>\n",
       "      <td>5.338215</td>\n",
       "      <td>5.554798</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MB-0006</td>\n",
       "      <td>0.228235</td>\n",
       "      <td>0.039574</td>\n",
       "      <td>0.012346</td>\n",
       "      <td>0.032258</td>\n",
       "      <td>0.190789</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>5.501439</td>\n",
       "      <td>5.356186</td>\n",
       "      <td>5.670296</td>\n",
       "      <td>6.377648</td>\n",
       "      <td>7.235799</td>\n",
       "      <td>6.089555</td>\n",
       "      <td>5.639904</td>\n",
       "      <td>9.058717</td>\n",
       "      <td>5.515105</td>\n",
       "      <td>5.881517</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MB-0008</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>6.018440</td>\n",
       "      <td>5.738390</td>\n",
       "      <td>5.705560</td>\n",
       "      <td>5.596711</td>\n",
       "      <td>7.947678</td>\n",
       "      <td>6.121529</td>\n",
       "      <td>5.328361</td>\n",
       "      <td>9.860345</td>\n",
       "      <td>5.375316</td>\n",
       "      <td>5.278561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1981</th>\n",
       "      <td>MB-7295</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>5.598353</td>\n",
       "      <td>5.185025</td>\n",
       "      <td>5.718490</td>\n",
       "      <td>6.096267</td>\n",
       "      <td>7.250477</td>\n",
       "      <td>6.841198</td>\n",
       "      <td>5.444054</td>\n",
       "      <td>9.119662</td>\n",
       "      <td>5.289611</td>\n",
       "      <td>5.461226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1982</th>\n",
       "      <td>MB-7296</td>\n",
       "      <td>0.491018</td>\n",
       "      <td>0.065934</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.030303</td>\n",
       "      <td>0.007895</td>\n",
       "      <td>0.007519</td>\n",
       "      <td>0.003571</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6.195056</td>\n",
       "      <td>5.512736</td>\n",
       "      <td>5.968755</td>\n",
       "      <td>5.843940</td>\n",
       "      <td>7.143147</td>\n",
       "      <td>5.865243</td>\n",
       "      <td>5.447092</td>\n",
       "      <td>9.195065</td>\n",
       "      <td>5.441914</td>\n",
       "      <td>5.688695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1983</th>\n",
       "      <td>MB-7297</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>5.408007</td>\n",
       "      <td>5.631185</td>\n",
       "      <td>6.096751</td>\n",
       "      <td>5.650109</td>\n",
       "      <td>7.113933</td>\n",
       "      <td>5.667898</td>\n",
       "      <td>5.381733</td>\n",
       "      <td>10.738350</td>\n",
       "      <td>5.311344</td>\n",
       "      <td>5.461644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1984</th>\n",
       "      <td>MB-7298</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>5.629645</td>\n",
       "      <td>5.543150</td>\n",
       "      <td>5.641894</td>\n",
       "      <td>5.715491</td>\n",
       "      <td>7.449389</td>\n",
       "      <td>6.630646</td>\n",
       "      <td>5.391292</td>\n",
       "      <td>12.146450</td>\n",
       "      <td>5.371765</td>\n",
       "      <td>5.449533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1985</th>\n",
       "      <td>MB-7299</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>5.804815</td>\n",
       "      <td>5.733710</td>\n",
       "      <td>5.717691</td>\n",
       "      <td>5.260670</td>\n",
       "      <td>7.913005</td>\n",
       "      <td>7.922645</td>\n",
       "      <td>5.583457</td>\n",
       "      <td>8.909626</td>\n",
       "      <td>5.235019</td>\n",
       "      <td>5.607719</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1980 rows × 224 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 33
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-25T02:29:40.980405Z",
     "start_time": "2026-01-25T02:29:40.948277Z"
    }
   },
   "cell_type": "code",
   "source": [
    "#99 methylation in the combined overall gwas feature set, may not be used\n",
    "gwas_overall_methy = gwas_overall[[\"PATIENT_ID\"] + list(gwas_overall.columns.intersection(methy_features))]\n",
    "gwas_overall_methy.shape"
   ],
   "id": "2b6d2dbd96e58c12",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1980, 99)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 9
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-25T02:29:42.065725Z",
     "start_time": "2026-01-25T02:29:42.031505Z"
    }
   },
   "cell_type": "code",
   "source": [
    "#125 mRNA in the combined overall gwas feature set\n",
    "gwas_overall_mRNA = gwas_overall[[\"PATIENT_ID\"] + list(gwas_overall.columns.intersection(mRNA_features))]\n",
    "gwas_overall_mRNA.shape"
   ],
   "id": "15dc1b0fc468680",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1980, 125)"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 10
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-25T02:29:43.577033Z",
     "start_time": "2026-01-25T02:29:43.548144Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##gwas feature (integrated cohort)\n",
    "gwas_integrate = gwas[gwas['PATIENT_ID'].isin(keep_ids_integrate)].copy()\n",
    "gwas_integrate"
   ],
   "id": "b5efe432890356c2",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "     PATIENT_ID  ACVRL1_methy  AHNAK_methy  AKT1_methy  ALK_methy  APC_methy  \\\n",
       "4       MB-0006      0.228235     0.039574    0.012346   0.032258   0.190789   \n",
       "11      MB-0028      0.194093     0.032573    0.042254   0.011905   0.110429   \n",
       "12      MB-0035      0.561047     0.000000    0.005587   0.000000   0.068966   \n",
       "16      MB-0046      0.314721     0.000000    0.000000   0.000000   0.000000   \n",
       "18      MB-0050      0.146341     0.049900         NaN   0.000000   0.007143   \n",
       "...         ...           ...          ...         ...        ...        ...   \n",
       "1976    MB-7289      0.040100     0.062780    0.025000   0.000000   0.021858   \n",
       "1977    MB-7291      0.114754     0.090261    0.000000   0.105882   0.127820   \n",
       "1978    MB-7292      0.360996     0.037288         NaN   0.028571   0.222222   \n",
       "1979    MB-7293      0.269841     0.017578    0.017857   0.252336   0.012579   \n",
       "1982    MB-7296      0.491018     0.065934         NaN   0.000000   0.030303   \n",
       "\n",
       "      ARID1A_methy  ARID1B_methy  ARID2_methy  ASXL1_methy  ...  EMID1_mrna  \\\n",
       "4         0.000000      0.000000     0.000000     0.000000  ...    5.501439   \n",
       "11        0.008048      0.002353     0.009881     0.000000  ...    5.562532   \n",
       "12        0.000000      0.001468     0.003091     0.000000  ...    5.496101   \n",
       "16        0.000000      0.001080     0.001684     0.012195  ...    5.288245   \n",
       "18        0.001410      0.000000     0.005906     0.000000  ...    5.388383   \n",
       "...            ...           ...          ...          ...  ...         ...   \n",
       "1976      0.001196      0.003783     0.000000     0.000000  ...    5.789173   \n",
       "1977      0.001185      0.000000     0.008565     0.000000  ...    5.534877   \n",
       "1978      0.006309      0.007782     0.028571     0.000000  ...    5.346964   \n",
       "1979      0.001594      0.000000     0.000000     0.000000  ...    5.794616   \n",
       "1982      0.007895      0.007519     0.003571     0.000000  ...    6.195056   \n",
       "\n",
       "      TTC28_mrna  ATAD5_mrna  PAX9_mrna  MLLT10_mrna  PTHLH_mrna  SKIDA1_mrna  \\\n",
       "4       5.356186    5.670296   6.377648     7.235799    6.089555     5.639904   \n",
       "11      5.819977    5.964095   5.661834     7.856708    5.466805     5.305149   \n",
       "12      5.344607    5.755204   6.733679     7.641674    5.324897     5.378147   \n",
       "16      5.454914    6.017270   5.939822     7.679091    6.873848     5.269159   \n",
       "18      5.980190    5.563753   5.534932     7.628535    7.864273     5.372053   \n",
       "...          ...         ...        ...          ...         ...          ...   \n",
       "1976    5.121171    5.529888   5.608332     7.383893   10.290670     5.448643   \n",
       "1977    5.431327    5.747914   5.631050     7.369142    6.034446     5.381699   \n",
       "1978    5.484218    5.924776   5.682275     7.246598    6.546055     5.612173   \n",
       "1979    5.494233    5.752061   6.126150     7.231557    6.389416     5.632143   \n",
       "1982    5.512736    5.968755   5.843940     7.143147    5.865243     5.447092   \n",
       "\n",
       "      MRPS30_mrna  APOBEC3A_mrna  SEC14L4_mrna  \n",
       "4        9.058717       5.515105      5.881517  \n",
       "11       8.713788       5.160860      5.739570  \n",
       "12       9.020583       5.407859      5.373921  \n",
       "16      11.045562       5.125100      5.904158  \n",
       "18       8.434989       5.295128      5.693610  \n",
       "...           ...            ...           ...  \n",
       "1976     8.446717       5.346037      5.567413  \n",
       "1977     8.832027       5.445283      5.470318  \n",
       "1978     8.947196       5.369745      5.362175  \n",
       "1979    10.417480       5.343353      5.425920  \n",
       "1982     9.195065       5.441914      5.688695  \n",
       "\n",
       "[1417 rows x 224 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PATIENT_ID</th>\n",
       "      <th>ACVRL1_methy</th>\n",
       "      <th>AHNAK_methy</th>\n",
       "      <th>AKT1_methy</th>\n",
       "      <th>ALK_methy</th>\n",
       "      <th>APC_methy</th>\n",
       "      <th>ARID1A_methy</th>\n",
       "      <th>ARID1B_methy</th>\n",
       "      <th>ARID2_methy</th>\n",
       "      <th>ASXL1_methy</th>\n",
       "      <th>...</th>\n",
       "      <th>EMID1_mrna</th>\n",
       "      <th>TTC28_mrna</th>\n",
       "      <th>ATAD5_mrna</th>\n",
       "      <th>PAX9_mrna</th>\n",
       "      <th>MLLT10_mrna</th>\n",
       "      <th>PTHLH_mrna</th>\n",
       "      <th>SKIDA1_mrna</th>\n",
       "      <th>MRPS30_mrna</th>\n",
       "      <th>APOBEC3A_mrna</th>\n",
       "      <th>SEC14L4_mrna</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MB-0006</td>\n",
       "      <td>0.228235</td>\n",
       "      <td>0.039574</td>\n",
       "      <td>0.012346</td>\n",
       "      <td>0.032258</td>\n",
       "      <td>0.190789</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>5.501439</td>\n",
       "      <td>5.356186</td>\n",
       "      <td>5.670296</td>\n",
       "      <td>6.377648</td>\n",
       "      <td>7.235799</td>\n",
       "      <td>6.089555</td>\n",
       "      <td>5.639904</td>\n",
       "      <td>9.058717</td>\n",
       "      <td>5.515105</td>\n",
       "      <td>5.881517</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>MB-0028</td>\n",
       "      <td>0.194093</td>\n",
       "      <td>0.032573</td>\n",
       "      <td>0.042254</td>\n",
       "      <td>0.011905</td>\n",
       "      <td>0.110429</td>\n",
       "      <td>0.008048</td>\n",
       "      <td>0.002353</td>\n",
       "      <td>0.009881</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>5.562532</td>\n",
       "      <td>5.819977</td>\n",
       "      <td>5.964095</td>\n",
       "      <td>5.661834</td>\n",
       "      <td>7.856708</td>\n",
       "      <td>5.466805</td>\n",
       "      <td>5.305149</td>\n",
       "      <td>8.713788</td>\n",
       "      <td>5.160860</td>\n",
       "      <td>5.739570</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MB-0035</td>\n",
       "      <td>0.561047</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.005587</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.068966</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001468</td>\n",
       "      <td>0.003091</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>5.496101</td>\n",
       "      <td>5.344607</td>\n",
       "      <td>5.755204</td>\n",
       "      <td>6.733679</td>\n",
       "      <td>7.641674</td>\n",
       "      <td>5.324897</td>\n",
       "      <td>5.378147</td>\n",
       "      <td>9.020583</td>\n",
       "      <td>5.407859</td>\n",
       "      <td>5.373921</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>MB-0046</td>\n",
       "      <td>0.314721</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001080</td>\n",
       "      <td>0.001684</td>\n",
       "      <td>0.012195</td>\n",
       "      <td>...</td>\n",
       "      <td>5.288245</td>\n",
       "      <td>5.454914</td>\n",
       "      <td>6.017270</td>\n",
       "      <td>5.939822</td>\n",
       "      <td>7.679091</td>\n",
       "      <td>6.873848</td>\n",
       "      <td>5.269159</td>\n",
       "      <td>11.045562</td>\n",
       "      <td>5.125100</td>\n",
       "      <td>5.904158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>MB-0050</td>\n",
       "      <td>0.146341</td>\n",
       "      <td>0.049900</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.007143</td>\n",
       "      <td>0.001410</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.005906</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>5.388383</td>\n",
       "      <td>5.980190</td>\n",
       "      <td>5.563753</td>\n",
       "      <td>5.534932</td>\n",
       "      <td>7.628535</td>\n",
       "      <td>7.864273</td>\n",
       "      <td>5.372053</td>\n",
       "      <td>8.434989</td>\n",
       "      <td>5.295128</td>\n",
       "      <td>5.693610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1976</th>\n",
       "      <td>MB-7289</td>\n",
       "      <td>0.040100</td>\n",
       "      <td>0.062780</td>\n",
       "      <td>0.025000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.021858</td>\n",
       "      <td>0.001196</td>\n",
       "      <td>0.003783</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>5.789173</td>\n",
       "      <td>5.121171</td>\n",
       "      <td>5.529888</td>\n",
       "      <td>5.608332</td>\n",
       "      <td>7.383893</td>\n",
       "      <td>10.290670</td>\n",
       "      <td>5.448643</td>\n",
       "      <td>8.446717</td>\n",
       "      <td>5.346037</td>\n",
       "      <td>5.567413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1977</th>\n",
       "      <td>MB-7291</td>\n",
       "      <td>0.114754</td>\n",
       "      <td>0.090261</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.105882</td>\n",
       "      <td>0.127820</td>\n",
       "      <td>0.001185</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.008565</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>5.534877</td>\n",
       "      <td>5.431327</td>\n",
       "      <td>5.747914</td>\n",
       "      <td>5.631050</td>\n",
       "      <td>7.369142</td>\n",
       "      <td>6.034446</td>\n",
       "      <td>5.381699</td>\n",
       "      <td>8.832027</td>\n",
       "      <td>5.445283</td>\n",
       "      <td>5.470318</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1978</th>\n",
       "      <td>MB-7292</td>\n",
       "      <td>0.360996</td>\n",
       "      <td>0.037288</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.028571</td>\n",
       "      <td>0.222222</td>\n",
       "      <td>0.006309</td>\n",
       "      <td>0.007782</td>\n",
       "      <td>0.028571</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>5.346964</td>\n",
       "      <td>5.484218</td>\n",
       "      <td>5.924776</td>\n",
       "      <td>5.682275</td>\n",
       "      <td>7.246598</td>\n",
       "      <td>6.546055</td>\n",
       "      <td>5.612173</td>\n",
       "      <td>8.947196</td>\n",
       "      <td>5.369745</td>\n",
       "      <td>5.362175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1979</th>\n",
       "      <td>MB-7293</td>\n",
       "      <td>0.269841</td>\n",
       "      <td>0.017578</td>\n",
       "      <td>0.017857</td>\n",
       "      <td>0.252336</td>\n",
       "      <td>0.012579</td>\n",
       "      <td>0.001594</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>5.794616</td>\n",
       "      <td>5.494233</td>\n",
       "      <td>5.752061</td>\n",
       "      <td>6.126150</td>\n",
       "      <td>7.231557</td>\n",
       "      <td>6.389416</td>\n",
       "      <td>5.632143</td>\n",
       "      <td>10.417480</td>\n",
       "      <td>5.343353</td>\n",
       "      <td>5.425920</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1982</th>\n",
       "      <td>MB-7296</td>\n",
       "      <td>0.491018</td>\n",
       "      <td>0.065934</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.030303</td>\n",
       "      <td>0.007895</td>\n",
       "      <td>0.007519</td>\n",
       "      <td>0.003571</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>6.195056</td>\n",
       "      <td>5.512736</td>\n",
       "      <td>5.968755</td>\n",
       "      <td>5.843940</td>\n",
       "      <td>7.143147</td>\n",
       "      <td>5.865243</td>\n",
       "      <td>5.447092</td>\n",
       "      <td>9.195065</td>\n",
       "      <td>5.441914</td>\n",
       "      <td>5.688695</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1417 rows × 224 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 11
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-25T02:29:46.106067Z",
     "start_time": "2026-01-25T02:29:46.070580Z"
    }
   },
   "cell_type": "code",
   "source": [
    "gwas_integrate_methy = gwas_integrate[[\"PATIENT_ID\"] + list(gwas_integrate.columns.intersection(methy_features))]\n",
    "gwas_integrate_methy"
   ],
   "id": "2c52edad73e5187a",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "     PATIENT_ID  ACVRL1_methy  AHNAK_methy  AKT1_methy  ALK_methy  APC_methy  \\\n",
       "4       MB-0006      0.228235     0.039574    0.012346   0.032258   0.190789   \n",
       "11      MB-0028      0.194093     0.032573    0.042254   0.011905   0.110429   \n",
       "12      MB-0035      0.561047     0.000000    0.005587   0.000000   0.068966   \n",
       "16      MB-0046      0.314721     0.000000    0.000000   0.000000   0.000000   \n",
       "18      MB-0050      0.146341     0.049900         NaN   0.000000   0.007143   \n",
       "...         ...           ...          ...         ...        ...        ...   \n",
       "1976    MB-7289      0.040100     0.062780    0.025000   0.000000   0.021858   \n",
       "1977    MB-7291      0.114754     0.090261    0.000000   0.105882   0.127820   \n",
       "1978    MB-7292      0.360996     0.037288         NaN   0.028571   0.222222   \n",
       "1979    MB-7293      0.269841     0.017578    0.017857   0.252336   0.012579   \n",
       "1982    MB-7296      0.491018     0.065934         NaN   0.000000   0.030303   \n",
       "\n",
       "      ARID1A_methy  ARID1B_methy  ARID2_methy  ASXL1_methy  ...  TCN2_methy  \\\n",
       "4         0.000000      0.000000     0.000000     0.000000  ...    0.068493   \n",
       "11        0.008048      0.002353     0.009881     0.000000  ...    0.000000   \n",
       "12        0.000000      0.001468     0.003091     0.000000  ...    0.022989   \n",
       "16        0.000000      0.001080     0.001684     0.012195  ...    0.023256   \n",
       "18        0.001410      0.000000     0.005906     0.000000  ...    0.000000   \n",
       "...            ...           ...          ...          ...  ...         ...   \n",
       "1976      0.001196      0.003783     0.000000     0.000000  ...    0.000000   \n",
       "1977      0.001185      0.000000     0.008565     0.000000  ...    0.125000   \n",
       "1978      0.006309      0.007782     0.028571     0.000000  ...    0.000000   \n",
       "1979      0.001594      0.000000     0.000000     0.000000  ...    0.024390   \n",
       "1982      0.007895      0.007519     0.003571     0.000000  ...    0.025000   \n",
       "\n",
       "      TEFM_methy  TERT_methy  TNNT3_methy  TOX3_methy  TRAK2_methy  \\\n",
       "4       0.000000    0.258065     0.641791    0.000000     0.000000   \n",
       "11      0.007712    0.320847     0.649123    0.016484     0.003891   \n",
       "12      0.000000    0.676724     0.727273    0.066339     0.002049   \n",
       "16      0.003044    0.150763     0.612245    0.002747     0.000000   \n",
       "18      0.002611    0.184486     0.790698    0.000000     0.001996   \n",
       "...          ...         ...          ...         ...          ...   \n",
       "1976    0.000000    0.028926     0.519481    0.012526     0.003856   \n",
       "1977    0.002119    0.103604     0.529412    0.059353     0.000000   \n",
       "1978    0.017391    0.668919          NaN    0.029412     0.005556   \n",
       "1979    0.006757    0.121911     0.813953    0.000000     0.000000   \n",
       "1982    0.008772    0.529010          NaN    0.022099     0.000000   \n",
       "\n",
       "      TTC28_methy  ZMIZ1_methy  ZNF365_methy  ZNRF3_methy  \n",
       "4        0.002174     0.000000      0.040219     0.000000  \n",
       "11       0.008830     0.008696      0.006780     0.000000  \n",
       "12       0.002849     0.000000      0.121951     0.000000  \n",
       "16       0.000000     0.000000      0.003333     0.000000  \n",
       "18       0.001276     0.000000      0.022556     0.000000  \n",
       "...           ...          ...           ...          ...  \n",
       "1976     0.000000     0.000000      0.047619     0.000000  \n",
       "1977     0.003289     0.000000      0.066327     0.010836  \n",
       "1978     0.015873     0.000000      0.006849     0.000000  \n",
       "1979     0.004193     0.004292      0.068293     0.003390  \n",
       "1982     0.027586     0.000000      0.071856     0.000000  \n",
       "\n",
       "[1417 rows x 99 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PATIENT_ID</th>\n",
       "      <th>ACVRL1_methy</th>\n",
       "      <th>AHNAK_methy</th>\n",
       "      <th>AKT1_methy</th>\n",
       "      <th>ALK_methy</th>\n",
       "      <th>APC_methy</th>\n",
       "      <th>ARID1A_methy</th>\n",
       "      <th>ARID1B_methy</th>\n",
       "      <th>ARID2_methy</th>\n",
       "      <th>ASXL1_methy</th>\n",
       "      <th>...</th>\n",
       "      <th>TCN2_methy</th>\n",
       "      <th>TEFM_methy</th>\n",
       "      <th>TERT_methy</th>\n",
       "      <th>TNNT3_methy</th>\n",
       "      <th>TOX3_methy</th>\n",
       "      <th>TRAK2_methy</th>\n",
       "      <th>TTC28_methy</th>\n",
       "      <th>ZMIZ1_methy</th>\n",
       "      <th>ZNF365_methy</th>\n",
       "      <th>ZNRF3_methy</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MB-0006</td>\n",
       "      <td>0.228235</td>\n",
       "      <td>0.039574</td>\n",
       "      <td>0.012346</td>\n",
       "      <td>0.032258</td>\n",
       "      <td>0.190789</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.068493</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.258065</td>\n",
       "      <td>0.641791</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002174</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.040219</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>MB-0028</td>\n",
       "      <td>0.194093</td>\n",
       "      <td>0.032573</td>\n",
       "      <td>0.042254</td>\n",
       "      <td>0.011905</td>\n",
       "      <td>0.110429</td>\n",
       "      <td>0.008048</td>\n",
       "      <td>0.002353</td>\n",
       "      <td>0.009881</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.007712</td>\n",
       "      <td>0.320847</td>\n",
       "      <td>0.649123</td>\n",
       "      <td>0.016484</td>\n",
       "      <td>0.003891</td>\n",
       "      <td>0.008830</td>\n",
       "      <td>0.008696</td>\n",
       "      <td>0.006780</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MB-0035</td>\n",
       "      <td>0.561047</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.005587</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.068966</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001468</td>\n",
       "      <td>0.003091</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.022989</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.676724</td>\n",
       "      <td>0.727273</td>\n",
       "      <td>0.066339</td>\n",
       "      <td>0.002049</td>\n",
       "      <td>0.002849</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.121951</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>MB-0046</td>\n",
       "      <td>0.314721</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001080</td>\n",
       "      <td>0.001684</td>\n",
       "      <td>0.012195</td>\n",
       "      <td>...</td>\n",
       "      <td>0.023256</td>\n",
       "      <td>0.003044</td>\n",
       "      <td>0.150763</td>\n",
       "      <td>0.612245</td>\n",
       "      <td>0.002747</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003333</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>MB-0050</td>\n",
       "      <td>0.146341</td>\n",
       "      <td>0.049900</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.007143</td>\n",
       "      <td>0.001410</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.005906</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002611</td>\n",
       "      <td>0.184486</td>\n",
       "      <td>0.790698</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001996</td>\n",
       "      <td>0.001276</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.022556</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1976</th>\n",
       "      <td>MB-7289</td>\n",
       "      <td>0.040100</td>\n",
       "      <td>0.062780</td>\n",
       "      <td>0.025000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.021858</td>\n",
       "      <td>0.001196</td>\n",
       "      <td>0.003783</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.028926</td>\n",
       "      <td>0.519481</td>\n",
       "      <td>0.012526</td>\n",
       "      <td>0.003856</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.047619</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1977</th>\n",
       "      <td>MB-7291</td>\n",
       "      <td>0.114754</td>\n",
       "      <td>0.090261</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.105882</td>\n",
       "      <td>0.127820</td>\n",
       "      <td>0.001185</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.008565</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.125000</td>\n",
       "      <td>0.002119</td>\n",
       "      <td>0.103604</td>\n",
       "      <td>0.529412</td>\n",
       "      <td>0.059353</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003289</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.066327</td>\n",
       "      <td>0.010836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1978</th>\n",
       "      <td>MB-7292</td>\n",
       "      <td>0.360996</td>\n",
       "      <td>0.037288</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.028571</td>\n",
       "      <td>0.222222</td>\n",
       "      <td>0.006309</td>\n",
       "      <td>0.007782</td>\n",
       "      <td>0.028571</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.017391</td>\n",
       "      <td>0.668919</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.029412</td>\n",
       "      <td>0.005556</td>\n",
       "      <td>0.015873</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.006849</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1979</th>\n",
       "      <td>MB-7293</td>\n",
       "      <td>0.269841</td>\n",
       "      <td>0.017578</td>\n",
       "      <td>0.017857</td>\n",
       "      <td>0.252336</td>\n",
       "      <td>0.012579</td>\n",
       "      <td>0.001594</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.024390</td>\n",
       "      <td>0.006757</td>\n",
       "      <td>0.121911</td>\n",
       "      <td>0.813953</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.004193</td>\n",
       "      <td>0.004292</td>\n",
       "      <td>0.068293</td>\n",
       "      <td>0.003390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1982</th>\n",
       "      <td>MB-7296</td>\n",
       "      <td>0.491018</td>\n",
       "      <td>0.065934</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.030303</td>\n",
       "      <td>0.007895</td>\n",
       "      <td>0.007519</td>\n",
       "      <td>0.003571</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.025000</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>0.529010</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.022099</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.027586</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.071856</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1417 rows × 99 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 12
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-25T02:29:48.091652Z",
     "start_time": "2026-01-25T02:29:48.054924Z"
    }
   },
   "cell_type": "code",
   "source": [
    "gwas_integrate_mRNA = gwas_integrate[[\"PATIENT_ID\"] + list(gwas_integrate.columns.intersection(mRNA_features))]\n",
    "gwas_integrate_mRNA"
   ],
   "id": "8e2045c247503d4a",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "     PATIENT_ID  AGMO_mrna  AHNAK_mrna  ALK_mrna  LAMB3_mrna  SMAD4_mrna  \\\n",
       "4       MB-0006   5.462942   11.511308  5.489501    5.898610    8.757572   \n",
       "11      MB-0028   5.343175   11.064371  5.597277    5.433681    8.496236   \n",
       "12      MB-0035   5.340526   11.787179  5.832625    6.112071    7.907685   \n",
       "16      MB-0046   5.358802   11.311807  5.567264    7.060030    8.168369   \n",
       "18      MB-0050   5.283746   12.004970  5.696334    7.409536    8.799622   \n",
       "...         ...        ...         ...       ...         ...         ...   \n",
       "1976    MB-7289   5.429778   11.567410  5.479520    8.470224    8.662044   \n",
       "1977    MB-7291   5.333743   10.877760  5.498422    6.400490    8.295540   \n",
       "1978    MB-7292   5.528522   10.810070  5.760783    5.687461    8.638073   \n",
       "1979    MB-7293   5.341804   11.904170  5.739238    6.408341    8.785697   \n",
       "1982    MB-7296   5.376283   10.339950  5.516525    6.207706    8.533860   \n",
       "\n",
       "      CDKN1B_mrna  EGFR_mrna  NF2_mrna  SETD2_mrna  ...  EMID1_mrna  \\\n",
       "4       10.380481   5.585357  5.408832    7.605424  ...    5.501439   \n",
       "11       8.307517   5.502281  5.366654    9.309069  ...    5.562532   \n",
       "12       7.890494   5.466188  5.357428    8.972442  ...    5.496101   \n",
       "16       7.136789   5.825170  5.479556    9.169579  ...    5.288245   \n",
       "18       7.416582   7.094461  5.544637    8.758379  ...    5.388383   \n",
       "...           ...        ...       ...         ...  ...         ...   \n",
       "1976     7.592850   6.666688  5.586575    8.824639  ...    5.789173   \n",
       "1977     7.528250   6.437883  5.668293    8.348152  ...    5.534877   \n",
       "1978     7.473152   5.917815  5.604277    8.421878  ...    5.346964   \n",
       "1979     8.211591   5.825643  5.737924    8.223599  ...    5.794616   \n",
       "1982     8.889112   5.411055  5.613963    8.037245  ...    6.195056   \n",
       "\n",
       "      TTC28_mrna  ATAD5_mrna  PAX9_mrna  MLLT10_mrna  PTHLH_mrna  SKIDA1_mrna  \\\n",
       "4       5.356186    5.670296   6.377648     7.235799    6.089555     5.639904   \n",
       "11      5.819977    5.964095   5.661834     7.856708    5.466805     5.305149   \n",
       "12      5.344607    5.755204   6.733679     7.641674    5.324897     5.378147   \n",
       "16      5.454914    6.017270   5.939822     7.679091    6.873848     5.269159   \n",
       "18      5.980190    5.563753   5.534932     7.628535    7.864273     5.372053   \n",
       "...          ...         ...        ...          ...         ...          ...   \n",
       "1976    5.121171    5.529888   5.608332     7.383893   10.290670     5.448643   \n",
       "1977    5.431327    5.747914   5.631050     7.369142    6.034446     5.381699   \n",
       "1978    5.484218    5.924776   5.682275     7.246598    6.546055     5.612173   \n",
       "1979    5.494233    5.752061   6.126150     7.231557    6.389416     5.632143   \n",
       "1982    5.512736    5.968755   5.843940     7.143147    5.865243     5.447092   \n",
       "\n",
       "      MRPS30_mrna  APOBEC3A_mrna  SEC14L4_mrna  \n",
       "4        9.058717       5.515105      5.881517  \n",
       "11       8.713788       5.160860      5.739570  \n",
       "12       9.020583       5.407859      5.373921  \n",
       "16      11.045562       5.125100      5.904158  \n",
       "18       8.434989       5.295128      5.693610  \n",
       "...           ...            ...           ...  \n",
       "1976     8.446717       5.346037      5.567413  \n",
       "1977     8.832027       5.445283      5.470318  \n",
       "1978     8.947196       5.369745      5.362175  \n",
       "1979    10.417480       5.343353      5.425920  \n",
       "1982     9.195065       5.441914      5.688695  \n",
       "\n",
       "[1417 rows x 125 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PATIENT_ID</th>\n",
       "      <th>AGMO_mrna</th>\n",
       "      <th>AHNAK_mrna</th>\n",
       "      <th>ALK_mrna</th>\n",
       "      <th>LAMB3_mrna</th>\n",
       "      <th>SMAD4_mrna</th>\n",
       "      <th>CDKN1B_mrna</th>\n",
       "      <th>EGFR_mrna</th>\n",
       "      <th>NF2_mrna</th>\n",
       "      <th>SETD2_mrna</th>\n",
       "      <th>...</th>\n",
       "      <th>EMID1_mrna</th>\n",
       "      <th>TTC28_mrna</th>\n",
       "      <th>ATAD5_mrna</th>\n",
       "      <th>PAX9_mrna</th>\n",
       "      <th>MLLT10_mrna</th>\n",
       "      <th>PTHLH_mrna</th>\n",
       "      <th>SKIDA1_mrna</th>\n",
       "      <th>MRPS30_mrna</th>\n",
       "      <th>APOBEC3A_mrna</th>\n",
       "      <th>SEC14L4_mrna</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MB-0006</td>\n",
       "      <td>5.462942</td>\n",
       "      <td>11.511308</td>\n",
       "      <td>5.489501</td>\n",
       "      <td>5.898610</td>\n",
       "      <td>8.757572</td>\n",
       "      <td>10.380481</td>\n",
       "      <td>5.585357</td>\n",
       "      <td>5.408832</td>\n",
       "      <td>7.605424</td>\n",
       "      <td>...</td>\n",
       "      <td>5.501439</td>\n",
       "      <td>5.356186</td>\n",
       "      <td>5.670296</td>\n",
       "      <td>6.377648</td>\n",
       "      <td>7.235799</td>\n",
       "      <td>6.089555</td>\n",
       "      <td>5.639904</td>\n",
       "      <td>9.058717</td>\n",
       "      <td>5.515105</td>\n",
       "      <td>5.881517</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>MB-0028</td>\n",
       "      <td>5.343175</td>\n",
       "      <td>11.064371</td>\n",
       "      <td>5.597277</td>\n",
       "      <td>5.433681</td>\n",
       "      <td>8.496236</td>\n",
       "      <td>8.307517</td>\n",
       "      <td>5.502281</td>\n",
       "      <td>5.366654</td>\n",
       "      <td>9.309069</td>\n",
       "      <td>...</td>\n",
       "      <td>5.562532</td>\n",
       "      <td>5.819977</td>\n",
       "      <td>5.964095</td>\n",
       "      <td>5.661834</td>\n",
       "      <td>7.856708</td>\n",
       "      <td>5.466805</td>\n",
       "      <td>5.305149</td>\n",
       "      <td>8.713788</td>\n",
       "      <td>5.160860</td>\n",
       "      <td>5.739570</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MB-0035</td>\n",
       "      <td>5.340526</td>\n",
       "      <td>11.787179</td>\n",
       "      <td>5.832625</td>\n",
       "      <td>6.112071</td>\n",
       "      <td>7.907685</td>\n",
       "      <td>7.890494</td>\n",
       "      <td>5.466188</td>\n",
       "      <td>5.357428</td>\n",
       "      <td>8.972442</td>\n",
       "      <td>...</td>\n",
       "      <td>5.496101</td>\n",
       "      <td>5.344607</td>\n",
       "      <td>5.755204</td>\n",
       "      <td>6.733679</td>\n",
       "      <td>7.641674</td>\n",
       "      <td>5.324897</td>\n",
       "      <td>5.378147</td>\n",
       "      <td>9.020583</td>\n",
       "      <td>5.407859</td>\n",
       "      <td>5.373921</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>MB-0046</td>\n",
       "      <td>5.358802</td>\n",
       "      <td>11.311807</td>\n",
       "      <td>5.567264</td>\n",
       "      <td>7.060030</td>\n",
       "      <td>8.168369</td>\n",
       "      <td>7.136789</td>\n",
       "      <td>5.825170</td>\n",
       "      <td>5.479556</td>\n",
       "      <td>9.169579</td>\n",
       "      <td>...</td>\n",
       "      <td>5.288245</td>\n",
       "      <td>5.454914</td>\n",
       "      <td>6.017270</td>\n",
       "      <td>5.939822</td>\n",
       "      <td>7.679091</td>\n",
       "      <td>6.873848</td>\n",
       "      <td>5.269159</td>\n",
       "      <td>11.045562</td>\n",
       "      <td>5.125100</td>\n",
       "      <td>5.904158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>MB-0050</td>\n",
       "      <td>5.283746</td>\n",
       "      <td>12.004970</td>\n",
       "      <td>5.696334</td>\n",
       "      <td>7.409536</td>\n",
       "      <td>8.799622</td>\n",
       "      <td>7.416582</td>\n",
       "      <td>7.094461</td>\n",
       "      <td>5.544637</td>\n",
       "      <td>8.758379</td>\n",
       "      <td>...</td>\n",
       "      <td>5.388383</td>\n",
       "      <td>5.980190</td>\n",
       "      <td>5.563753</td>\n",
       "      <td>5.534932</td>\n",
       "      <td>7.628535</td>\n",
       "      <td>7.864273</td>\n",
       "      <td>5.372053</td>\n",
       "      <td>8.434989</td>\n",
       "      <td>5.295128</td>\n",
       "      <td>5.693610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1976</th>\n",
       "      <td>MB-7289</td>\n",
       "      <td>5.429778</td>\n",
       "      <td>11.567410</td>\n",
       "      <td>5.479520</td>\n",
       "      <td>8.470224</td>\n",
       "      <td>8.662044</td>\n",
       "      <td>7.592850</td>\n",
       "      <td>6.666688</td>\n",
       "      <td>5.586575</td>\n",
       "      <td>8.824639</td>\n",
       "      <td>...</td>\n",
       "      <td>5.789173</td>\n",
       "      <td>5.121171</td>\n",
       "      <td>5.529888</td>\n",
       "      <td>5.608332</td>\n",
       "      <td>7.383893</td>\n",
       "      <td>10.290670</td>\n",
       "      <td>5.448643</td>\n",
       "      <td>8.446717</td>\n",
       "      <td>5.346037</td>\n",
       "      <td>5.567413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1977</th>\n",
       "      <td>MB-7291</td>\n",
       "      <td>5.333743</td>\n",
       "      <td>10.877760</td>\n",
       "      <td>5.498422</td>\n",
       "      <td>6.400490</td>\n",
       "      <td>8.295540</td>\n",
       "      <td>7.528250</td>\n",
       "      <td>6.437883</td>\n",
       "      <td>5.668293</td>\n",
       "      <td>8.348152</td>\n",
       "      <td>...</td>\n",
       "      <td>5.534877</td>\n",
       "      <td>5.431327</td>\n",
       "      <td>5.747914</td>\n",
       "      <td>5.631050</td>\n",
       "      <td>7.369142</td>\n",
       "      <td>6.034446</td>\n",
       "      <td>5.381699</td>\n",
       "      <td>8.832027</td>\n",
       "      <td>5.445283</td>\n",
       "      <td>5.470318</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1978</th>\n",
       "      <td>MB-7292</td>\n",
       "      <td>5.528522</td>\n",
       "      <td>10.810070</td>\n",
       "      <td>5.760783</td>\n",
       "      <td>5.687461</td>\n",
       "      <td>8.638073</td>\n",
       "      <td>7.473152</td>\n",
       "      <td>5.917815</td>\n",
       "      <td>5.604277</td>\n",
       "      <td>8.421878</td>\n",
       "      <td>...</td>\n",
       "      <td>5.346964</td>\n",
       "      <td>5.484218</td>\n",
       "      <td>5.924776</td>\n",
       "      <td>5.682275</td>\n",
       "      <td>7.246598</td>\n",
       "      <td>6.546055</td>\n",
       "      <td>5.612173</td>\n",
       "      <td>8.947196</td>\n",
       "      <td>5.369745</td>\n",
       "      <td>5.362175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1979</th>\n",
       "      <td>MB-7293</td>\n",
       "      <td>5.341804</td>\n",
       "      <td>11.904170</td>\n",
       "      <td>5.739238</td>\n",
       "      <td>6.408341</td>\n",
       "      <td>8.785697</td>\n",
       "      <td>8.211591</td>\n",
       "      <td>5.825643</td>\n",
       "      <td>5.737924</td>\n",
       "      <td>8.223599</td>\n",
       "      <td>...</td>\n",
       "      <td>5.794616</td>\n",
       "      <td>5.494233</td>\n",
       "      <td>5.752061</td>\n",
       "      <td>6.126150</td>\n",
       "      <td>7.231557</td>\n",
       "      <td>6.389416</td>\n",
       "      <td>5.632143</td>\n",
       "      <td>10.417480</td>\n",
       "      <td>5.343353</td>\n",
       "      <td>5.425920</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1982</th>\n",
       "      <td>MB-7296</td>\n",
       "      <td>5.376283</td>\n",
       "      <td>10.339950</td>\n",
       "      <td>5.516525</td>\n",
       "      <td>6.207706</td>\n",
       "      <td>8.533860</td>\n",
       "      <td>8.889112</td>\n",
       "      <td>5.411055</td>\n",
       "      <td>5.613963</td>\n",
       "      <td>8.037245</td>\n",
       "      <td>...</td>\n",
       "      <td>6.195056</td>\n",
       "      <td>5.512736</td>\n",
       "      <td>5.968755</td>\n",
       "      <td>5.843940</td>\n",
       "      <td>7.143147</td>\n",
       "      <td>5.865243</td>\n",
       "      <td>5.447092</td>\n",
       "      <td>9.195065</td>\n",
       "      <td>5.441914</td>\n",
       "      <td>5.688695</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1417 rows × 125 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 13
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-25T02:29:50.115272Z",
     "start_time": "2026-01-25T02:29:50.081265Z"
    }
   },
   "cell_type": "code",
   "source": [
    "nan_counts_mRNA = gwas_overall_mRNA.isna().sum()\n",
    "nan_counts_meth = gwas_integrate_methy.isna().sum()\n",
    "missing_pct_mRNA= gwas_overall_mRNA.isna().mean() * 100\n",
    "missing_pct_meth= gwas_integrate_methy.isna().mean() * 100\n",
    "print(\"Columns with NaN in mRNA in GWAS overall:\", (nan_counts_mRNA > 0).sum())\n",
    "print(\"Columns with NaN in methylation in GWAS integrated:\", (nan_counts_meth > 0).sum())\n",
    "print(\"Percentage of missing in mRNA in GWAS overall\",missing_pct_mRNA.describe())\n",
    "print(\"Percentage of missing in methylation in GWAS integrated\",missing_pct_meth.describe())\n",
    "##the missingness percentage is acceptable, let's move on for KNN imputation"
   ],
   "id": "65aef04dcf93a1c",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Columns with NaN in mRNA in GWAS overall: 0\n",
      "Columns with NaN in methylation in GWAS integrated: 75\n",
      "Percentage of missing in mRNA in GWAS overall count    125.0\n",
      "mean       0.0\n",
      "std        0.0\n",
      "min        0.0\n",
      "25%        0.0\n",
      "50%        0.0\n",
      "75%        0.0\n",
      "max        0.0\n",
      "dtype: float64\n",
      "Percentage of missing in methylation in GWAS integrated count    99.000000\n",
      "mean      3.758830\n",
      "std       6.782153\n",
      "min       0.000000\n",
      "25%       0.070572\n",
      "50%       0.423430\n",
      "75%       3.705011\n",
      "max      28.863797\n",
      "dtype: float64\n"
     ]
    }
   ],
   "execution_count": 14
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-25T02:29:52.821997Z",
     "start_time": "2026-01-25T02:29:52.796555Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##data-driven feature set\n",
    "#integrated cohort case 1 (n=1417, N=200)\n",
    "var_data_driven_integrate_1 = pd.read_csv('/Users/mac1/Desktop/DATAFEST/strategy_3_Importance_Cohort1_Case1_SeparateCox.csv')\n",
    "top200_features_1 = var_data_driven_integrate_1[\"feature_name\"].head(200).tolist()\n",
    "##integrated cohort case 2 (n=1417, N=300)\n",
    "var_data_driven_integrate_2 = pd.read_csv('/Users/mac1/Desktop/DATAFEST/strategy_3_Importance_Cohort1_Case2_CombinedCox.csv')\n",
    "top300_features_2 = var_data_driven_integrate_2[\"feature_name\"].head(300).tolist()\n",
    "#overall cohort (n=1980, N mRNA=100)\n",
    "var_data_driven_overall = pd.read_csv('/Users/mac1/Desktop/DATAFEST/strategy_3_Importance_C2_Case1_mRNA_only.csv')\n",
    "top100_features_3 = var_data_driven_overall[\"feature_name\"].head(100).tolist()"
   ],
   "id": "b790d68d926e2681",
   "outputs": [],
   "execution_count": 15
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-25T02:29:54.632387Z",
     "start_time": "2026-01-25T02:29:54.285525Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##data-driven dataset (integrate cohort case 1)\n",
    "genetic_sub = methy_sub.merge(mRNA_sub, on=\"PATIENT_ID\", how=\"inner\")\n",
    "data_driven_integrate_1 = genetic_sub[[\"PATIENT_ID\"] + list(genetic_sub.columns.intersection(top200_features_1))]\n",
    "##data-driven dataset (integrate cohort case 2)\n",
    "data_driven_integrate_2 = genetic_sub[[\"PATIENT_ID\"] + list(genetic_sub.columns.intersection(top300_features_2))]\n",
    "##data-driven dataset (overall cohort)\n",
    "data_driven_overall = mRNA_genetic[[\"PATIENT_ID\"] + list(mRNA_genetic.columns.intersection(top100_features_3))]"
   ],
   "id": "83bc3c93dcef9e61",
   "outputs": [],
   "execution_count": 16
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-25T05:47:42.146653Z",
     "start_time": "2026-01-25T05:47:42.115568Z"
    }
   },
   "cell_type": "code",
   "source": [
    "cols = [\"PATIENT_ID\"] + [c for c in data_driven_integrate_1.columns if c.endswith(\"_mrna\")]\n",
    "data_driven_integrate_1_mrna = data_driven_integrate_1[cols]\n",
    "data_driven_integrate_1_mrna"
   ],
   "id": "8eb0e91b15e47f0d",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "     PATIENT_ID  LINC00937_mrna  TAT_mrna  C14orf184_mrna  GLA_mrna  \\\n",
       "0       MB-0006        5.652845  5.746114        5.757450  8.485949   \n",
       "1       MB-0028        5.408407  5.433514        5.689489  8.731710   \n",
       "2       MB-0035        5.525525  5.563156        6.037567  8.149366   \n",
       "3       MB-0046        5.520658  5.556175        6.511966  9.690913   \n",
       "4       MB-0050        5.553870  7.874150        6.072143  9.028392   \n",
       "...         ...             ...       ...             ...       ...   \n",
       "1412    MB-7289        5.311013  6.588677        5.627716  9.365496   \n",
       "1413    MB-7291        5.639454  5.806753        5.801780  9.135438   \n",
       "1414    MB-7292        5.320487  5.475669        5.907214  8.669601   \n",
       "1415    MB-7293        5.472402  5.459333        5.476251  9.409805   \n",
       "1416    MB-7296        5.461060  5.399961        6.095062  8.967495   \n",
       "\n",
       "      SLC16A3_mrna  IGFN1_mrna  DUSP15_mrna  DNASE1L3_mrna  INPP5A_mrna  ...  \\\n",
       "0         9.983743    5.706636     5.719950       5.644084     7.895471  ...   \n",
       "1         8.608753    5.613722     5.440101       5.163537     7.682070  ...   \n",
       "2         6.730225    5.874200     5.353390       5.365097     8.872877  ...   \n",
       "3         8.949934    5.417182     5.469190       5.181842     7.397672  ...   \n",
       "4         7.074297    5.399167     5.223366       6.589710     7.625433  ...   \n",
       "...            ...         ...          ...            ...          ...  ...   \n",
       "1412      7.828994    5.240209     5.473240       6.396629     7.374513  ...   \n",
       "1413      9.178206    5.490531     5.689773       7.174227     7.855154  ...   \n",
       "1414     10.047511    5.196646     5.485982       5.566037     7.430706  ...   \n",
       "1415      8.348674    5.071341     5.424013       6.719742     7.437116  ...   \n",
       "1416     10.303957    5.155927     5.914946       5.672096     7.151103  ...   \n",
       "\n",
       "      ENC1_mrna  SH3BGRL2_mrna  SLC38A1_mrna  HILPDA_mrna  ALDOC_mrna  \\\n",
       "0      7.690706       7.451934      9.537609     7.726851    7.130907   \n",
       "1      7.401007       6.445898     10.360132     7.323750    7.429350   \n",
       "2      8.611371       6.614811     11.006116     7.551945    6.484451   \n",
       "3      8.526717       7.703336      7.738962     7.294387    8.326490   \n",
       "4      8.284187       8.118200     10.092764     6.739002    8.355693   \n",
       "...         ...            ...           ...          ...         ...   \n",
       "1412   8.609249       7.780088     10.208355     6.374919    8.955945   \n",
       "1413   8.796033       7.243271      9.858089     6.272637    7.444408   \n",
       "1414   8.587751       6.867430     10.248435     7.438962    7.759453   \n",
       "1415   8.656620       7.449284      9.302515     6.397087    7.558685   \n",
       "1416   7.989299       5.902987     10.212643     6.364783   11.395064   \n",
       "\n",
       "      GRB2_mrna  MMP3_mrna  VANGL1_mrna  HMGB3_mrna  EXPH5_mrna  \n",
       "0      6.655574   6.375535     5.368648    7.702250    5.852908  \n",
       "1     10.484214   5.335386     5.495902    8.134939    6.638296  \n",
       "2      6.771495   5.717184     5.609675    7.148153    7.305261  \n",
       "3      9.456882   6.112901     6.165212    8.016967    5.850510  \n",
       "4      7.460314   7.397655     5.381572    7.639669    7.079205  \n",
       "...         ...        ...          ...         ...         ...  \n",
       "1412   7.438374   8.345558     5.551011    6.389354    6.075920  \n",
       "1413   8.620689   7.491018     6.134674    8.126522    6.458758  \n",
       "1414   7.811448   6.195857     5.710864    8.460674    6.164186  \n",
       "1415   7.019974   8.311320     5.593818    6.092144    6.310746  \n",
       "1416   8.287243   7.185083     5.512210    8.437577    6.154567  \n",
       "\n",
       "[1417 rows x 135 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PATIENT_ID</th>\n",
       "      <th>LINC00937_mrna</th>\n",
       "      <th>TAT_mrna</th>\n",
       "      <th>C14orf184_mrna</th>\n",
       "      <th>GLA_mrna</th>\n",
       "      <th>SLC16A3_mrna</th>\n",
       "      <th>IGFN1_mrna</th>\n",
       "      <th>DUSP15_mrna</th>\n",
       "      <th>DNASE1L3_mrna</th>\n",
       "      <th>INPP5A_mrna</th>\n",
       "      <th>...</th>\n",
       "      <th>ENC1_mrna</th>\n",
       "      <th>SH3BGRL2_mrna</th>\n",
       "      <th>SLC38A1_mrna</th>\n",
       "      <th>HILPDA_mrna</th>\n",
       "      <th>ALDOC_mrna</th>\n",
       "      <th>GRB2_mrna</th>\n",
       "      <th>MMP3_mrna</th>\n",
       "      <th>VANGL1_mrna</th>\n",
       "      <th>HMGB3_mrna</th>\n",
       "      <th>EXPH5_mrna</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MB-0006</td>\n",
       "      <td>5.652845</td>\n",
       "      <td>5.746114</td>\n",
       "      <td>5.757450</td>\n",
       "      <td>8.485949</td>\n",
       "      <td>9.983743</td>\n",
       "      <td>5.706636</td>\n",
       "      <td>5.719950</td>\n",
       "      <td>5.644084</td>\n",
       "      <td>7.895471</td>\n",
       "      <td>...</td>\n",
       "      <td>7.690706</td>\n",
       "      <td>7.451934</td>\n",
       "      <td>9.537609</td>\n",
       "      <td>7.726851</td>\n",
       "      <td>7.130907</td>\n",
       "      <td>6.655574</td>\n",
       "      <td>6.375535</td>\n",
       "      <td>5.368648</td>\n",
       "      <td>7.702250</td>\n",
       "      <td>5.852908</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MB-0028</td>\n",
       "      <td>5.408407</td>\n",
       "      <td>5.433514</td>\n",
       "      <td>5.689489</td>\n",
       "      <td>8.731710</td>\n",
       "      <td>8.608753</td>\n",
       "      <td>5.613722</td>\n",
       "      <td>5.440101</td>\n",
       "      <td>5.163537</td>\n",
       "      <td>7.682070</td>\n",
       "      <td>...</td>\n",
       "      <td>7.401007</td>\n",
       "      <td>6.445898</td>\n",
       "      <td>10.360132</td>\n",
       "      <td>7.323750</td>\n",
       "      <td>7.429350</td>\n",
       "      <td>10.484214</td>\n",
       "      <td>5.335386</td>\n",
       "      <td>5.495902</td>\n",
       "      <td>8.134939</td>\n",
       "      <td>6.638296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MB-0035</td>\n",
       "      <td>5.525525</td>\n",
       "      <td>5.563156</td>\n",
       "      <td>6.037567</td>\n",
       "      <td>8.149366</td>\n",
       "      <td>6.730225</td>\n",
       "      <td>5.874200</td>\n",
       "      <td>5.353390</td>\n",
       "      <td>5.365097</td>\n",
       "      <td>8.872877</td>\n",
       "      <td>...</td>\n",
       "      <td>8.611371</td>\n",
       "      <td>6.614811</td>\n",
       "      <td>11.006116</td>\n",
       "      <td>7.551945</td>\n",
       "      <td>6.484451</td>\n",
       "      <td>6.771495</td>\n",
       "      <td>5.717184</td>\n",
       "      <td>5.609675</td>\n",
       "      <td>7.148153</td>\n",
       "      <td>7.305261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MB-0046</td>\n",
       "      <td>5.520658</td>\n",
       "      <td>5.556175</td>\n",
       "      <td>6.511966</td>\n",
       "      <td>9.690913</td>\n",
       "      <td>8.949934</td>\n",
       "      <td>5.417182</td>\n",
       "      <td>5.469190</td>\n",
       "      <td>5.181842</td>\n",
       "      <td>7.397672</td>\n",
       "      <td>...</td>\n",
       "      <td>8.526717</td>\n",
       "      <td>7.703336</td>\n",
       "      <td>7.738962</td>\n",
       "      <td>7.294387</td>\n",
       "      <td>8.326490</td>\n",
       "      <td>9.456882</td>\n",
       "      <td>6.112901</td>\n",
       "      <td>6.165212</td>\n",
       "      <td>8.016967</td>\n",
       "      <td>5.850510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MB-0050</td>\n",
       "      <td>5.553870</td>\n",
       "      <td>7.874150</td>\n",
       "      <td>6.072143</td>\n",
       "      <td>9.028392</td>\n",
       "      <td>7.074297</td>\n",
       "      <td>5.399167</td>\n",
       "      <td>5.223366</td>\n",
       "      <td>6.589710</td>\n",
       "      <td>7.625433</td>\n",
       "      <td>...</td>\n",
       "      <td>8.284187</td>\n",
       "      <td>8.118200</td>\n",
       "      <td>10.092764</td>\n",
       "      <td>6.739002</td>\n",
       "      <td>8.355693</td>\n",
       "      <td>7.460314</td>\n",
       "      <td>7.397655</td>\n",
       "      <td>5.381572</td>\n",
       "      <td>7.639669</td>\n",
       "      <td>7.079205</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1412</th>\n",
       "      <td>MB-7289</td>\n",
       "      <td>5.311013</td>\n",
       "      <td>6.588677</td>\n",
       "      <td>5.627716</td>\n",
       "      <td>9.365496</td>\n",
       "      <td>7.828994</td>\n",
       "      <td>5.240209</td>\n",
       "      <td>5.473240</td>\n",
       "      <td>6.396629</td>\n",
       "      <td>7.374513</td>\n",
       "      <td>...</td>\n",
       "      <td>8.609249</td>\n",
       "      <td>7.780088</td>\n",
       "      <td>10.208355</td>\n",
       "      <td>6.374919</td>\n",
       "      <td>8.955945</td>\n",
       "      <td>7.438374</td>\n",
       "      <td>8.345558</td>\n",
       "      <td>5.551011</td>\n",
       "      <td>6.389354</td>\n",
       "      <td>6.075920</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1413</th>\n",
       "      <td>MB-7291</td>\n",
       "      <td>5.639454</td>\n",
       "      <td>5.806753</td>\n",
       "      <td>5.801780</td>\n",
       "      <td>9.135438</td>\n",
       "      <td>9.178206</td>\n",
       "      <td>5.490531</td>\n",
       "      <td>5.689773</td>\n",
       "      <td>7.174227</td>\n",
       "      <td>7.855154</td>\n",
       "      <td>...</td>\n",
       "      <td>8.796033</td>\n",
       "      <td>7.243271</td>\n",
       "      <td>9.858089</td>\n",
       "      <td>6.272637</td>\n",
       "      <td>7.444408</td>\n",
       "      <td>8.620689</td>\n",
       "      <td>7.491018</td>\n",
       "      <td>6.134674</td>\n",
       "      <td>8.126522</td>\n",
       "      <td>6.458758</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1414</th>\n",
       "      <td>MB-7292</td>\n",
       "      <td>5.320487</td>\n",
       "      <td>5.475669</td>\n",
       "      <td>5.907214</td>\n",
       "      <td>8.669601</td>\n",
       "      <td>10.047511</td>\n",
       "      <td>5.196646</td>\n",
       "      <td>5.485982</td>\n",
       "      <td>5.566037</td>\n",
       "      <td>7.430706</td>\n",
       "      <td>...</td>\n",
       "      <td>8.587751</td>\n",
       "      <td>6.867430</td>\n",
       "      <td>10.248435</td>\n",
       "      <td>7.438962</td>\n",
       "      <td>7.759453</td>\n",
       "      <td>7.811448</td>\n",
       "      <td>6.195857</td>\n",
       "      <td>5.710864</td>\n",
       "      <td>8.460674</td>\n",
       "      <td>6.164186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1415</th>\n",
       "      <td>MB-7293</td>\n",
       "      <td>5.472402</td>\n",
       "      <td>5.459333</td>\n",
       "      <td>5.476251</td>\n",
       "      <td>9.409805</td>\n",
       "      <td>8.348674</td>\n",
       "      <td>5.071341</td>\n",
       "      <td>5.424013</td>\n",
       "      <td>6.719742</td>\n",
       "      <td>7.437116</td>\n",
       "      <td>...</td>\n",
       "      <td>8.656620</td>\n",
       "      <td>7.449284</td>\n",
       "      <td>9.302515</td>\n",
       "      <td>6.397087</td>\n",
       "      <td>7.558685</td>\n",
       "      <td>7.019974</td>\n",
       "      <td>8.311320</td>\n",
       "      <td>5.593818</td>\n",
       "      <td>6.092144</td>\n",
       "      <td>6.310746</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1416</th>\n",
       "      <td>MB-7296</td>\n",
       "      <td>5.461060</td>\n",
       "      <td>5.399961</td>\n",
       "      <td>6.095062</td>\n",
       "      <td>8.967495</td>\n",
       "      <td>10.303957</td>\n",
       "      <td>5.155927</td>\n",
       "      <td>5.914946</td>\n",
       "      <td>5.672096</td>\n",
       "      <td>7.151103</td>\n",
       "      <td>...</td>\n",
       "      <td>7.989299</td>\n",
       "      <td>5.902987</td>\n",
       "      <td>10.212643</td>\n",
       "      <td>6.364783</td>\n",
       "      <td>11.395064</td>\n",
       "      <td>8.287243</td>\n",
       "      <td>7.185083</td>\n",
       "      <td>5.512210</td>\n",
       "      <td>8.437577</td>\n",
       "      <td>6.154567</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1417 rows × 135 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 22
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-25T05:49:18.711151Z",
     "start_time": "2026-01-25T05:49:18.663874Z"
    }
   },
   "cell_type": "code",
   "source": [
    "cols = [\"PATIENT_ID\"] + [c for c in data_driven_integrate_2.columns if c.endswith(\"_mrna\")]\n",
    "data_driven_integrate_2_mrna = data_driven_integrate_2[cols]\n",
    "data_driven_integrate_2_mrna"
   ],
   "id": "21b95f774fabdadf",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "     PATIENT_ID  SHMT2_mrna  KRT8_mrna  RANBP6_mrna  C14orf184_mrna  GLA_mrna  \\\n",
       "0       MB-0006    9.431167  12.925422     6.761162        5.757450  8.485949   \n",
       "1       MB-0028   10.367760  10.294378     6.343706        5.689489  8.731710   \n",
       "2       MB-0035    9.837096   9.838600     6.414310        6.037567  8.149366   \n",
       "3       MB-0046   10.387976  11.794499     6.072853        6.511966  9.690913   \n",
       "4       MB-0050    9.488406   9.456052     6.514110        6.072143  9.028392   \n",
       "...         ...         ...        ...          ...             ...       ...   \n",
       "1412    MB-7289    9.802586  10.208570     6.732579        5.627716  9.365496   \n",
       "1413    MB-7291   10.365286  10.840858     6.050577        5.801780  9.135438   \n",
       "1414    MB-7292   10.063176  10.210223     6.100147        5.907214  8.669601   \n",
       "1415    MB-7293    9.271953   9.792816     6.604081        5.476251  9.409805   \n",
       "1416    MB-7296   11.022124  10.133212     6.780975        6.095062  8.967495   \n",
       "\n",
       "      ZNF214_mrna  EFR3A_mrna  ANKAR_mrna  KIAA1958_mrna  ...  DNAH11_mrna  \\\n",
       "0        5.956145   10.742071    5.758045       5.509959  ...     5.157518   \n",
       "1        5.486175   10.219419    5.832128       6.035359  ...     5.210486   \n",
       "2        5.793395    9.525547    5.827017       6.157515  ...     5.232120   \n",
       "3        5.655006    9.368818    5.755084       5.769001  ...     5.282381   \n",
       "4        5.567433    8.983147    5.911851       5.652870  ...     5.327767   \n",
       "...           ...         ...         ...            ...  ...          ...   \n",
       "1412     5.435067    8.996555    5.777710       5.708150  ...     5.295160   \n",
       "1413     5.451489    9.861437    5.711542       5.462414  ...     5.297712   \n",
       "1414     5.431562    9.570582    5.638522       5.861783  ...     5.296351   \n",
       "1415     5.853228    9.328939    5.909755       5.524639  ...     5.341369   \n",
       "1416     5.569103    9.660797    5.548795       5.581806  ...     5.236329   \n",
       "\n",
       "      ALDOC_mrna  GRB2_mrna  MMP3_mrna  SELENBP1_mrna  GPR156_mrna  \\\n",
       "0       7.130907   6.655574   6.375535       7.660406     5.804127   \n",
       "1       7.429350  10.484214   5.335386       8.973843     5.705585   \n",
       "2       6.484451   6.771495   5.717184      10.820817     5.658895   \n",
       "3       8.326490   9.456882   6.112901       9.732011     5.686241   \n",
       "4       8.355693   7.460314   7.397655       9.311124     5.677137   \n",
       "...          ...        ...        ...            ...          ...   \n",
       "1412    8.955945   7.438374   8.345558       8.963071     5.424336   \n",
       "1413    7.444408   8.620689   7.491018       6.819179     5.388680   \n",
       "1414    7.759453   7.811448   6.195857       8.892469     5.527376   \n",
       "1415    7.558685   7.019974   8.311320       8.512068     5.601245   \n",
       "1416   11.395064   8.287243   7.185083       9.368818     5.480140   \n",
       "\n",
       "      VANGL1_mrna  NKX2-2_mrna  NCEH1_mrna  UMODL1_mrna  \n",
       "0        5.368648     5.429745    6.994440     5.443783  \n",
       "1        5.495902     5.330070    6.666313     5.353278  \n",
       "2        5.609675     6.226791    8.741210     7.520124  \n",
       "3        6.165212     5.496586    7.395477     5.470029  \n",
       "4        5.381572     5.323528    6.517581     5.254414  \n",
       "...           ...          ...         ...          ...  \n",
       "1412     5.551011     5.462697    6.555181     5.689578  \n",
       "1413     6.134674     5.553103    7.038777     5.676197  \n",
       "1414     5.710864     5.343891    6.952544     5.645671  \n",
       "1415     5.593818     5.439948    6.963754     5.300404  \n",
       "1416     5.512210     9.180790    6.966874     5.570184  \n",
       "\n",
       "[1417 rows x 239 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PATIENT_ID</th>\n",
       "      <th>SHMT2_mrna</th>\n",
       "      <th>KRT8_mrna</th>\n",
       "      <th>RANBP6_mrna</th>\n",
       "      <th>C14orf184_mrna</th>\n",
       "      <th>GLA_mrna</th>\n",
       "      <th>ZNF214_mrna</th>\n",
       "      <th>EFR3A_mrna</th>\n",
       "      <th>ANKAR_mrna</th>\n",
       "      <th>KIAA1958_mrna</th>\n",
       "      <th>...</th>\n",
       "      <th>DNAH11_mrna</th>\n",
       "      <th>ALDOC_mrna</th>\n",
       "      <th>GRB2_mrna</th>\n",
       "      <th>MMP3_mrna</th>\n",
       "      <th>SELENBP1_mrna</th>\n",
       "      <th>GPR156_mrna</th>\n",
       "      <th>VANGL1_mrna</th>\n",
       "      <th>NKX2-2_mrna</th>\n",
       "      <th>NCEH1_mrna</th>\n",
       "      <th>UMODL1_mrna</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MB-0006</td>\n",
       "      <td>9.431167</td>\n",
       "      <td>12.925422</td>\n",
       "      <td>6.761162</td>\n",
       "      <td>5.757450</td>\n",
       "      <td>8.485949</td>\n",
       "      <td>5.956145</td>\n",
       "      <td>10.742071</td>\n",
       "      <td>5.758045</td>\n",
       "      <td>5.509959</td>\n",
       "      <td>...</td>\n",
       "      <td>5.157518</td>\n",
       "      <td>7.130907</td>\n",
       "      <td>6.655574</td>\n",
       "      <td>6.375535</td>\n",
       "      <td>7.660406</td>\n",
       "      <td>5.804127</td>\n",
       "      <td>5.368648</td>\n",
       "      <td>5.429745</td>\n",
       "      <td>6.994440</td>\n",
       "      <td>5.443783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MB-0028</td>\n",
       "      <td>10.367760</td>\n",
       "      <td>10.294378</td>\n",
       "      <td>6.343706</td>\n",
       "      <td>5.689489</td>\n",
       "      <td>8.731710</td>\n",
       "      <td>5.486175</td>\n",
       "      <td>10.219419</td>\n",
       "      <td>5.832128</td>\n",
       "      <td>6.035359</td>\n",
       "      <td>...</td>\n",
       "      <td>5.210486</td>\n",
       "      <td>7.429350</td>\n",
       "      <td>10.484214</td>\n",
       "      <td>5.335386</td>\n",
       "      <td>8.973843</td>\n",
       "      <td>5.705585</td>\n",
       "      <td>5.495902</td>\n",
       "      <td>5.330070</td>\n",
       "      <td>6.666313</td>\n",
       "      <td>5.353278</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MB-0035</td>\n",
       "      <td>9.837096</td>\n",
       "      <td>9.838600</td>\n",
       "      <td>6.414310</td>\n",
       "      <td>6.037567</td>\n",
       "      <td>8.149366</td>\n",
       "      <td>5.793395</td>\n",
       "      <td>9.525547</td>\n",
       "      <td>5.827017</td>\n",
       "      <td>6.157515</td>\n",
       "      <td>...</td>\n",
       "      <td>5.232120</td>\n",
       "      <td>6.484451</td>\n",
       "      <td>6.771495</td>\n",
       "      <td>5.717184</td>\n",
       "      <td>10.820817</td>\n",
       "      <td>5.658895</td>\n",
       "      <td>5.609675</td>\n",
       "      <td>6.226791</td>\n",
       "      <td>8.741210</td>\n",
       "      <td>7.520124</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MB-0046</td>\n",
       "      <td>10.387976</td>\n",
       "      <td>11.794499</td>\n",
       "      <td>6.072853</td>\n",
       "      <td>6.511966</td>\n",
       "      <td>9.690913</td>\n",
       "      <td>5.655006</td>\n",
       "      <td>9.368818</td>\n",
       "      <td>5.755084</td>\n",
       "      <td>5.769001</td>\n",
       "      <td>...</td>\n",
       "      <td>5.282381</td>\n",
       "      <td>8.326490</td>\n",
       "      <td>9.456882</td>\n",
       "      <td>6.112901</td>\n",
       "      <td>9.732011</td>\n",
       "      <td>5.686241</td>\n",
       "      <td>6.165212</td>\n",
       "      <td>5.496586</td>\n",
       "      <td>7.395477</td>\n",
       "      <td>5.470029</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MB-0050</td>\n",
       "      <td>9.488406</td>\n",
       "      <td>9.456052</td>\n",
       "      <td>6.514110</td>\n",
       "      <td>6.072143</td>\n",
       "      <td>9.028392</td>\n",
       "      <td>5.567433</td>\n",
       "      <td>8.983147</td>\n",
       "      <td>5.911851</td>\n",
       "      <td>5.652870</td>\n",
       "      <td>...</td>\n",
       "      <td>5.327767</td>\n",
       "      <td>8.355693</td>\n",
       "      <td>7.460314</td>\n",
       "      <td>7.397655</td>\n",
       "      <td>9.311124</td>\n",
       "      <td>5.677137</td>\n",
       "      <td>5.381572</td>\n",
       "      <td>5.323528</td>\n",
       "      <td>6.517581</td>\n",
       "      <td>5.254414</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1412</th>\n",
       "      <td>MB-7289</td>\n",
       "      <td>9.802586</td>\n",
       "      <td>10.208570</td>\n",
       "      <td>6.732579</td>\n",
       "      <td>5.627716</td>\n",
       "      <td>9.365496</td>\n",
       "      <td>5.435067</td>\n",
       "      <td>8.996555</td>\n",
       "      <td>5.777710</td>\n",
       "      <td>5.708150</td>\n",
       "      <td>...</td>\n",
       "      <td>5.295160</td>\n",
       "      <td>8.955945</td>\n",
       "      <td>7.438374</td>\n",
       "      <td>8.345558</td>\n",
       "      <td>8.963071</td>\n",
       "      <td>5.424336</td>\n",
       "      <td>5.551011</td>\n",
       "      <td>5.462697</td>\n",
       "      <td>6.555181</td>\n",
       "      <td>5.689578</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1413</th>\n",
       "      <td>MB-7291</td>\n",
       "      <td>10.365286</td>\n",
       "      <td>10.840858</td>\n",
       "      <td>6.050577</td>\n",
       "      <td>5.801780</td>\n",
       "      <td>9.135438</td>\n",
       "      <td>5.451489</td>\n",
       "      <td>9.861437</td>\n",
       "      <td>5.711542</td>\n",
       "      <td>5.462414</td>\n",
       "      <td>...</td>\n",
       "      <td>5.297712</td>\n",
       "      <td>7.444408</td>\n",
       "      <td>8.620689</td>\n",
       "      <td>7.491018</td>\n",
       "      <td>6.819179</td>\n",
       "      <td>5.388680</td>\n",
       "      <td>6.134674</td>\n",
       "      <td>5.553103</td>\n",
       "      <td>7.038777</td>\n",
       "      <td>5.676197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1414</th>\n",
       "      <td>MB-7292</td>\n",
       "      <td>10.063176</td>\n",
       "      <td>10.210223</td>\n",
       "      <td>6.100147</td>\n",
       "      <td>5.907214</td>\n",
       "      <td>8.669601</td>\n",
       "      <td>5.431562</td>\n",
       "      <td>9.570582</td>\n",
       "      <td>5.638522</td>\n",
       "      <td>5.861783</td>\n",
       "      <td>...</td>\n",
       "      <td>5.296351</td>\n",
       "      <td>7.759453</td>\n",
       "      <td>7.811448</td>\n",
       "      <td>6.195857</td>\n",
       "      <td>8.892469</td>\n",
       "      <td>5.527376</td>\n",
       "      <td>5.710864</td>\n",
       "      <td>5.343891</td>\n",
       "      <td>6.952544</td>\n",
       "      <td>5.645671</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1415</th>\n",
       "      <td>MB-7293</td>\n",
       "      <td>9.271953</td>\n",
       "      <td>9.792816</td>\n",
       "      <td>6.604081</td>\n",
       "      <td>5.476251</td>\n",
       "      <td>9.409805</td>\n",
       "      <td>5.853228</td>\n",
       "      <td>9.328939</td>\n",
       "      <td>5.909755</td>\n",
       "      <td>5.524639</td>\n",
       "      <td>...</td>\n",
       "      <td>5.341369</td>\n",
       "      <td>7.558685</td>\n",
       "      <td>7.019974</td>\n",
       "      <td>8.311320</td>\n",
       "      <td>8.512068</td>\n",
       "      <td>5.601245</td>\n",
       "      <td>5.593818</td>\n",
       "      <td>5.439948</td>\n",
       "      <td>6.963754</td>\n",
       "      <td>5.300404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1416</th>\n",
       "      <td>MB-7296</td>\n",
       "      <td>11.022124</td>\n",
       "      <td>10.133212</td>\n",
       "      <td>6.780975</td>\n",
       "      <td>6.095062</td>\n",
       "      <td>8.967495</td>\n",
       "      <td>5.569103</td>\n",
       "      <td>9.660797</td>\n",
       "      <td>5.548795</td>\n",
       "      <td>5.581806</td>\n",
       "      <td>...</td>\n",
       "      <td>5.236329</td>\n",
       "      <td>11.395064</td>\n",
       "      <td>8.287243</td>\n",
       "      <td>7.185083</td>\n",
       "      <td>9.368818</td>\n",
       "      <td>5.480140</td>\n",
       "      <td>5.512210</td>\n",
       "      <td>9.180790</td>\n",
       "      <td>6.966874</td>\n",
       "      <td>5.570184</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1417 rows × 239 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 23
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-25T05:57:45.432037Z",
     "start_time": "2026-01-25T05:57:45.326386Z"
    }
   },
   "cell_type": "code",
   "source": [
    "ID_COL = \"PATIENT_ID\"\n",
    "\n",
    "def drop_overlap_keep_gwas(gwas_df, dd_df, id_col=ID_COL):\n",
    "    overlap = (set(gwas_df.columns) & set(dd_df.columns)) - {id_col}\n",
    "    if overlap:\n",
    "        print(f\"Dropping {len(overlap)} overlapping columns from data-driven set (keeping GWAS).\")\n",
    "        dd_df = dd_df.drop(columns=list(overlap))\n",
    "    return dd_df\n",
    "\n",
    "# integrate cohort case 1\n",
    "data_driven_integrate_1 = drop_overlap_keep_gwas(gwas_integrate, data_driven_integrate_1)\n",
    "\n",
    "# integrate cohort case 2\n",
    "data_driven_integrate_2 = drop_overlap_keep_gwas(gwas_integrate, data_driven_integrate_2)\n",
    "\n",
    "# overall cohort (match the GWAS df you merge with)\n",
    "data_driven_overall = drop_overlap_keep_gwas(gwas_overall_mRNA, data_driven_overall)"
   ],
   "id": "dcc6c46a172bfea6",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dropping 3 overlapping columns from data-driven set (keeping GWAS).\n",
      "Dropping 4 overlapping columns from data-driven set (keeping GWAS).\n",
      "Dropping 1 overlapping columns from data-driven set (keeping GWAS).\n"
     ]
    }
   ],
   "execution_count": 24
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-25T06:00:45.728390Z",
     "start_time": "2026-01-25T06:00:45.623901Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##Combine clinical+gwas+data-driven\n",
    "#integrate cohort gwas （n=1417，N=252)\n",
    "data_all_integrate_gwas = clinical_impute_integrate.merge(gwas_integrate, on=\"PATIENT_ID\", how=\"inner\") #clinical+gwas\n",
    "#integrate cohort case 1 (N=1417,N=449)\n",
    "data_all_integrate_1 = clinical_impute_integrate.merge(gwas_integrate, on=\"PATIENT_ID\", how=\"inner\") #clinical+gwas\n",
    "data_all_integrate_1 = data_all_integrate_1.merge(data_driven_integrate_1, on=\"PATIENT_ID\", how=\"inner\") # +data-driven\n",
    "data_all_integrate_1cli = clinical_impute_integrate.merge(data_driven_integrate_1, on=\"PATIENT_ID\", how=\"inner\")  #clinical+gwas\n",
    "#integrate cohort case 2 (n=1417, N=548)\n",
    "data_all_integrate_2 = clinical_impute_integrate.merge(gwas_integrate, on=\"PATIENT_ID\", how=\"inner\") #clinical+gwas\n",
    "data_all_integrate_2 = data_all_integrate_2.merge(data_driven_integrate_2, on=\"PATIENT_ID\", how=\"inner\") # +data-driven\n",
    "data_all_integrate_2cli = clinical_impute_integrate.merge(data_driven_integrate_2, on=\"PATIENT_ID\", how=\"inner\")\n",
    "#overall cohort (n=1980, N=252)\n",
    "data_all_overall_mrna = clinical_impute_overall.merge(gwas_overall_mRNA, on=\"PATIENT_ID\", how=\"inner\") #clinical+gwas (mrna)\n",
    "data_all_overall = data_all_overall_mrna.merge(data_driven_overall, on=\"PATIENT_ID\", how=\"inner\") # +data-driven\n",
    "#integrate cohort gwas （n=1980，N=153)\n",
    "data_all_overall_gwas = clinical_impute_overall.merge(gwas_overall_mRNA, on=\"PATIENT_ID\", how=\"inner\") #clinical+gwas\n",
    "#integrate cohort case1 (mrna only)\n",
    "data_driven_integrate_1_mrna = clinical_impute_integrate.merge(data_driven_integrate_1_mrna, on=\"PATIENT_ID\", how=\"inner\")\n",
    "data_driven_integrate_2_mrna = clinical_impute_integrate.merge(data_driven_integrate_2_mrna, on=\"PATIENT_ID\", how=\"inner\")"
   ],
   "id": "efbf4a1862dc5065",
   "outputs": [],
   "execution_count": 25
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-25T06:05:27.303012Z",
     "start_time": "2026-01-25T06:05:27.220894Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##split into training and testing (70% vs 30%)\n",
    "#integrate cohort gwas （n=1417，N=252)\n",
    "train_all_integrate_gwas, test_all_integrate_gwas = train_test_split(\n",
    "    data_all_integrate_gwas,\n",
    "    test_size=0.15,\n",
    "    stratify=data_all_integrate_gwas['OS_STATUS'],\n",
    "    random_state=123\n",
    ")\n",
    "X_train_integrate_gwas = train_all_integrate_gwas.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_train_integrate_gwas = train_all_integrate_gwas[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "X_test_integrate_gwas = test_all_integrate_gwas.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_test_integrate_gwas = test_all_integrate_gwas[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "\n",
    "#integrate cohort case 1 (N=1417,N=449)\n",
    "train_all_integrate_1, test_all_integrate_1 = train_test_split(\n",
    "    data_all_integrate_1,\n",
    "    test_size=0.15,\n",
    "    stratify=data_all_integrate_1['OS_STATUS'],\n",
    "    random_state=123\n",
    ")\n",
    "X_train_integrate_1 = train_all_integrate_1.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_train_integrate_1 = train_all_integrate_1[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "X_test_integrate_1 = test_all_integrate_1.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_test_integrate_1 = test_all_integrate_1[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "\n",
    "#integrate cohort case 1 +clinical (n=1417,N=226)\n",
    "train_all_integrate_1cli, test_all_integrate_1cli = train_test_split(\n",
    "    data_all_integrate_1cli,\n",
    "    test_size=0.15,\n",
    "    stratify=data_all_integrate_1cli['OS_STATUS'],\n",
    "    random_state=123\n",
    ")\n",
    "X_train_integrate_1cli = train_all_integrate_1cli.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_train_integrate_1cli = train_all_integrate_1cli[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "X_test_integrate_1cli = test_all_integrate_1cli.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_test_integrate_1cli = test_all_integrate_1cli[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "\n",
    "\n",
    "#integrate cohort case 2 (N=1417,N=548)\n",
    "train_all_integrate_2, test_all_integrate_2 = train_test_split(\n",
    "    data_all_integrate_2,\n",
    "    test_size=0.15,\n",
    "    stratify=data_all_integrate_2['OS_STATUS'],\n",
    "    random_state=123\n",
    ")\n",
    "X_train_integrate_2 = train_all_integrate_2.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_train_integrate_2 = train_all_integrate_2[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "X_test_integrate_2 = test_all_integrate_2.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_test_integrate_2 = test_all_integrate_2[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "\n",
    "#integrate cohort case 2 +clinical (n=1417,N=326)\n",
    "train_all_integrate_2cli, test_all_integrate_2cli = train_test_split(\n",
    "    data_all_integrate_2cli,\n",
    "    test_size=0.15,\n",
    "    stratify=data_all_integrate_2cli['OS_STATUS'],\n",
    "    random_state=123\n",
    ")\n",
    "X_train_integrate_2cli = train_all_integrate_2cli.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_train_integrate_2cli = train_all_integrate_2cli[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "X_test_integrate_2cli = test_all_integrate_2cli.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_test_integrate_2cli = test_all_integrate_2cli[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "\n",
    "#overall cohort mRNA+GWAS (N=1980,N=251)\n",
    "train_all_overall, test_all_overall = train_test_split(\n",
    "    data_all_overall,\n",
    "    test_size=0.15,\n",
    "    stratify=data_all_overall['OS_STATUS'],\n",
    "    random_state=123\n",
    ")\n",
    "X_train_overall = train_all_overall.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_train_overall = train_all_overall[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "X_test_overall = test_all_overall.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_test_overall = test_all_overall[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "\n",
    "#overall cohort clinical+mRNA (N=1980,N=251)\n",
    "train_all_overall_mrna, test_all_overall_mrna = train_test_split(\n",
    "    data_all_overall_mrna,\n",
    "    test_size=0.15,\n",
    "    stratify=data_all_overall_mrna['OS_STATUS'],\n",
    "    random_state=123\n",
    ")\n",
    "X_train_overall_mrna = train_all_overall_mrna.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_train_overall_mrna = train_all_overall_mrna[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "X_test_overall_mrna = test_all_overall_mrna.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_test_overall_mrna = test_all_overall_mrna[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "\n",
    "#integrate cohort gwas （n=1980，N=153)\n",
    "train_all_overall_gwas, test_all_overall_gwas = train_test_split(\n",
    "    data_all_overall_gwas,\n",
    "    test_size=0.15,\n",
    "    stratify=data_all_overall_gwas['OS_STATUS'],\n",
    "    random_state=123\n",
    ")\n",
    "X_train_overall_gwas = train_all_overall_gwas.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_train_overall_gwas = train_all_overall_gwas[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "X_test_overall_gwas = test_all_overall_gwas.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_test_overall_gwas = test_all_overall_gwas[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "\n",
    "#integrate cohort case1(mrna only)\n",
    "train_all_integrate_1_mrna, test_all_integrate_1_mrna = train_test_split(\n",
    "    data_driven_integrate_1_mrna,\n",
    "    test_size=0.15,\n",
    "    stratify=data_driven_integrate_1_mrna['OS_STATUS'],\n",
    "    random_state=123\n",
    ")\n",
    "X_train_integrate_1_mrna = train_all_integrate_1_mrna.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_train_integrate_1_mrna = train_all_integrate_1_mrna[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "X_test_integrate_1_mrna = test_all_integrate_1_mrna.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_test_integrate_1_mrna = test_all_integrate_1_mrna[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "\n",
    "#integrate cohort case2(mrna only)\n",
    "train_all_integrate_2_mrna, test_all_integrate_2_mrna = train_test_split(\n",
    "    data_driven_integrate_2_mrna,\n",
    "    test_size=0.15,\n",
    "    stratify=data_driven_integrate_2_mrna['OS_STATUS'],\n",
    "    random_state=123\n",
    ")\n",
    "X_train_integrate_2_mrna = train_all_integrate_2_mrna.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_train_integrate_2_mrna = train_all_integrate_2_mrna[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "X_test_integrate_2_mrna = test_all_integrate_2_mrna.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_test_integrate_2_mrna = test_all_integrate_2_mrna[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "\n"
   ],
   "id": "7ba3955a970c84ed",
   "outputs": [],
   "execution_count": 28
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-23T00:17:36.271102Z",
     "start_time": "2026-01-23T00:17:35.893006Z"
    }
   },
   "cell_type": "code",
   "source": [
    "## KNN imputation (genetic only + std first)\n",
    "#integrate cohort gwas (N=1417,N=252)\n",
    "Xc_train_integrate_gwas = X_train_integrate_gwas.iloc[:, :26]\n",
    "Xg_train_integrate_gwas = X_train_integrate_gwas.iloc[:, 26:] #only impute on training\n",
    "Xc_test_integrate_gwas  = X_test_integrate_gwas.iloc[:, :26]\n",
    "Xg_test_integrate_gwas  = X_test_integrate_gwas.iloc[:, 26:]\n",
    "\n",
    "scaler = StandardScaler()\n",
    "Xg_train_integrate_gwas_scaled = scaler.fit_transform(Xg_train_integrate_gwas) #std first\n",
    "\n",
    "imputer = KNNImputer(n_neighbors=5,weights=\"distance\",metric=\"nan_euclidean\") #knn\n",
    "Xg_train_integrate_gwas_imp = imputer.fit_transform(Xg_train_integrate_gwas_scaled)\n",
    "\n",
    "Xg_train_integrate_gwas_imp = pd.DataFrame( #get columns' name back\n",
    "    Xg_train_integrate_gwas_imp,\n",
    "    columns=Xg_train_integrate_gwas.columns,\n",
    "    index=Xg_train_integrate_gwas.index)\n",
    "\n",
    "X_train_integrate_gwas_imp = pd.concat([Xc_train_integrate_gwas, Xg_train_integrate_gwas_imp], axis=1)\n",
    "\n",
    "Xg_test_integrate_gwas_scaled = scaler.transform(Xg_test_integrate_gwas) #Scale and std test using TRAIN statistics\n",
    "Xg_test_integrate_gwas_imp = imputer.transform(Xg_test_integrate_gwas_scaled)\n",
    "Xg_test_integrate_gwas_imp = pd.DataFrame(\n",
    "    Xg_test_integrate_gwas_imp,\n",
    "    columns=Xg_test_integrate_gwas.columns,\n",
    "    index=Xg_test_integrate_gwas.index\n",
    ")\n",
    "X_test_integrate_gwas_imp = pd.concat([Xc_test_integrate_gwas, Xg_test_integrate_gwas_imp], axis=1)\n",
    "\n",
    "##add ID back\n",
    "X_train_integrate_gwas_imp[\"PATIENT_ID\"] = train_all_integrate_gwas[\"PATIENT_ID\"].values\n",
    "X_test_integrate_gwas_imp[\"PATIENT_ID\"]  = test_all_integrate_gwas[\"PATIENT_ID\"].values\n",
    "def move_id_first(df, id_col=\"PATIENT_ID\"): #move ID to the first column\n",
    "    cols = [id_col] + [c for c in df.columns if c != id_col]\n",
    "    return df[cols]\n",
    "X_train_integrate_gwas_imp = move_id_first(X_train_integrate_gwas_imp)\n",
    "X_test_integrate_gwas_imp  = move_id_first(X_test_integrate_gwas_imp)"
   ],
   "id": "49af578b977a1eec",
   "outputs": [],
   "execution_count": 43
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-22T17:51:17.470394Z",
     "start_time": "2026-01-22T17:51:16.660901Z"
    }
   },
   "cell_type": "code",
   "source": [
    "## KNN imputation (genetic only + std first)\n",
    "#integrate cohort case 1 (N=1417,N=446)\n",
    "Xc_train_integrate_1 = X_train_integrate_1.iloc[:, :26]\n",
    "Xg_train_integrate_1 = X_train_integrate_1.iloc[:, 26:] #only impute on training\n",
    "Xc_test_integrate_1  = X_test_integrate_1.iloc[:, :26]\n",
    "Xg_test_integrate_1  = X_test_integrate_1.iloc[:, 26:]\n",
    "\n",
    "scaler = StandardScaler()\n",
    "Xg_train_integrate_1_scaled = scaler.fit_transform(Xg_train_integrate_1) #std first\n",
    "\n",
    "imputer = KNNImputer(n_neighbors=5,weights=\"distance\",metric=\"nan_euclidean\") #knn\n",
    "Xg_train_integrate_1_imp = imputer.fit_transform(Xg_train_integrate_1_scaled)\n",
    "\n",
    "Xg_train_integrate_1_imp = pd.DataFrame( #get columns' name back\n",
    "    Xg_train_integrate_1_imp,\n",
    "    columns=Xg_train_integrate_1.columns,\n",
    "    index=Xg_train_integrate_1.index)\n",
    "\n",
    "X_train_integrate_1_imp = pd.concat([Xc_train_integrate_1, Xg_train_integrate_1_imp], axis=1)\n",
    "\n",
    "Xg_test_integrate_1_scaled = scaler.transform(Xg_test_integrate_1) #Scale and std test using TRAIN statistics\n",
    "Xg_test_integrate_1_imp = imputer.transform(Xg_test_integrate_1_scaled)\n",
    "Xg_test_integrate_1_imp = pd.DataFrame(\n",
    "    Xg_test_integrate_1_imp,\n",
    "    columns=Xg_test_integrate_1.columns,\n",
    "    index=Xg_test_integrate_1.index\n",
    ")\n",
    "X_test_integrate_1_imp = pd.concat([Xc_test_integrate_1, Xg_test_integrate_1_imp], axis=1)\n",
    "\n",
    "##add ID back\n",
    "X_train_integrate_1_imp[\"PATIENT_ID\"] = train_all_integrate_1[\"PATIENT_ID\"].values\n",
    "X_test_integrate_1_imp[\"PATIENT_ID\"]  = test_all_integrate_1[\"PATIENT_ID\"].values\n",
    "def move_id_first(df, id_col=\"PATIENT_ID\"): #move ID to the first column\n",
    "    cols = [id_col] + [c for c in df.columns if c != id_col]\n",
    "    return df[cols]\n",
    "X_train_integrate_1_imp = move_id_first(X_train_integrate_1_imp)\n",
    "X_test_integrate_1_imp  = move_id_first(X_test_integrate_1_imp)"
   ],
   "id": "b58f07e86dcdd83e",
   "outputs": [],
   "execution_count": 28
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-23T04:42:20.773047Z",
     "start_time": "2026-01-23T04:42:20.419055Z"
    }
   },
   "cell_type": "code",
   "source": [
    "## KNN imputation (genetic only + std first)\n",
    "#integrate cohort case 1 + clinical (N=1417,N=446)\n",
    "Xc_train_integrate_1cli = X_train_integrate_1cli.iloc[:, :26]\n",
    "Xg_train_integrate_1cli = X_train_integrate_1cli.iloc[:, 26:]  #only impute on training\n",
    "Xc_test_integrate_1cli = X_test_integrate_1cli.iloc[:, :26]\n",
    "Xg_test_integrate_1cli = X_test_integrate_1cli.iloc[:, 26:]\n",
    "\n",
    "scaler = StandardScaler()\n",
    "Xg_train_integrate_1cli_scaled = scaler.fit_transform(Xg_train_integrate_1cli)  #std first\n",
    "\n",
    "imputer = KNNImputer(n_neighbors=5, weights=\"distance\", metric=\"nan_euclidean\")  #knn\n",
    "Xg_train_integrate_1cli_imp = imputer.fit_transform(Xg_train_integrate_1cli_scaled)\n",
    "\n",
    "Xg_train_integrate_1cli_imp = pd.DataFrame(  #get columns' name back\n",
    "    Xg_train_integrate_1cli_imp,\n",
    "    columns=Xg_train_integrate_1cli.columns,\n",
    "    index=Xg_train_integrate_1cli.index)\n",
    "\n",
    "X_train_integrate_1cli_imp = pd.concat([Xc_train_integrate_1cli, Xg_train_integrate_1cli_imp], axis=1)\n",
    "\n",
    "Xg_test_integrate_1cli_scaled = scaler.transform(Xg_test_integrate_1cli)  #Scale and std test using TRAIN statistics\n",
    "Xg_test_integrate_1cli_imp = imputer.transform(Xg_test_integrate_1cli_scaled)\n",
    "Xg_test_integrate_1cli_imp = pd.DataFrame(\n",
    "    Xg_test_integrate_1cli_imp,\n",
    "    columns=Xg_test_integrate_1cli.columns,\n",
    "    index=Xg_test_integrate_1cli.index\n",
    ")\n",
    "X_test_integrate_1cli_imp = pd.concat([Xc_test_integrate_1cli, Xg_test_integrate_1cli_imp], axis=1)\n",
    "\n",
    "##add ID back\n",
    "X_train_integrate_1cli_imp[\"PATIENT_ID\"] = train_all_integrate_1cli[\"PATIENT_ID\"].values\n",
    "X_test_integrate_1cli_imp[\"PATIENT_ID\"] = test_all_integrate_1cli[\"PATIENT_ID\"].values\n",
    "\n",
    "\n",
    "def move_id_first(df, id_col=\"PATIENT_ID\"):  #move ID to the first column\n",
    "    cols = [id_col] + [c for c in df.columns if c != id_col]\n",
    "    return df[cols]\n",
    "\n",
    "\n",
    "X_train_integrate_1cli_imp = move_id_first(X_train_integrate_1cli_imp)\n",
    "X_test_integrate_1cli_imp = move_id_first(X_test_integrate_1cli_imp)"
   ],
   "id": "89d5b7ef1ee2aa5c",
   "outputs": [],
   "execution_count": 60
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-22T17:51:24.887943Z",
     "start_time": "2026-01-22T17:51:23.927853Z"
    }
   },
   "cell_type": "code",
   "source": [
    "## KNN imputation (genetic only + std first)\n",
    "#integrate cohort case 2 (N=1417,N=545)\n",
    "Xc_train_integrate_2 = X_train_integrate_2.iloc[:, :26]\n",
    "Xg_train_integrate_2 = X_train_integrate_2.iloc[:, 26:]  #only impute on training\n",
    "Xc_test_integrate_2 = X_test_integrate_2.iloc[:, :26]\n",
    "Xg_test_integrate_2 = X_test_integrate_2.iloc[:, 26:]\n",
    "\n",
    "scaler = StandardScaler()\n",
    "Xg_train_integrate_2_scaled = scaler.fit_transform(Xg_train_integrate_2)  #std first\n",
    "\n",
    "imputer = KNNImputer(n_neighbors=5, weights=\"distance\", metric=\"nan_euclidean\")  #knn\n",
    "Xg_train_integrate_2_imp = imputer.fit_transform(Xg_train_integrate_2_scaled)\n",
    "\n",
    "Xg_train_integrate_2_imp = pd.DataFrame(  #get columns' name back\n",
    "    Xg_train_integrate_2_imp,\n",
    "    columns=Xg_train_integrate_2.columns,\n",
    "    index=Xg_train_integrate_2.index)\n",
    "\n",
    "X_train_integrate_2_imp = pd.concat([Xc_train_integrate_2, Xg_train_integrate_2_imp], axis=1)\n",
    "\n",
    "Xg_test_integrate_2_scaled = scaler.transform(Xg_test_integrate_2)  #Scale and std test using TRAIN statistics\n",
    "Xg_test_integrate_2_imp = imputer.transform(Xg_test_integrate_2_scaled)\n",
    "Xg_test_integrate_2_imp = pd.DataFrame(\n",
    "    Xg_test_integrate_2_imp,\n",
    "    columns=Xg_test_integrate_2.columns,\n",
    "    index=Xg_test_integrate_2.index\n",
    ")\n",
    "X_test_integrate_2_imp = pd.concat([Xc_test_integrate_2, Xg_test_integrate_2_imp], axis=1)\n",
    "\n",
    "##add ID back\n",
    "X_train_integrate_2_imp[\"PATIENT_ID\"] = train_all_integrate_2[\"PATIENT_ID\"].values\n",
    "X_test_integrate_2_imp[\"PATIENT_ID\"]  = test_all_integrate_2[\"PATIENT_ID\"].values\n",
    "X_train_integrate_2_imp = move_id_first(X_train_integrate_2_imp)\n",
    "X_test_integrate_2_imp  = move_id_first(X_test_integrate_2_imp)"
   ],
   "id": "ea37d6106924379c",
   "outputs": [],
   "execution_count": 29
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-23T04:47:19.661227Z",
     "start_time": "2026-01-23T04:47:19.230432Z"
    }
   },
   "cell_type": "code",
   "source": [
    "## KNN imputation (genetic only + std first)\n",
    "#integrate cohort case 1 + clinical (N=1417,N=446)\n",
    "Xc_train_integrate_2cli = X_train_integrate_2cli.iloc[:, :26]\n",
    "Xg_train_integrate_2cli = X_train_integrate_2cli.iloc[:, 26:]  #only impute on training\n",
    "Xc_test_integrate_2cli = X_test_integrate_2cli.iloc[:, :26]\n",
    "Xg_test_integrate_2cli = X_test_integrate_2cli.iloc[:, 26:]\n",
    "\n",
    "scaler = StandardScaler()\n",
    "Xg_train_integrate_2cli_scaled = scaler.fit_transform(Xg_train_integrate_2cli)  #std first\n",
    "\n",
    "imputer = KNNImputer(n_neighbors=5, weights=\"distance\", metric=\"nan_euclidean\")  #knn\n",
    "Xg_train_integrate_2cli_imp = imputer.fit_transform(Xg_train_integrate_2cli_scaled)\n",
    "\n",
    "Xg_train_integrate_2cli_imp = pd.DataFrame(  #get columns' name back\n",
    "    Xg_train_integrate_2cli_imp,\n",
    "    columns=Xg_train_integrate_2cli.columns,\n",
    "    index=Xg_train_integrate_2cli.index)\n",
    "\n",
    "X_train_integrate_2cli_imp = pd.concat([Xc_train_integrate_2cli, Xg_train_integrate_2cli_imp], axis=1)\n",
    "\n",
    "Xg_test_integrate_2cli_scaled = scaler.transform(Xg_test_integrate_2cli)  #Scale and std test using TRAIN statistics\n",
    "Xg_test_integrate_2cli_imp = imputer.transform(Xg_test_integrate_2cli_scaled)\n",
    "Xg_test_integrate_2cli_imp = pd.DataFrame(\n",
    "    Xg_test_integrate_2cli_imp,\n",
    "    columns=Xg_test_integrate_2cli.columns,\n",
    "    index=Xg_test_integrate_2cli.index\n",
    ")\n",
    "X_test_integrate_2cli_imp = pd.concat([Xc_test_integrate_2cli, Xg_test_integrate_2cli_imp], axis=1)\n",
    "\n",
    "##add ID back\n",
    "X_train_integrate_2cli_imp[\"PATIENT_ID\"] = train_all_integrate_2cli[\"PATIENT_ID\"].values\n",
    "X_test_integrate_2cli_imp[\"PATIENT_ID\"] = test_all_integrate_2cli[\"PATIENT_ID\"].values\n",
    "\n",
    "\n",
    "def move_id_first(df, id_col=\"PATIENT_ID\"):  #move ID to the first column\n",
    "    cols = [id_col] + [c for c in df.columns if c != id_col]\n",
    "    return df[cols]\n",
    "\n",
    "\n",
    "X_train_integrate_2cli_imp = move_id_first(X_train_integrate_2cli_imp)\n",
    "X_test_integrate_2cli_imp = move_id_first(X_test_integrate_2cli_imp)"
   ],
   "id": "85425a462cb5c5fb",
   "outputs": [],
   "execution_count": 62
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-22T17:51:26.337521Z",
     "start_time": "2026-01-22T17:51:26.312826Z"
    }
   },
   "cell_type": "code",
   "source": [
    "## KNN imputation (genetic only + std first)\n",
    "#overall cohort (N=1980,N=248)\n",
    "Xc_train_overall = X_train_overall.iloc[:, :26]\n",
    "Xg_train_overall = X_train_overall.iloc[:, 26:]  #only impute on training\n",
    "Xc_test_overall = X_test_overall.iloc[:, :26]\n",
    "Xg_test_overall = X_test_overall.iloc[:, 26:]\n",
    "\n",
    "scaler = StandardScaler()\n",
    "Xg_train_overall_scaled = scaler.fit_transform(Xg_train_overall)  #std first\n",
    "\n",
    "imputer = KNNImputer(n_neighbors=5, weights=\"distance\", metric=\"nan_euclidean\")  #knn\n",
    "Xg_train_overall_imp = imputer.fit_transform(Xg_train_overall_scaled)\n",
    "\n",
    "Xg_train_overall_imp = pd.DataFrame(  #get columns' name back\n",
    "    Xg_train_overall_imp,\n",
    "    columns=Xg_train_overall.columns,\n",
    "    index=Xg_train_overall.index)\n",
    "\n",
    "X_train_overall_imp = pd.concat([Xc_train_overall, Xg_train_overall_imp], axis=1)\n",
    "\n",
    "Xg_test_overall_scaled = scaler.transform(Xg_test_overall)  #Scale and std test using TRAIN statistics\n",
    "Xg_test_overall_imp = imputer.transform(Xg_test_overall_scaled)\n",
    "Xg_test_overall_imp = pd.DataFrame(\n",
    "    Xg_test_overall_imp,\n",
    "    columns=Xg_test_overall.columns,\n",
    "    index=Xg_test_overall.index\n",
    ")\n",
    "X_test_overall_imp = pd.concat([Xc_test_overall, Xg_test_overall_imp], axis=1)\n",
    "\n",
    "##add ID back\n",
    "X_train_overall_imp[\"PATIENT_ID\"] = train_all_overall[\"PATIENT_ID\"].values\n",
    "X_test_overall_imp[\"PATIENT_ID\"]  = test_all_overall[\"PATIENT_ID\"].values\n",
    "X_train_overall_imp = move_id_first(X_train_overall_imp)\n",
    "X_test_overall_imp  = move_id_first(X_test_overall_imp)"
   ],
   "id": "1422fb06374facbc",
   "outputs": [],
   "execution_count": 30
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-23T05:23:18.328721Z",
     "start_time": "2026-01-23T05:23:18.296798Z"
    }
   },
   "cell_type": "code",
   "source": [
    "## KNN imputation (genetic only + std first)\n",
    "#overall cohort mrna (N=1980,N=248)\n",
    "Xc_train_overall_mrna = X_train_overall_mrna.iloc[:, :26]\n",
    "Xg_train_overall_mrna = X_train_overall_mrna.iloc[:, 26:]  #only impute on training\n",
    "Xc_test_overall_mrna = X_test_overall_mrna.iloc[:, :26]\n",
    "Xg_test_overall_mrna = X_test_overall_mrna.iloc[:, 26:]\n",
    "\n",
    "scaler = StandardScaler()\n",
    "Xg_train_overall_mrna_scaled = scaler.fit_transform(Xg_train_overall_mrna)  #std first\n",
    "\n",
    "imputer = KNNImputer(n_neighbors=5, weights=\"distance\", metric=\"nan_euclidean\")  #knn\n",
    "Xg_train_overall_mrna_imp = imputer.fit_transform(Xg_train_overall_mrna_scaled)\n",
    "\n",
    "Xg_train_overall_mrna_imp = pd.DataFrame(  #get columns' name back\n",
    "    Xg_train_overall_mrna_imp,\n",
    "    columns=Xg_train_overall_mrna.columns,\n",
    "    index=Xg_train_overall_mrna.index)\n",
    "\n",
    "X_train_overall_mrna_imp = pd.concat([Xc_train_overall_mrna, Xg_train_overall_mrna_imp], axis=1)\n",
    "\n",
    "Xg_test_overall_mrna_scaled = scaler.transform(Xg_test_overall_mrna)  #Scale and std test using TRAIN statistics\n",
    "Xg_test_overall_mrna_imp = imputer.transform(Xg_test_overall_mrna_scaled)\n",
    "Xg_test_overall_mrna_imp = pd.DataFrame(\n",
    "    Xg_test_overall_mrna_imp,\n",
    "    columns=Xg_test_overall_mrna.columns,\n",
    "    index=Xg_test_overall_mrna.index\n",
    ")\n",
    "X_test_overall_mrna_imp = pd.concat([Xc_test_overall_mrna, Xg_test_overall_mrna_imp], axis=1)\n",
    "\n",
    "##add ID back\n",
    "X_train_overall_mrna_imp[\"PATIENT_ID\"] = train_all_overall_mrna[\"PATIENT_ID\"].values\n",
    "X_test_overall_mrna_imp[\"PATIENT_ID\"] = test_all_overall_mrna[\"PATIENT_ID\"].values\n",
    "X_train_overall_mrna_imp = move_id_first(X_train_overall_mrna_imp)\n",
    "X_test_overall_mrna_imp = move_id_first(X_test_overall_mrna_imp)"
   ],
   "id": "b59227649fd93c77",
   "outputs": [],
   "execution_count": 68
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-23T00:28:05.373288Z",
     "start_time": "2026-01-23T00:28:05.335376Z"
    }
   },
   "cell_type": "code",
   "source": [
    "#integrate cohort gwas (N=1980,N=153)\n",
    "Xc_train_overall_gwas = X_train_overall_gwas.iloc[:, :26]\n",
    "Xg_train_overall_gwas = X_train_overall_gwas.iloc[:, 26:] #only impute on training\n",
    "Xc_test_overall_gwas  = X_test_overall_gwas.iloc[:, :26]\n",
    "Xg_test_overall_gwas  = X_test_overall_gwas.iloc[:, 26:]\n",
    "\n",
    "scaler = StandardScaler()\n",
    "Xg_train_overall_gwas_scaled = scaler.fit_transform(Xg_train_overall_gwas) #std first\n",
    "\n",
    "imputer = KNNImputer(n_neighbors=5,weights=\"distance\",metric=\"nan_euclidean\") #knn\n",
    "Xg_train_overall_gwas_imp = imputer.fit_transform(Xg_train_overall_gwas_scaled)\n",
    "\n",
    "Xg_train_overall_gwas_imp = pd.DataFrame( #get columns' name back\n",
    "    Xg_train_overall_gwas_imp,\n",
    "    columns=Xg_train_overall_gwas.columns,\n",
    "    index=Xg_train_overall_gwas.index)\n",
    "\n",
    "X_train_overall_gwas_imp = pd.concat([Xc_train_overall_gwas, Xg_train_overall_gwas_imp], axis=1)\n",
    "\n",
    "Xg_test_overall_gwas_scaled = scaler.transform(Xg_test_overall_gwas) #Scale and std test using TRAIN statistics\n",
    "Xg_test_overall_gwas_imp = imputer.transform(Xg_test_overall_gwas_scaled)\n",
    "Xg_test_overall_gwas_imp = pd.DataFrame(\n",
    "    Xg_test_overall_gwas_imp,\n",
    "    columns=Xg_test_overall_gwas.columns,\n",
    "    index=Xg_test_overall_gwas.index\n",
    ")\n",
    "X_test_overall_gwas_imp = pd.concat([Xc_test_overall_gwas, Xg_test_overall_gwas_imp], axis=1)\n",
    "\n",
    "##add ID back\n",
    "X_train_overall_gwas_imp[\"PATIENT_ID\"] = train_all_overall_gwas[\"PATIENT_ID\"].values\n",
    "X_test_overall_gwas_imp[\"PATIENT_ID\"]  = test_all_overall_gwas[\"PATIENT_ID\"].values\n",
    "def move_id_first(df, id_col=\"PATIENT_ID\"): #move ID to the first column\n",
    "    cols = [id_col] + [c for c in df.columns if c != id_col]\n",
    "    return df[cols]\n",
    "X_train_overall_gwas_imp = move_id_first(X_train_overall_gwas_imp)\n",
    "X_test_overall_gwas_imp  = move_id_first(X_test_overall_gwas_imp)"
   ],
   "id": "5396fa4f60df029b",
   "outputs": [],
   "execution_count": 51
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-25T06:12:30.522512Z",
     "start_time": "2026-01-25T06:12:30.487545Z"
    }
   },
   "cell_type": "code",
   "source": [
    "#integrate cohort case 1(mrna only)\n",
    "Xc_train_integrate_1_mrna = X_train_integrate_1_mrna.iloc[:, :26]\n",
    "Xg_train_integrate_1_mrna = X_train_integrate_1_mrna.iloc[:, 26:] #only impute on training\n",
    "Xc_test_integrate_1_mrna  = X_test_integrate_1_mrna.iloc[:, :26]\n",
    "Xg_test_integrate_1_mrna  = X_test_integrate_1_mrna.iloc[:, 26:]\n",
    "\n",
    "scaler = StandardScaler()\n",
    "Xg_train_integrate_1_mrna_scaled = scaler.fit_transform(Xg_train_integrate_1_mrna) #std first\n",
    "\n",
    "imputer = KNNImputer(n_neighbors=5,weights=\"distance\",metric=\"nan_euclidean\") #knn\n",
    "Xg_train_integrate_1_mrna_imp = imputer.fit_transform(Xg_train_integrate_1_mrna_scaled)\n",
    "\n",
    "Xg_train_integrate_1_mrna_imp = pd.DataFrame( #get columns' name back\n",
    "    Xg_train_integrate_1_mrna_imp,\n",
    "    columns=Xg_train_integrate_1_mrna.columns,\n",
    "    index=Xg_train_integrate_1_mrna.index)\n",
    "\n",
    "X_train_integrate_1_mrna_imp = pd.concat([Xc_train_integrate_1_mrna, Xg_train_integrate_1_mrna_imp], axis=1)\n",
    "\n",
    "Xg_test_integrate_1_mrna_scaled = scaler.transform(Xg_test_integrate_1_mrna) #Scale and std test using TRAIN statistics\n",
    "Xg_test_integrate_1_mrna_imp = imputer.transform(Xg_test_integrate_1_mrna_scaled)\n",
    "Xg_test_integrate_1_mrna_imp = pd.DataFrame(\n",
    "    Xg_test_integrate_1_mrna_imp,\n",
    "    columns=Xg_test_integrate_1_mrna.columns,\n",
    "    index=Xg_test_integrate_1_mrna.index\n",
    ")\n",
    "X_test_integrate_1_mrna_imp = pd.concat([Xc_test_integrate_1_mrna, Xg_test_integrate_1_mrna_imp], axis=1)\n",
    "\n",
    "##add ID back\n",
    "X_train_integrate_1_mrna_imp[\"PATIENT_ID\"] = train_all_integrate_1_mrna[\"PATIENT_ID\"].values\n",
    "X_test_integrate_1_mrna_imp[\"PATIENT_ID\"]  = test_all_integrate_1_mrna[\"PATIENT_ID\"].values\n",
    "def move_id_first(df, id_col=\"PATIENT_ID\"): #move ID to the first column\n",
    "    cols = [id_col] + [c for c in df.columns if c != id_col]\n",
    "    return df[cols]\n",
    "X_train_integrate_1_mrna_imp = move_id_first(X_train_integrate_1_mrna_imp)\n",
    "X_test_integrate_1_mrna_imp  = move_id_first(X_test_integrate_1_mrna_imp)"
   ],
   "id": "ab6c7ead2b3cf0d0",
   "outputs": [],
   "execution_count": 31
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-25T06:10:52.605769Z",
     "start_time": "2026-01-25T06:10:52.553993Z"
    }
   },
   "cell_type": "code",
   "source": [
    "#integrate cohort case 2(mrna only)\n",
    "Xc_train_integrate_2_mrna = X_train_integrate_2_mrna.iloc[:, :26]\n",
    "Xg_train_integrate_2_mrna = X_train_integrate_2_mrna.iloc[:, 26:] #only impute on training\n",
    "Xc_test_integrate_2_mrna  = X_test_integrate_2_mrna.iloc[:, :26]\n",
    "Xg_test_integrate_2_mrna  = X_test_integrate_2_mrna.iloc[:, 26:]\n",
    "\n",
    "scaler = StandardScaler()\n",
    "Xg_train_integrate_2_mrna_scaled = scaler.fit_transform(Xg_train_integrate_2_mrna) #std first\n",
    "\n",
    "imputer = KNNImputer(n_neighbors=5,weights=\"distance\",metric=\"nan_euclidean\") #knn\n",
    "Xg_train_integrate_2_mrna_imp = imputer.fit_transform(Xg_train_integrate_2_mrna_scaled)\n",
    "\n",
    "Xg_train_integrate_2_mrna_imp = pd.DataFrame( #get columns' name back\n",
    "    Xg_train_integrate_2_mrna_imp,\n",
    "    columns=Xg_train_integrate_2_mrna.columns,\n",
    "    index=Xg_train_integrate_2_mrna.index)\n",
    "\n",
    "X_train_integrate_2_mrna_imp = pd.concat([Xc_train_integrate_2_mrna, Xg_train_integrate_2_mrna_imp], axis=1)\n",
    "\n",
    "Xg_test_integrate_2_mrna_scaled = scaler.transform(Xg_test_integrate_2_mrna) #Scale and std test using TRAIN statistics\n",
    "Xg_test_integrate_2_mrna_imp = imputer.transform(Xg_test_integrate_2_mrna_scaled)\n",
    "Xg_test_integrate_2_mrna_imp = pd.DataFrame(\n",
    "    Xg_test_integrate_2_mrna_imp,\n",
    "    columns=Xg_test_integrate_2_mrna.columns,\n",
    "    index=Xg_test_integrate_2_mrna.index\n",
    ")\n",
    "X_test_integrate_2_mrna_imp = pd.concat([Xc_test_integrate_2_mrna, Xg_test_integrate_2_mrna_imp], axis=1)\n",
    "\n",
    "##add ID back\n",
    "X_train_integrate_2_mrna_imp[\"PATIENT_ID\"] = train_all_integrate_2_mrna[\"PATIENT_ID\"].values\n",
    "X_test_integrate_2_mrna_imp[\"PATIENT_ID\"]  = test_all_integrate_2_mrna[\"PATIENT_ID\"].values\n",
    "def move_id_first(df, id_col=\"PATIENT_ID\"): #move ID to the first column\n",
    "    cols = [id_col] + [c for c in df.columns if c != id_col]\n",
    "    return df[cols]\n",
    "X_train_integrate_2_mrna_imp = move_id_first(X_train_integrate_2_mrna_imp)\n",
    "X_test_integrate_2_mrna_imp  = move_id_first(X_test_integrate_2_mrna_imp)"
   ],
   "id": "c499bd4b4c419472",
   "outputs": [],
   "execution_count": 30
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-22T19:15:58.478271Z",
     "start_time": "2026-01-22T19:15:56.772725Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##output all dataset ready for prediction models\n",
    "X_train_integrate_1_imp.to_csv(\"/Users/mac1/Desktop/X_train_integrate_1_imp.csv\", index=False)\n",
    "X_test_integrate_1_imp.to_csv(\"/Users/mac1/Desktop/X_test_integrate_1_imp.csv\", index=False)\n",
    "y_train_integrate_1.to_csv(\"/Users/mac1/Desktop/y_train_integrate_1.csv\", index=False)\n",
    "y_test_integrate_1.to_csv(\"/Users/mac1/Desktop/y_test_integrate_1.csv\", index=False)\n",
    "\n",
    "X_train_integrate_2_imp.to_csv(\"/Users/mac1/Desktop/X_train_integrate_2_imp.csv\", index=False)\n",
    "X_test_integrate_2_imp.to_csv(\"/Users/mac1/Desktop/X_test_integrate_2_imp.csv\", index=False)\n",
    "y_train_integrate_2.to_csv(\"/Users/mac1/Desktop/y_train_integrate_2.csv\", index=False)\n",
    "y_test_integrate_2.to_csv(\"/Users/mac1/Desktop/y_test_integrate_2.csv\", index=False)\n",
    "\n",
    "X_train_overall_imp.to_csv(\"/Users/mac1/Desktop/X_train_overall_imp.csv\", index=False)\n",
    "X_test_overall_imp.to_csv(\"/Users/mac1/Desktop/X_test_overall_imp.csv\", index=False)\n",
    "y_train_overall.to_csv(\"/Users/mac1/Desktop/y_train_overall.csv\", index=False)\n",
    "y_test_overall.to_csv(\"/Users/mac1/Desktop/y_test_overall.csv\", index=False)"
   ],
   "id": "8dcf44f40d22a512",
   "outputs": [],
   "execution_count": 36
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-23T00:18:41.734657Z",
     "start_time": "2026-01-23T00:18:41.483394Z"
    }
   },
   "cell_type": "code",
   "source": [
    "X_train_integrate_gwas_imp.to_csv(\"/Users/mac1/Desktop/X_train_integrate_gwas_imp.csv\", index=False)\n",
    "X_test_integrate_gwas_imp.to_csv(\"/Users/mac1/Desktop/X_test_integrate_gwas_imp.csv\", index=False)\n",
    "X_train_overall_gwas_imp.to_csv(\"/Users/mac1/Desktop/X_train_overall_gwas_imp.csv\", index=False)\n",
    "X_test_overall_gwas_imp.to_csv(\"/Users/mac1/Desktop/X_test_overall_gwas_imp.csv\", index=False)\n",
    "y_train_integrate_gwas.to_csv(\"/Users/mac1/Desktop/y_train_integrate_gwas.csv\", index=False)\n",
    "y_test_integrate_gwas.to_csv(\"/Users/mac1/Desktop/y_test_integrate_gwas.csv\", index=False)\n",
    "y_train_overall_gwas.to_csv(\"/Users/mac1/Desktop/y_train_overall_gwas.csv\", index=False)\n",
    "y_test_overall_gwas.to_csv(\"/Users/mac1/Desktop/y_test_overall_gwas.csv\", index=False)"
   ],
   "id": "aef4b87cf4802eed",
   "outputs": [],
   "execution_count": 45
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-23T04:49:35.507166Z",
     "start_time": "2026-01-23T04:49:34.753390Z"
    }
   },
   "cell_type": "code",
   "source": [
    "X_train_integrate_1cli_imp.to_csv(\"/Users/mac1/Desktop/X_train_integrate_1cli_imp.csv\", index=False)\n",
    "X_test_integrate_1cli_imp.to_csv(\"/Users/mac1/Desktop/X_test_integrate_1cli_imp.csv\", index=False)\n",
    "y_train_integrate_1cli.to_csv(\"/Users/mac1/Desktop/y_train_integrate_1cli.csv\", index=False)\n",
    "y_test_integrate_1cli.to_csv(\"/Users/mac1/Desktop/y_test_integrate_1cli.csv\", index=False)\n",
    "\n",
    "X_train_integrate_2cli_imp.to_csv(\"/Users/mac1/Desktop/X_train_integrate_2cli_imp.csv\", index=False)\n",
    "X_test_integrate_2cli_imp.to_csv(\"/Users/mac1/Desktop/X_test_integrate_2cli_imp.csv\", index=False)\n",
    "y_train_integrate_2cli.to_csv(\"/Users/mac1/Desktop/y_train_integrate_2cli.csv\", index=False)\n",
    "y_test_integrate_2cli.to_csv(\"/Users/mac1/Desktop/y_test_integrate_2cli.csv\", index=False)"
   ],
   "id": "9ea935a2f7dfc187",
   "outputs": [],
   "execution_count": 64
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-23T14:33:52.268534Z",
     "start_time": "2026-01-23T14:33:52.231259Z"
    }
   },
   "cell_type": "code",
   "source": [
    "df=X_train_integrate_1cli_imp\n",
    "n_methy = df.columns.str.endswith(\"_methy\").sum()\n",
    "n_mrna  = df.columns.str.endswith(\"_mrna\").sum()\n",
    "print(\"methy:\", n_methy)\n",
    "print(\"mrna :\", n_mrna)"
   ],
   "id": "4737ad0dfbe2cc00",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "methy: 66\n",
      "mrna : 131\n"
     ]
    }
   ],
   "execution_count": 79
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-23T05:26:17.114699Z",
     "start_time": "2026-01-23T05:26:16.823164Z"
    }
   },
   "cell_type": "code",
   "outputs": [],
   "execution_count": 71,
   "source": [
    "X_train_overall_mrna_imp.to_csv(\"/Users/mac1/Desktop/X_train_overall_mrna_imp.csv\", index=False)\n",
    "X_test_overall_mrna_imp.to_csv(\"/Users/mac1/Desktop/X_test_overall_mrna_imp.csv\", index=False)\n",
    "y_train_overall_mrna.to_csv(\"/Users/mac1/Desktop/y_train_overall_mrna.csv\", index=False)\n",
    "y_test_overall_mrna.to_csv(\"/Users/mac1/Desktop/y_test_overall_mrna.csv\", index=False)"
   ],
   "id": "3248d085690f458b"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-25T06:16:14.929882Z",
     "start_time": "2026-01-25T06:16:14.353995Z"
    }
   },
   "cell_type": "code",
   "source": [
    "X_train_integrate_1_mrna_imp.to_csv(\"/Users/mac1/Desktop/X_train_integrate_1_mrna_imp.csv\", index=False)\n",
    "X_test_integrate_1_mrna_imp.to_csv(\"/Users/mac1/Desktop/X_test_integrate_1_mrna_imp.csv\", index=False)\n",
    "y_train_integrate_1_mrna.to_csv(\"/Users/mac1/Desktop/y_train_integrate_1_mrna.csv\", index=False)\n",
    "y_test_integrate_1_mrna.to_csv(\"/Users/mac1/Desktop/y_test_integrate_1_mrna.csv\", index=False)\n",
    "\n",
    "X_train_integrate_2_mrna_imp.to_csv(\"/Users/mac1/Desktop/X_train_integrate_2_mrna_imp.csv\", index=False)\n",
    "X_test_integrate_2_mrna_imp.to_csv(\"/Users/mac1/Desktop/X_test_integrate_2_mrna_imp.csv\", index=False)\n",
    "y_train_integrate_2_mrna.to_csv(\"/Users/mac1/Desktop/y_train_integrate_2_mrna.csv\", index=False)\n",
    "y_test_integrate_2_mrna.to_csv(\"/Users/mac1/Desktop/y_test_integrate_2_mrna.csv\", index=False)"
   ],
   "id": "c3979abaacc5b351",
   "outputs": [],
   "execution_count": 32
  }
 ],
 "metadata": {
  "kernelspec": {
   "name": "python3",
   "language": "python",
   "display_name": "Python 3 (ipykernel)"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
