{
 "cells": [
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-20T01:33:56.121692Z",
     "start_time": "2026-01-20T01:33:54.837874Z"
    }
   },
   "cell_type": "code",
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "from sklearn.impute import KNNImputer\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split"
   ],
   "id": "a05b393c220d6ffd",
   "outputs": [],
   "execution_count": 2
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-20T02:56:50.948319Z",
     "start_time": "2026-01-20T02:56:32.084872Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##upload the original mRNA and methylation\n",
    "#interated mRNA and methylation (n=1417) without any operation and exclusion of features\n",
    "methy_sub = pd.read_csv('/Users/mac1/Desktop/Test_Train/methylation_sub.csv') #(1417, 13188)\n",
    "mRNA_sub = pd.read_csv('/Users/mac1/Desktop/Test_Train/mRNA_sub.csv') #(1417, 20604)\n",
    "#overall mRNA (n=1980) without any operation and exclusion of features\n",
    "mRNA_genetic = pd.read_csv('/Users/mac1/Desktop/Test_Train/mRNA_genetic.csv') #(1980, 20604)"
   ],
   "id": "92d7f60a08c32b03",
   "outputs": [],
   "execution_count": 28
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-20T02:56:54.091818Z",
     "start_time": "2026-01-20T02:56:54.057423Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##extract ID row index for cohort\n",
    "keep_ids_overall = mRNA_genetic['PATIENT_ID'] #overall\n",
    "keep_ids_integrate = methy_sub['PATIENT_ID'] #integrate\n",
    "##extract feature sets for methylation or mRNA\n",
    "methy_features = [c for c in methy_sub.columns if c not in [\"PATIENT_ID\"]]\n",
    "mRNA_features = [c for c in mRNA_sub.columns if c not in [\"PATIENT_ID\"]]"
   ],
   "id": "e2a781aa876b456",
   "outputs": [],
   "execution_count": 29
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-20T03:28:55.795542Z",
     "start_time": "2026-01-20T03:28:55.764641Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##clinical features ï¼ˆoverall cohort)\n",
    "clinical_impute_overall = pd.read_csv(\"/Users/mac1/Desktop/DATAFEST/clinical_imputed_1.csv\")\n",
    "clinical_impute_overall.shape"
   ],
   "id": "844d0bc2bf2b920f",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1980, 29)"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 51
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-20T03:28:57.739747Z",
     "start_time": "2026-01-20T03:28:57.691038Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##clinical feature (integrated cohort)\n",
    "clinical_impute_integrate = clinical_impute_overall[clinical_impute_overall['PATIENT_ID'].isin(keep_ids_integrate)].copy()\n",
    "clinical_impute_integrate.shape"
   ],
   "id": "97a95800c68c3c69",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1417, 29)"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 52
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-20T03:29:10.039660Z",
     "start_time": "2026-01-20T03:29:09.950925Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##gwas feature (overall cohort)\n",
    "gwas=pd.read_csv('/Users/mac1/Desktop/DATAFEST/genomics_selected_gwas_ensembl.csv',index_col=0)\n",
    "gwas_overall = gwas[gwas['PATIENT_ID'].isin(keep_ids_overall)].copy()"
   ],
   "id": "ecea1239e13f7474",
   "outputs": [],
   "execution_count": 53
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-20T03:29:11.150214Z",
     "start_time": "2026-01-20T03:29:11.110230Z"
    }
   },
   "cell_type": "code",
   "source": [
    "#99 methylation in the combined overall gwas feature set, may not be used\n",
    "gwas_overall_methy = gwas_overall[[\"PATIENT_ID\"] + list(gwas_overall.columns.intersection(methy_features))]\n",
    "gwas_overall_methy.shape"
   ],
   "id": "2b6d2dbd96e58c12",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1980, 99)"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 54
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-20T03:29:13.835649Z",
     "start_time": "2026-01-20T03:29:13.792088Z"
    }
   },
   "cell_type": "code",
   "source": [
    "#125 mRNA in the combined overall gwas feature set\n",
    "gwas_overall_mRNA = gwas_overall[[\"PATIENT_ID\"] + list(gwas_overall.columns.intersection(mRNA_features))]\n",
    "gwas_overall_mRNA.shape"
   ],
   "id": "15dc1b0fc468680",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1980, 125)"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 55
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-20T03:29:16.173098Z",
     "start_time": "2026-01-20T03:29:16.138166Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##gwas feature (integrated cohort)\n",
    "gwas_integrate = gwas[gwas['PATIENT_ID'].isin(keep_ids_integrate)].copy()\n",
    "gwas_integrate.shape"
   ],
   "id": "b5efe432890356c2",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1417, 224)"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 56
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-20T03:02:10.042534Z",
     "start_time": "2026-01-20T03:02:10.004906Z"
    }
   },
   "cell_type": "code",
   "source": [
    "gwas_integrate_methy = gwas_integrate[[\"PATIENT_ID\"] + list(gwas_integrate.columns.intersection(methy_features))]\n",
    "gwas_integrate_methy.shape"
   ],
   "id": "2c52edad73e5187a",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1417, 99)"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 34
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-20T04:24:11.449440Z",
     "start_time": "2026-01-20T04:24:11.395114Z"
    }
   },
   "cell_type": "code",
   "source": [
    "gwas_integrate_mRNA = gwas_integrate[[\"PATIENT_ID\"] + list(gwas_integrate.columns.intersection(mRNA_features))]\n",
    "gwas_integrate_mRNA.shape"
   ],
   "id": "8e2045c247503d4a",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1417, 125)"
      ]
     },
     "execution_count": 135,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 135
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-20T03:31:33.410488Z",
     "start_time": "2026-01-20T03:31:33.355466Z"
    }
   },
   "cell_type": "code",
   "source": [
    "nan_counts_mRNA = gwas_overall_mRNA.isna().sum()\n",
    "nan_counts_meth = gwas_integrate_methy.isna().sum()\n",
    "missing_pct_mRNA= gwas_overall_mRNA.isna().mean() * 100\n",
    "missing_pct_meth= gwas_integrate_methy.isna().mean() * 100\n",
    "print(\"Columns with NaN in mRNA in GWAS overall:\", (nan_counts_mRNA > 0).sum())\n",
    "print(\"Columns with NaN in methylation in GWAS integrated:\", (nan_counts_meth > 0).sum())\n",
    "print(\"Percentage of missing in mRNA in GWAS overall\",missing_pct_mRNA.describe())\n",
    "print(\"Percentage of missing in methylation in GWAS integrated\",missing_pct_meth.describe())\n",
    "##the missingness percentage is acceptable, let's move on for KNN imputation"
   ],
   "id": "65aef04dcf93a1c",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Columns with NaN in mRNA in GWAS overall: 0\n",
      "Columns with NaN in methylation in GWAS integrated: 75\n",
      "Percentage of missing in mRNA in GWAS overall count    125.0\n",
      "mean       0.0\n",
      "std        0.0\n",
      "min        0.0\n",
      "25%        0.0\n",
      "50%        0.0\n",
      "75%        0.0\n",
      "max        0.0\n",
      "dtype: float64\n",
      "Percentage of missing in methylation in GWAS integrated count    99.000000\n",
      "mean      3.758830\n",
      "std       6.782153\n",
      "min       0.000000\n",
      "25%       0.070572\n",
      "50%       0.423430\n",
      "75%       3.705011\n",
      "max      28.863797\n",
      "dtype: float64\n"
     ]
    }
   ],
   "execution_count": 58
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-20T03:33:08.965996Z",
     "start_time": "2026-01-20T03:33:08.913671Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##data-driven feature set\n",
    "#integrated cohort case 1 (n=1417, N=200)\n",
    "var_data_driven_integrate_1 = pd.read_csv('/Users/mac1/Desktop/DATAFEST/strategy_3_Importance_Cohort1_Case1_SeparateCox.csv')\n",
    "top200_features_1 = var_data_driven_integrate_1[\"feature_name\"].head(200).tolist()\n",
    "##integrated cohort case 2 (n=1417, N=300)\n",
    "var_data_driven_integrate_2 = pd.read_csv('/Users/mac1/Desktop/DATAFEST/strategy_3_Importance_Cohort1_Case2_CombinedCox.csv')\n",
    "top300_features_2 = var_data_driven_integrate_2[\"feature_name\"].head(300).tolist()\n",
    "#overall cohort (n=1980, N mRNA=100)\n",
    "var_data_driven_overall = pd.read_csv('/Users/mac1/Desktop/DATAFEST/strategy_3_Importance_C2_Case1_mRNA_only.csv')\n",
    "top100_features_3 = var_data_driven_overall[\"feature_name\"].head(100).tolist()"
   ],
   "id": "b790d68d926e2681",
   "outputs": [],
   "execution_count": 64
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-20T03:33:25.803892Z",
     "start_time": "2026-01-20T03:33:25.330576Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##data-driven dataset (integrate cohort case 1)\n",
    "genetic_sub = methy_sub.merge(mRNA_sub, on=\"PATIENT_ID\", how=\"inner\")\n",
    "data_driven_integrate_1 = genetic_sub[[\"PATIENT_ID\"] + list(genetic_sub.columns.intersection(top200_features_1))]\n",
    "##data-driven dataset (integrate cohort case 2)\n",
    "data_driven_integrate_2 = genetic_sub[[\"PATIENT_ID\"] + list(genetic_sub.columns.intersection(top300_features_2))]\n",
    "##data-driven dataset (overall cohort)\n",
    "data_driven_overall = mRNA_genetic[[\"PATIENT_ID\"] + list(mRNA_genetic.columns.intersection(top100_features_3))]"
   ],
   "id": "83bc3c93dcef9e61",
   "outputs": [],
   "execution_count": 66
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-20T03:33:39.999233Z",
     "start_time": "2026-01-20T03:33:39.948741Z"
    }
   },
   "cell_type": "code",
   "source": [
    "ID_COL = \"PATIENT_ID\"\n",
    "\n",
    "def drop_overlap_keep_gwas(gwas_df, dd_df, id_col=ID_COL):\n",
    "    overlap = (set(gwas_df.columns) & set(dd_df.columns)) - {id_col}\n",
    "    if overlap:\n",
    "        print(f\"Dropping {len(overlap)} overlapping columns from data-driven set (keeping GWAS).\")\n",
    "        dd_df = dd_df.drop(columns=list(overlap))\n",
    "    return dd_df\n",
    "\n",
    "# integrate cohort case 1\n",
    "data_driven_integrate_1 = drop_overlap_keep_gwas(gwas_integrate, data_driven_integrate_1)\n",
    "\n",
    "# integrate cohort case 2\n",
    "data_driven_integrate_2 = drop_overlap_keep_gwas(gwas_integrate, data_driven_integrate_2)\n",
    "\n",
    "# overall cohort (match the GWAS df you merge with)\n",
    "data_driven_overall = drop_overlap_keep_gwas(gwas_overall_mRNA, data_driven_overall)"
   ],
   "id": "dcc6c46a172bfea6",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dropping 3 overlapping columns from data-driven set (keeping GWAS).\n",
      "Dropping 4 overlapping columns from data-driven set (keeping GWAS).\n",
      "Dropping 1 overlapping columns from data-driven set (keeping GWAS).\n"
     ]
    }
   ],
   "execution_count": 68
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-20T03:33:45.624369Z",
     "start_time": "2026-01-20T03:33:45.542945Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##Combine clinical+gwas+data-driven\n",
    "#integrate cohort case 1 (N=1417,N=449)\n",
    "data_all_integrate_1 = clinical_impute_integrate.merge(gwas_integrate, on=\"PATIENT_ID\", how=\"inner\") #clinical+gwas\n",
    "data_all_integrate_1 = data_all_integrate_1.merge(data_driven_integrate_1, on=\"PATIENT_ID\", how=\"inner\") # +data-driven\n",
    "#integrate cohort case 2 (n=1417, N=548)\n",
    "data_all_integrate_2 = clinical_impute_integrate.merge(gwas_integrate, on=\"PATIENT_ID\", how=\"inner\") #clinical+gwas\n",
    "data_all_integrate_2 = data_all_integrate_2.merge(data_driven_integrate_2, on=\"PATIENT_ID\", how=\"inner\") # +data-driven\n",
    "#overall cohort (n=1417, N=252)\n",
    "data_all_overall = clinical_impute_overall.merge(gwas_overall_mRNA, on=\"PATIENT_ID\", how=\"inner\") #clinical+gwas (mrna)\n",
    "data_all_overall = data_all_overall.merge(data_driven_overall, on=\"PATIENT_ID\", how=\"inner\") # +data-driven"
   ],
   "id": "efbf4a1862dc5065",
   "outputs": [],
   "execution_count": 69
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-20T04:06:46.705782Z",
     "start_time": "2026-01-20T04:06:46.654034Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##split into training and testing (70% vs 30%)\n",
    "#integrate cohort case 1 (N=1417,N=449)\n",
    "train_all_integrate_1, test_all_integrate_1 = train_test_split(\n",
    "    data_all_integrate_1,\n",
    "    test_size=0.20,\n",
    "    stratify=data_all_integrate_1['OS_STATUS'],\n",
    "    random_state=123\n",
    ")\n",
    "X_train_integrate_1 = train_all_integrate_1.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_train_integrate_1 = train_all_integrate_1[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "X_test_integrate_1 = test_all_integrate_1.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_test_integrate_1 = test_all_integrate_1[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "\n",
    "#integrate cohort case 2 (N=1417,N=548)\n",
    "train_all_integrate_2, test_all_integrate_2 = train_test_split(\n",
    "    data_all_integrate_2,\n",
    "    test_size=0.20,\n",
    "    stratify=data_all_integrate_2['OS_STATUS'],\n",
    "    random_state=123\n",
    ")\n",
    "X_train_integrate_2 = train_all_integrate_2.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_train_integrate_2 = train_all_integrate_2[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "X_test_integrate_2 = test_all_integrate_2.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_test_integrate_2 = test_all_integrate_2[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "\n",
    "#overall cohort case 2 (N=1980,N=251)\n",
    "train_all_overall, test_all_overall = train_test_split(\n",
    "    data_all_overall,\n",
    "    test_size=0.20,\n",
    "    stratify=data_all_overall['OS_STATUS'],\n",
    "    random_state=123\n",
    ")\n",
    "X_train_overall = train_all_overall.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_train_overall = train_all_overall[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]\n",
    "X_test_overall = test_all_overall.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_test_overall = test_all_overall[['PATIENT_ID','OS_MONTHS', 'OS_STATUS']]"
   ],
   "id": "7ba3955a970c84ed",
   "outputs": [],
   "execution_count": 109
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-20T04:07:09.566158Z",
     "start_time": "2026-01-20T04:07:08.699570Z"
    }
   },
   "cell_type": "code",
   "source": [
    "## KNN imputation (genetic only + std first)\n",
    "#integrate cohort case 1 (N=1417,N=446)\n",
    "Xc_train_integrate_1 = X_train_integrate_1.iloc[:, :26]\n",
    "Xg_train_integrate_1 = X_train_integrate_1.iloc[:, 26:] #only impute on training\n",
    "Xc_test_integrate_1  = X_test_integrate_1.iloc[:, :26]\n",
    "Xg_test_integrate_1  = X_test_integrate_1.iloc[:, 26:]\n",
    "\n",
    "scaler = StandardScaler()\n",
    "Xg_train_integrate_1_scaled = scaler.fit_transform(Xg_train_integrate_1) #std first\n",
    "\n",
    "imputer = KNNImputer(n_neighbors=5,weights=\"distance\",metric=\"nan_euclidean\") #knn\n",
    "Xg_train_integrate_1_imp = imputer.fit_transform(Xg_train_integrate_1_scaled)\n",
    "\n",
    "Xg_train_integrate_1_imp = pd.DataFrame( #get columns' name back\n",
    "    Xg_train_integrate_1_imp,\n",
    "    columns=Xg_train_integrate_1.columns,\n",
    "    index=Xg_train_integrate_1.index)\n",
    "\n",
    "X_train_integrate_1_imp = pd.concat([Xc_train_integrate_1, Xg_train_integrate_1_imp], axis=1)\n",
    "\n",
    "Xg_test_integrate_1_scaled = scaler.transform(Xg_test_integrate_1) #Scale and std test using TRAIN statistics\n",
    "Xg_test_integrate_1_imp = imputer.transform(Xg_test_integrate_1_scaled)\n",
    "Xg_test_integrate_1_imp = pd.DataFrame(\n",
    "    Xg_test_integrate_1_imp,\n",
    "    columns=Xg_test_integrate_1.columns,\n",
    "    index=Xg_test_integrate_1.index\n",
    ")\n",
    "X_test_integrate_1_imp = pd.concat([Xc_test_integrate_1, Xg_test_integrate_1_imp], axis=1)\n",
    "\n",
    "##add ID back\n",
    "X_train_integrate_1_imp[\"PATIENT_ID\"] = train_all_integrate_1[\"PATIENT_ID\"].values\n",
    "X_test_integrate_1_imp[\"PATIENT_ID\"]  = test_all_integrate_1[\"PATIENT_ID\"].values\n",
    "def move_id_first(df, id_col=\"PATIENT_ID\"): #move ID to the first column\n",
    "    cols = [id_col] + [c for c in df.columns if c != id_col]\n",
    "    return df[cols]\n",
    "X_train_integrate_1_imp = move_id_first(X_train_integrate_1_imp)\n",
    "X_test_integrate_1_imp  = move_id_first(X_test_integrate_1_imp)"
   ],
   "id": "b58f07e86dcdd83e",
   "outputs": [],
   "execution_count": 110
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-20T04:07:22.067475Z",
     "start_time": "2026-01-20T04:07:21.082246Z"
    }
   },
   "cell_type": "code",
   "source": [
    "## KNN imputation (genetic only + std first)\n",
    "#integrate cohort case 2 (N=1417,N=545)\n",
    "Xc_train_integrate_2 = X_train_integrate_2.iloc[:, :26]\n",
    "Xg_train_integrate_2 = X_train_integrate_2.iloc[:, 26:]  #only impute on training\n",
    "Xc_test_integrate_2 = X_test_integrate_2.iloc[:, :26]\n",
    "Xg_test_integrate_2 = X_test_integrate_2.iloc[:, 26:]\n",
    "\n",
    "scaler = StandardScaler()\n",
    "Xg_train_integrate_2_scaled = scaler.fit_transform(Xg_train_integrate_2)  #std first\n",
    "\n",
    "imputer = KNNImputer(n_neighbors=5, weights=\"distance\", metric=\"nan_euclidean\")  #knn\n",
    "Xg_train_integrate_2_imp = imputer.fit_transform(Xg_train_integrate_2_scaled)\n",
    "\n",
    "Xg_train_integrate_2_imp = pd.DataFrame(  #get columns' name back\n",
    "    Xg_train_integrate_2_imp,\n",
    "    columns=Xg_train_integrate_2.columns,\n",
    "    index=Xg_train_integrate_2.index)\n",
    "\n",
    "X_train_integrate_2_imp = pd.concat([Xc_train_integrate_2, Xg_train_integrate_2_imp], axis=1)\n",
    "\n",
    "Xg_test_integrate_2_scaled = scaler.transform(Xg_test_integrate_2)  #Scale and std test using TRAIN statistics\n",
    "Xg_test_integrate_2_imp = imputer.transform(Xg_test_integrate_2_scaled)\n",
    "Xg_test_integrate_2_imp = pd.DataFrame(\n",
    "    Xg_test_integrate_2_imp,\n",
    "    columns=Xg_test_integrate_2.columns,\n",
    "    index=Xg_test_integrate_2.index\n",
    ")\n",
    "X_test_integrate_2_imp = pd.concat([Xc_test_integrate_2, Xg_test_integrate_2_imp], axis=1)\n",
    "\n",
    "##add ID back\n",
    "X_train_integrate_2_imp[\"PATIENT_ID\"] = train_all_integrate_2[\"PATIENT_ID\"].values\n",
    "X_test_integrate_2_imp[\"PATIENT_ID\"]  = test_all_integrate_2[\"PATIENT_ID\"].values\n",
    "X_train_integrate_2_imp = move_id_first(X_train_integrate_2_imp)\n",
    "X_test_integrate_2_imp  = move_id_first(X_test_integrate_2_imp)"
   ],
   "id": "ea37d6106924379c",
   "outputs": [],
   "execution_count": 111
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-20T04:07:30.476038Z",
     "start_time": "2026-01-20T04:07:30.440034Z"
    }
   },
   "cell_type": "code",
   "source": [
    "## KNN imputation (genetic only + std first)\n",
    "#overall cohort (N=1980,N=248)\n",
    "Xc_train_overall = X_train_overall.iloc[:, :26]\n",
    "Xg_train_overall = X_train_overall.iloc[:, 26:]  #only impute on training\n",
    "Xc_test_overall = X_test_overall.iloc[:, :26]\n",
    "Xg_test_overall = X_test_overall.iloc[:, 26:]\n",
    "\n",
    "scaler = StandardScaler()\n",
    "Xg_train_overall_scaled = scaler.fit_transform(Xg_train_overall)  #std first\n",
    "\n",
    "imputer = KNNImputer(n_neighbors=5, weights=\"distance\", metric=\"nan_euclidean\")  #knn\n",
    "Xg_train_overall_imp = imputer.fit_transform(Xg_train_overall_scaled)\n",
    "\n",
    "Xg_train_overall_imp = pd.DataFrame(  #get columns' name back\n",
    "    Xg_train_overall_imp,\n",
    "    columns=Xg_train_overall.columns,\n",
    "    index=Xg_train_overall.index)\n",
    "\n",
    "X_train_overall_imp = pd.concat([Xc_train_overall, Xg_train_overall_imp], axis=1)\n",
    "\n",
    "Xg_test_overall_scaled = scaler.transform(Xg_test_overall)  #Scale and std test using TRAIN statistics\n",
    "Xg_test_overall_imp = imputer.transform(Xg_test_overall_scaled)\n",
    "Xg_test_overall_imp = pd.DataFrame(\n",
    "    Xg_test_overall_imp,\n",
    "    columns=Xg_test_overall.columns,\n",
    "    index=Xg_test_overall.index\n",
    ")\n",
    "X_test_overall_imp = pd.concat([Xc_test_overall, Xg_test_overall_imp], axis=1)\n",
    "\n",
    "##add ID back\n",
    "X_train_overall_imp[\"PATIENT_ID\"] = train_all_overall[\"PATIENT_ID\"].values\n",
    "X_test_overall_imp[\"PATIENT_ID\"]  = test_all_overall[\"PATIENT_ID\"].values\n",
    "X_train_overall_imp = move_id_first(X_train_overall_imp)\n",
    "X_test_overall_imp  = move_id_first(X_test_overall_imp)"
   ],
   "id": "1422fb06374facbc",
   "outputs": [],
   "execution_count": 112
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-20T04:08:29.144342Z",
     "start_time": "2026-01-20T04:08:27.344311Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##output all dataset ready for prediction models\n",
    "X_train_integrate_1_imp.to_csv(\"/Users/mac1/Desktop/X_train_integrate_1_imp.csv\", index=False)\n",
    "X_test_integrate_1_imp.to_csv(\"/Users/mac1/Desktop/X_test_integrate_1_imp.csv\", index=False)\n",
    "y_train_integrate_1.to_csv(\"/Users/mac1/Desktop/y_train_integrate_1.csv\", index=False)\n",
    "y_test_integrate_1.to_csv(\"/Users/mac1/Desktop/y_test_integrate_1.csv\", index=False)\n",
    "\n",
    "X_train_integrate_2_imp.to_csv(\"/Users/mac1/Desktop/X_train_integrate_2_imp.csv\", index=False)\n",
    "X_test_integrate_2_imp.to_csv(\"/Users/mac1/Desktop/X_test_integrate_2_imp.csv\", index=False)\n",
    "y_train_integrate_2.to_csv(\"/Users/mac1/Desktop/y_train_integrate_2.csv\", index=False)\n",
    "y_test_integrate_2.to_csv(\"/Users/mac1/Desktop/y_test_integrate_2.csv\", index=False)\n",
    "\n",
    "X_train_overall_imp.to_csv(\"/Users/mac1/Desktop/X_train_overall_imp.csv\", index=False)\n",
    "X_test_overall_imp.to_csv(\"/Users/mac1/Desktop/X_test_overall_imp.csv\", index=False)\n",
    "y_train_overall.to_csv(\"/Users/mac1/Desktop/y_train_overall.csv\", index=False)\n",
    "y_test_overall.to_csv(\"/Users/mac1/Desktop/y_test_overall.csv\", index=False)"
   ],
   "id": "8dcf44f40d22a512",
   "outputs": [],
   "execution_count": 121
  }
 ],
 "metadata": {
  "kernelspec": {
   "name": "python3",
   "language": "python",
   "display_name": "Python 3 (ipykernel)"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
