{
 "cells": [
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-19T22:06:00.540620Z",
     "start_time": "2026-01-19T22:06:00.432032Z"
    }
   },
   "cell_type": "code",
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "from sklearn.impute import KNNImputer\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.model_selection import train_test_split"
   ],
   "id": "a05b393c220d6ffd",
   "outputs": [],
   "execution_count": 35
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-19T21:09:20.124289Z",
     "start_time": "2026-01-19T21:09:01.760191Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##upload the original mRNA and methylation\n",
    "#interated mRNA and methylation (n=1417) without any operation and exclusion of features\n",
    "methy_sub = pd.read_csv('/Users/mac1/Desktop/methylation_sub.csv')\n",
    "mRNA_sub = pd.read_csv('/Users/mac1/Desktop/mRNA_sub.csv')\n",
    "#overall mRNA (n=1980) without any operation and exclusion of features\n",
    "mRNA_genetic = pd.read_csv('/Users/mac1/Desktop/mRNA_genetic.csv')"
   ],
   "id": "92d7f60a08c32b03",
   "outputs": [],
   "execution_count": 4
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-19T21:09:21.890528Z",
     "start_time": "2026-01-19T21:09:21.877476Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##extract ID row index for cohort\n",
    "keep_ids_overall = mRNA_genetic['PATIENT_ID'] #overall\n",
    "keep_ids_integrate = methy_sub['PATIENT_ID'] #integrate\n",
    "##extract feature sets for methylation or mRNA\n",
    "methy_features = [c for c in methy_sub.columns if c not in [\"PATIENT_ID\"]]\n",
    "mRNA_features = [c for c in mRNA_sub.columns if c not in [\"PATIENT_ID\"]]"
   ],
   "id": "e2a781aa876b456",
   "outputs": [],
   "execution_count": 5
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-19T21:09:23.571252Z",
     "start_time": "2026-01-19T21:09:23.518501Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##clinical features （overall cohort)\n",
    "clinical_impute_overall = pd.read_csv(\"/Users/mac1/Desktop/DATAFEST/clinical_imputed_1.csv\")\n",
    "clinical_impute_overall.shape"
   ],
   "id": "844d0bc2bf2b920f",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1980, 29)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 6
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-19T21:09:25.286912Z",
     "start_time": "2026-01-19T21:09:25.241425Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##clinical feature (integrated cohort)\n",
    "clinical_impute_integrate = clinical_impute_overall[clinical_impute_overall['PATIENT_ID'].isin(keep_ids_integrate)].copy()\n",
    "clinical_impute_integrate.shape"
   ],
   "id": "97a95800c68c3c69",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1417, 29)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 7
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-19T22:30:44.377648Z",
     "start_time": "2026-01-19T22:30:44.200246Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##gwas feature (overall cohort)\n",
    "gwas=pd.read_csv('/Users/mac1/Desktop/DATAFEST/genomics_selected_gwas_ensembl.csv',index_col=0)\n",
    "gwas_overall = gwas[gwas['PATIENT_ID'].isin(keep_ids_overall)].copy()"
   ],
   "id": "ecea1239e13f7474",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "     PATIENT_ID  ACVRL1_methy  AHNAK_methy  AKT1_methy  ALK_methy  APC_methy  \\\n",
       "1       MB-0000           NaN          NaN         NaN        NaN        NaN   \n",
       "2       MB-0002           NaN          NaN         NaN        NaN        NaN   \n",
       "3       MB-0005           NaN          NaN         NaN        NaN        NaN   \n",
       "4       MB-0006      0.228235     0.039574    0.012346   0.032258   0.190789   \n",
       "5       MB-0008           NaN          NaN         NaN        NaN        NaN   \n",
       "...         ...           ...          ...         ...        ...        ...   \n",
       "1981    MB-7295           NaN          NaN         NaN        NaN        NaN   \n",
       "1982    MB-7296      0.491018     0.065934         NaN   0.000000   0.030303   \n",
       "1983    MB-7297           NaN          NaN         NaN        NaN        NaN   \n",
       "1984    MB-7298           NaN          NaN         NaN        NaN        NaN   \n",
       "1985    MB-7299           NaN          NaN         NaN        NaN        NaN   \n",
       "\n",
       "      ARID1A_methy  ARID1B_methy  ARID2_methy  ASXL1_methy  ...  EMID1_mrna  \\\n",
       "1              NaN           NaN          NaN          NaN  ...    5.825269   \n",
       "2              NaN           NaN          NaN          NaN  ...    5.698300   \n",
       "3              NaN           NaN          NaN          NaN  ...    5.713216   \n",
       "4         0.000000      0.000000     0.000000          0.0  ...    5.501439   \n",
       "5              NaN           NaN          NaN          NaN  ...    6.018440   \n",
       "...            ...           ...          ...          ...  ...         ...   \n",
       "1981           NaN           NaN          NaN          NaN  ...    5.598353   \n",
       "1982      0.007895      0.007519     0.003571          0.0  ...    6.195056   \n",
       "1983           NaN           NaN          NaN          NaN  ...    5.408007   \n",
       "1984           NaN           NaN          NaN          NaN  ...    5.629645   \n",
       "1985           NaN           NaN          NaN          NaN  ...    5.804815   \n",
       "\n",
       "      TTC28_mrna  ATAD5_mrna  PAX9_mrna  MLLT10_mrna  PTHLH_mrna  SKIDA1_mrna  \\\n",
       "1       5.798647    5.762455   5.469058     7.263043    6.013555     5.608817   \n",
       "2       5.609069    5.807519   6.411290     8.121593    5.764321     6.050104   \n",
       "3       5.485918    5.704256   5.817795     6.877284    9.808890     5.321103   \n",
       "4       5.356186    5.670296   6.377648     7.235799    6.089555     5.639904   \n",
       "5       5.738390    5.705560   5.596711     7.947678    6.121529     5.328361   \n",
       "...          ...         ...        ...          ...         ...          ...   \n",
       "1981    5.185025    5.718490   6.096267     7.250477    6.841198     5.444054   \n",
       "1982    5.512736    5.968755   5.843940     7.143147    5.865243     5.447092   \n",
       "1983    5.631185    6.096751   5.650109     7.113933    5.667898     5.381733   \n",
       "1984    5.543150    5.641894   5.715491     7.449389    6.630646     5.391292   \n",
       "1985    5.733710    5.717691   5.260670     7.913005    7.922645     5.583457   \n",
       "\n",
       "      MRPS30_mrna  APOBEC3A_mrna  SEC14L4_mrna  \n",
       "1        8.535214       5.336678      5.868042  \n",
       "2        9.628202       5.377759      5.747465  \n",
       "3        8.915721       5.338215      5.554798  \n",
       "4        9.058717       5.515105      5.881517  \n",
       "5        9.860345       5.375316      5.278561  \n",
       "...           ...            ...           ...  \n",
       "1981     9.119662       5.289611      5.461226  \n",
       "1982     9.195065       5.441914      5.688695  \n",
       "1983    10.738350       5.311344      5.461644  \n",
       "1984    12.146450       5.371765      5.449533  \n",
       "1985     8.909626       5.235019      5.607719  \n",
       "\n",
       "[1980 rows x 224 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PATIENT_ID</th>\n",
       "      <th>ACVRL1_methy</th>\n",
       "      <th>AHNAK_methy</th>\n",
       "      <th>AKT1_methy</th>\n",
       "      <th>ALK_methy</th>\n",
       "      <th>APC_methy</th>\n",
       "      <th>ARID1A_methy</th>\n",
       "      <th>ARID1B_methy</th>\n",
       "      <th>ARID2_methy</th>\n",
       "      <th>ASXL1_methy</th>\n",
       "      <th>...</th>\n",
       "      <th>EMID1_mrna</th>\n",
       "      <th>TTC28_mrna</th>\n",
       "      <th>ATAD5_mrna</th>\n",
       "      <th>PAX9_mrna</th>\n",
       "      <th>MLLT10_mrna</th>\n",
       "      <th>PTHLH_mrna</th>\n",
       "      <th>SKIDA1_mrna</th>\n",
       "      <th>MRPS30_mrna</th>\n",
       "      <th>APOBEC3A_mrna</th>\n",
       "      <th>SEC14L4_mrna</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MB-0000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>5.825269</td>\n",
       "      <td>5.798647</td>\n",
       "      <td>5.762455</td>\n",
       "      <td>5.469058</td>\n",
       "      <td>7.263043</td>\n",
       "      <td>6.013555</td>\n",
       "      <td>5.608817</td>\n",
       "      <td>8.535214</td>\n",
       "      <td>5.336678</td>\n",
       "      <td>5.868042</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MB-0002</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>5.698300</td>\n",
       "      <td>5.609069</td>\n",
       "      <td>5.807519</td>\n",
       "      <td>6.411290</td>\n",
       "      <td>8.121593</td>\n",
       "      <td>5.764321</td>\n",
       "      <td>6.050104</td>\n",
       "      <td>9.628202</td>\n",
       "      <td>5.377759</td>\n",
       "      <td>5.747465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MB-0005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>5.713216</td>\n",
       "      <td>5.485918</td>\n",
       "      <td>5.704256</td>\n",
       "      <td>5.817795</td>\n",
       "      <td>6.877284</td>\n",
       "      <td>9.808890</td>\n",
       "      <td>5.321103</td>\n",
       "      <td>8.915721</td>\n",
       "      <td>5.338215</td>\n",
       "      <td>5.554798</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MB-0006</td>\n",
       "      <td>0.228235</td>\n",
       "      <td>0.039574</td>\n",
       "      <td>0.012346</td>\n",
       "      <td>0.032258</td>\n",
       "      <td>0.190789</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>5.501439</td>\n",
       "      <td>5.356186</td>\n",
       "      <td>5.670296</td>\n",
       "      <td>6.377648</td>\n",
       "      <td>7.235799</td>\n",
       "      <td>6.089555</td>\n",
       "      <td>5.639904</td>\n",
       "      <td>9.058717</td>\n",
       "      <td>5.515105</td>\n",
       "      <td>5.881517</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MB-0008</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>6.018440</td>\n",
       "      <td>5.738390</td>\n",
       "      <td>5.705560</td>\n",
       "      <td>5.596711</td>\n",
       "      <td>7.947678</td>\n",
       "      <td>6.121529</td>\n",
       "      <td>5.328361</td>\n",
       "      <td>9.860345</td>\n",
       "      <td>5.375316</td>\n",
       "      <td>5.278561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1981</th>\n",
       "      <td>MB-7295</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>5.598353</td>\n",
       "      <td>5.185025</td>\n",
       "      <td>5.718490</td>\n",
       "      <td>6.096267</td>\n",
       "      <td>7.250477</td>\n",
       "      <td>6.841198</td>\n",
       "      <td>5.444054</td>\n",
       "      <td>9.119662</td>\n",
       "      <td>5.289611</td>\n",
       "      <td>5.461226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1982</th>\n",
       "      <td>MB-7296</td>\n",
       "      <td>0.491018</td>\n",
       "      <td>0.065934</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.030303</td>\n",
       "      <td>0.007895</td>\n",
       "      <td>0.007519</td>\n",
       "      <td>0.003571</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>6.195056</td>\n",
       "      <td>5.512736</td>\n",
       "      <td>5.968755</td>\n",
       "      <td>5.843940</td>\n",
       "      <td>7.143147</td>\n",
       "      <td>5.865243</td>\n",
       "      <td>5.447092</td>\n",
       "      <td>9.195065</td>\n",
       "      <td>5.441914</td>\n",
       "      <td>5.688695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1983</th>\n",
       "      <td>MB-7297</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>5.408007</td>\n",
       "      <td>5.631185</td>\n",
       "      <td>6.096751</td>\n",
       "      <td>5.650109</td>\n",
       "      <td>7.113933</td>\n",
       "      <td>5.667898</td>\n",
       "      <td>5.381733</td>\n",
       "      <td>10.738350</td>\n",
       "      <td>5.311344</td>\n",
       "      <td>5.461644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1984</th>\n",
       "      <td>MB-7298</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>5.629645</td>\n",
       "      <td>5.543150</td>\n",
       "      <td>5.641894</td>\n",
       "      <td>5.715491</td>\n",
       "      <td>7.449389</td>\n",
       "      <td>6.630646</td>\n",
       "      <td>5.391292</td>\n",
       "      <td>12.146450</td>\n",
       "      <td>5.371765</td>\n",
       "      <td>5.449533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1985</th>\n",
       "      <td>MB-7299</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>5.804815</td>\n",
       "      <td>5.733710</td>\n",
       "      <td>5.717691</td>\n",
       "      <td>5.260670</td>\n",
       "      <td>7.913005</td>\n",
       "      <td>7.922645</td>\n",
       "      <td>5.583457</td>\n",
       "      <td>8.909626</td>\n",
       "      <td>5.235019</td>\n",
       "      <td>5.607719</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1980 rows × 224 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 47
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-19T22:30:57.395767Z",
     "start_time": "2026-01-19T22:30:57.346423Z"
    }
   },
   "cell_type": "code",
   "source": [
    "#99 methylation in the combined overall gwas feature set, may not be used\n",
    "gwas_overall_methy = gwas_overall[[\"PATIENT_ID\"] + list(gwas_overall.columns.intersection(methy_features))]\n",
    "gwas_overall_methy.shape"
   ],
   "id": "2b6d2dbd96e58c12",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1980, 99)"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 48
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-19T22:30:58.767110Z",
     "start_time": "2026-01-19T22:30:58.728751Z"
    }
   },
   "cell_type": "code",
   "source": [
    "#125 mRNA in the combined overall gwas feature set\n",
    "gwas_overall_mRNA = gwas_overall[[\"PATIENT_ID\"] + list(gwas_overall.columns.intersection(mRNA_features))]\n",
    "gwas_overall_mRNA.shape"
   ],
   "id": "15dc1b0fc468680",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1980, 125)"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 49
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-19T22:31:00.489417Z",
     "start_time": "2026-01-19T22:31:00.436096Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##gwas feature (integrated cohort)\n",
    "gwas_integrate = gwas[gwas['PATIENT_ID'].isin(keep_ids_integrate)].copy()\n",
    "gwas_integrate.shape"
   ],
   "id": "b5efe432890356c2",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1417, 224)"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 50
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-19T22:31:08.143608Z",
     "start_time": "2026-01-19T22:31:08.099350Z"
    }
   },
   "cell_type": "code",
   "source": [
    "gwas_integrate_methy = gwas_integrate[[\"PATIENT_ID\"] + list(gwas_integrate.columns.intersection(methy_features))]\n",
    "gwas_integrate_methy.shape"
   ],
   "id": "2c52edad73e5187a",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1417, 99)"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 51
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-19T22:31:09.447255Z",
     "start_time": "2026-01-19T22:31:09.410418Z"
    }
   },
   "cell_type": "code",
   "source": [
    "gwas_integrate_mRNA = gwas_integrate[[\"PATIENT_ID\"] + list(gwas_integrate.columns.intersection(mRNA_features))]\n",
    "gwas_integrate_mRNA.shape"
   ],
   "id": "8e2045c247503d4a",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1417, 125)"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 52
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-19T22:31:11.401928Z",
     "start_time": "2026-01-19T22:31:11.359906Z"
    }
   },
   "cell_type": "code",
   "source": [
    "nan_counts_mRNA = gwas_overall_mRNA.isna().sum()\n",
    "nan_counts_meth = gwas_integrate_methy.isna().sum()\n",
    "missing_pct_mRNA= gwas_overall_mRNA.isna().mean() * 100\n",
    "missing_pct_meth= gwas_integrate_methy.isna().mean() * 100\n",
    "print(\"Columns with NaN in mRNA in GWAS overall:\", (nan_counts_mRNA > 0).sum())\n",
    "print(\"Columns with NaN in methylation in GWAS integrated:\", (nan_counts_meth > 0).sum())\n",
    "print(\"Percentage of missing in mRNA in GWAS overall\",missing_pct_mRNA.describe())\n",
    "print(\"Percentage of missing in methylation in GWAS integrated\",missing_pct_meth.describe())\n",
    "##the missingness percentage is acceptable, let's move on for KNN imputation"
   ],
   "id": "65aef04dcf93a1c",
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Columns with NaN in mRNA in GWAS overall: 0\n",
      "Columns with NaN in methylation in GWAS integrated: 75\n",
      "Percentage of missing in mRNA in GWAS overall count    125.0\n",
      "mean       0.0\n",
      "std        0.0\n",
      "min        0.0\n",
      "25%        0.0\n",
      "50%        0.0\n",
      "75%        0.0\n",
      "max        0.0\n",
      "dtype: float64\n",
      "Percentage of missing in methylation in GWAS integrated count    99.000000\n",
      "mean      3.758830\n",
      "std       6.782153\n",
      "min       0.000000\n",
      "25%       0.070572\n",
      "50%       0.423430\n",
      "75%       3.705011\n",
      "max      28.863797\n",
      "dtype: float64\n"
     ]
    }
   ],
   "execution_count": 53
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-19T22:31:15.003028Z",
     "start_time": "2026-01-19T22:31:14.982634Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##data-driven feature set\n",
    "#integrated cohort case 1 (n=1417, N=200)\n",
    "var_data_driven_integrate_1 = pd.read_csv('/Users/mac1/Desktop/DATAFEST/strategy_3_Importance_Cohort1_Case1_SeparateCox.csv')\n",
    "top200_features_1 = var_data_driven_integrate_1[\"feature_name\"].head(200).tolist()\n",
    "##integrated cohort case 2 (n=1417, N=300)\n",
    "var_data_driven_integrate_2 = pd.read_csv('/Users/mac1/Desktop/DATAFEST/strategy_3_Importance_Cohort1_Case2_CombinedCox.csv')\n",
    "top300_features_2 = var_data_driven_integrate_2[\"feature_name\"].head(300).tolist()\n",
    "#overall cohort (n=1980, N mRNA=100)\n",
    "var_data_driven_overall = pd.read_csv('/Users/mac1/Desktop/DATAFEST/strategy_3_Importance_Cohort1_Case2_CombinedCox.csv')\n",
    "top100_features_3 = var_data_driven_overall[\"feature_name\"].head(100).tolist()"
   ],
   "id": "b790d68d926e2681",
   "outputs": [],
   "execution_count": 54
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-19T22:31:17.920568Z",
     "start_time": "2026-01-19T22:31:16.914514Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##data-driven dataset (integrate cohort case 1)\n",
    "genetic_sub = methy_sub.merge(mRNA_sub, on=\"PATIENT_ID\", how=\"inner\")\n",
    "data_driven_integrate_1 = genetic_sub[[\"PATIENT_ID\"] + list(genetic_sub.columns.intersection(top200_features_1))]\n",
    "##data-driven dataset (integrate cohort case 2)\n",
    "data_driven_integrate_2 = genetic_sub[[\"PATIENT_ID\"] + list(genetic_sub.columns.intersection(top300_features_2))]\n",
    "##data-driven dataset (overall cohort)\n",
    "data_driven_overall = genetic_sub[[\"PATIENT_ID\"] + list(genetic_sub.columns.intersection(top100_features_3))]"
   ],
   "id": "83bc3c93dcef9e61",
   "outputs": [],
   "execution_count": 55
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-19T22:31:18.917062Z",
     "start_time": "2026-01-19T22:31:18.862645Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##Combine clinical+gwas+data-driven\n",
    "#integrate cohort case 1 (N=1417,N=452)\n",
    "data_all_integrate_1 = clinical_impute_integrate.merge(gwas_integrate, on=\"PATIENT_ID\", how=\"inner\") #clinical+gwas\n",
    "data_all_integrate_1 = data_all_integrate_1.merge(data_driven_integrate_1, on=\"PATIENT_ID\", how=\"inner\") # +data-driven\n",
    "#integrate cohort case 2 (n=1417, N=552)\n",
    "data_all_integrate_2 = clinical_impute_integrate.merge(gwas_integrate, on=\"PATIENT_ID\", how=\"inner\") #clinical+gwas\n",
    "data_all_integrate_2 = data_all_integrate_2.merge(data_driven_integrate_2, on=\"PATIENT_ID\", how=\"inner\") # +data-driven\n",
    "#overall cohort (n=1417, N=253)\n",
    "data_all_overall = clinical_impute_overall.merge(gwas_overall_mRNA, on=\"PATIENT_ID\", how=\"inner\") #clinical+gwas (mrna)\n",
    "data_all_overall = data_all_overall.merge(data_driven_overall, on=\"PATIENT_ID\", how=\"inner\") # +data-driven"
   ],
   "id": "efbf4a1862dc5065",
   "outputs": [],
   "execution_count": 56
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-19T22:31:46.965831Z",
     "start_time": "2026-01-19T22:31:46.819707Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##split into training and testing (70% vs 30%)\n",
    "#integrate cohort case 1 (N=1417,N=452)\n",
    "train_all_integrate_1, test_all_integrate_1 = train_test_split(\n",
    "    data_all_integrate_1,\n",
    "    test_size=0.30,\n",
    "    stratify=data_all_integrate_1['OS_STATUS'],\n",
    "    random_state=123\n",
    ")\n",
    "X_train_integrate_1 = train_all_integrate_1.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_train_integrate_1 = train_all_integrate_1[['OS_MONTHS', 'OS_STATUS']]\n",
    "X_test_integrate_1 = test_all_integrate_1.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_test_integrate_1 = test_all_integrate_1[['OS_MONTHS', 'OS_STATUS']]\n",
    "\n",
    "#integrate cohort case 2 (N=1417,N=552)\n",
    "train_all_integrate_2, test_all_integrate_2 = train_test_split(\n",
    "    data_all_integrate_2,\n",
    "    test_size=0.30,\n",
    "    stratify=data_all_integrate_2['OS_STATUS'],\n",
    "    random_state=123\n",
    ")\n",
    "X_train_integrate_2 = train_all_integrate_2.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_train_integrate_2 = train_all_integrate_2[['OS_MONTHS', 'OS_STATUS']]\n",
    "X_test_integrate_2 = test_all_integrate_2.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_test_integrate_2 = test_all_integrate_2[['OS_MONTHS', 'OS_STATUS']]\n",
    "\n",
    "#overall cohort case 2 (N=1980,N=253)\n",
    "train_all_overall, test_all_overall = train_test_split(\n",
    "    data_all_overall,\n",
    "    test_size=0.30,\n",
    "    stratify=data_all_overall['OS_STATUS'],\n",
    "    random_state=123\n",
    ")\n",
    "X_train_overall = train_all_overall.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_train_overall = train_all_overall[['OS_MONTHS', 'OS_STATUS']]\n",
    "X_test_overall = test_all_overall.drop(columns=['PATIENT_ID', 'OS_MONTHS', 'OS_STATUS'])\n",
    "y_test_overall = test_all_overall[['OS_MONTHS', 'OS_STATUS']]"
   ],
   "id": "7ba3955a970c84ed",
   "outputs": [],
   "execution_count": 58
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-19T22:52:36.462172Z",
     "start_time": "2026-01-19T22:52:35.783557Z"
    }
   },
   "cell_type": "code",
   "source": [
    "## KNN imputation (genetic only + std first)\n",
    "#integrate cohort case 1 (N=1417,N=452)\n",
    "Xc_train_integrate_1 = X_train_integrate_1.iloc[:, :26]\n",
    "Xg_train_integrate_1 = X_train_integrate_1.iloc[:, 26:] #only impute on training\n",
    "Xc_test_integrate_1  = X_test_integrate_1.iloc[:, :26]\n",
    "Xg_test_integrate_1  = X_test_integrate_1.iloc[:, 26:]\n",
    "\n",
    "scaler = StandardScaler()\n",
    "Xg_train_integrate_1_scaled = scaler.fit_transform(Xg_train_integrate_1) #std first\n",
    "\n",
    "imputer = KNNImputer(n_neighbors=5,weights=\"distance\",metric=\"nan_euclidean\") #knn\n",
    "Xg_train_integrate_1_imp = imputer.fit_transform(Xg_train_integrate_1_scaled)\n",
    "\n",
    "Xg_train_integrate_1_imp = pd.DataFrame( #get columns' name back\n",
    "    Xg_train_integrate_1_imp,\n",
    "    columns=Xg_train_integrate_1.columns,\n",
    "    index=Xg_train_integrate_1.index)\n",
    "\n",
    "X_train_integrate_1_imp = pd.concat([Xc_train_integrate_1, Xg_train_integrate_1_imp], axis=1)\n",
    "\n",
    "Xg_test_integrate_1_scaled = scaler.transform(Xg_test_integrate_1) #Scale and std test using TRAIN statistics\n",
    "Xg_test_integrate_1_imp = imputer.transform(Xg_test_integrate_1_scaled)\n",
    "Xg_test_integrate_1_imp = pd.DataFrame(\n",
    "    Xg_test_integrate_1_imp,\n",
    "    columns=Xg_test_integrate_1.columns,\n",
    "    index=Xg_test_integrate_1.index\n",
    ")\n",
    "X_test_integrate_1_imp = pd.concat([Xc_test_integrate_1, Xg_test_integrate_1_imp], axis=1)"
   ],
   "id": "b58f07e86dcdd83e",
   "outputs": [],
   "execution_count": 69
  },
  {
   "metadata": {},
   "cell_type": "code",
   "outputs": [],
   "execution_count": null,
   "source": [
    "## KNN imputation (genetic only + std first)\n",
    "#integrate cohort case 2 (N=1417,N=552)\n",
    "Xc_train_integrate_2 = X_train_integrate_2.iloc[:, :26]\n",
    "Xg_train_integrate_2 = X_train_integrate_2.iloc[:, 26:]  #only impute on training\n",
    "Xc_test_integrate_2 = X_test_integrate_2.iloc[:, :26]\n",
    "Xg_test_integrate_2 = X_test_integrate_2.iloc[:, 26:]\n",
    "\n",
    "scaler = StandardScaler()\n",
    "Xg_train_integrate_2_scaled = scaler.fit_transform(Xg_train_integrate_2)  #std first\n",
    "\n",
    "imputer = KNNImputer(n_neighbors=5, weights=\"distance\", metric=\"nan_euclidean\")  #knn\n",
    "Xg_train_integrate_2_imp = imputer.fit_transform(Xg_train_integrate_2_scaled)\n",
    "\n",
    "Xg_train_integrate_2_imp = pd.DataFrame(  #get columns' name back\n",
    "    Xg_train_integrate_2_imp,\n",
    "    columns=Xg_train_integrate_2.columns,\n",
    "    index=Xg_train_integrate_2.index)\n",
    "\n",
    "X_train_integrate_2_imp = pd.concat([Xc_train_integrate_2, Xg_train_integrate_2_imp], axis=1)\n",
    "\n",
    "Xg_test_integrate_2_scaled = scaler.transform(Xg_test_integrate_2)  #Scale and std test using TRAIN statistics\n",
    "Xg_test_integrate_2_imp = imputer.transform(Xg_test_integrate_2_scaled)\n",
    "Xg_test_integrate_2_imp = pd.DataFrame(\n",
    "    Xg_test_integrate_2_imp,\n",
    "    columns=Xg_test_integrate_2.columns,\n",
    "    index=Xg_test_integrate_2.index\n",
    ")\n",
    "X_test_integrate_2_imp = pd.concat([Xc_test_integrate_2, Xg_test_integrate_2_imp], axis=1)"
   ],
   "id": "ea37d6106924379c"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-19T22:57:00.574545Z",
     "start_time": "2026-01-19T22:57:00.419897Z"
    }
   },
   "cell_type": "code",
   "outputs": [],
   "execution_count": 74,
   "source": [
    "## KNN imputation (genetic only + std first)\n",
    "#overall cohort (N=1980,N=252)\n",
    "Xc_train_overall = X_train_overall.iloc[:, :26]\n",
    "Xg_train_overall = X_train_overall.iloc[:, 26:]  #only impute on training\n",
    "Xc_test_overall = X_test_overall.iloc[:, :26]\n",
    "Xg_test_overall = X_test_overall.iloc[:, 26:]\n",
    "\n",
    "scaler = StandardScaler()\n",
    "Xg_train_overall_scaled = scaler.fit_transform(Xg_train_overall)  #std first\n",
    "\n",
    "imputer = KNNImputer(n_neighbors=5, weights=\"distance\", metric=\"nan_euclidean\")  #knn\n",
    "Xg_train_overall_imp = imputer.fit_transform(Xg_train_overall_scaled)\n",
    "\n",
    "Xg_train_overall_imp = pd.DataFrame(  #get columns' name back\n",
    "    Xg_train_overall_imp,\n",
    "    columns=Xg_train_overall.columns,\n",
    "    index=Xg_train_overall.index)\n",
    "\n",
    "X_train_overall_imp = pd.concat([Xc_train_overall, Xg_train_overall_imp], axis=1)\n",
    "\n",
    "Xg_test_overall_scaled = scaler.transform(Xg_test_overall)  #Scale and std test using TRAIN statistics\n",
    "Xg_test_overall_imp = imputer.transform(Xg_test_overall_scaled)\n",
    "Xg_test_overall_imp = pd.DataFrame(\n",
    "    Xg_test_overall_imp,\n",
    "    columns=Xg_test_overall.columns,\n",
    "    index=Xg_test_overall.index\n",
    ")\n",
    "X_test_overall_imp = pd.concat([Xc_test_overall, Xg_test_overall_imp], axis=1)\n"
   ],
   "id": "1422fb06374facbc"
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-19T23:02:55.652777Z",
     "start_time": "2026-01-19T23:02:55.586010Z"
    }
   },
   "cell_type": "code",
   "source": "X_test_overall_imp",
   "id": "3d1af7b4718f1ea9",
   "outputs": [
    {
     "data": {
      "text/plain": [
       "      RFS_MONTHS      RFS_STATUS          VITAL_STATUS  \\\n",
       "342   121.966667  0:Not Recurred                Living   \n",
       "325   149.866667  0:Not Recurred                Living   \n",
       "946    45.533333      1:Recurred       Died of Disease   \n",
       "1011  187.033333  0:Not Recurred  Died of Other Causes   \n",
       "1168   21.066667      1:Recurred       Died of Disease   \n",
       "...          ...             ...                   ...   \n",
       "1385  239.166667  0:Not Recurred                Living   \n",
       "965    62.633333      1:Recurred       Died of Disease   \n",
       "282   130.900000  0:Not Recurred                Living   \n",
       "607   199.966667  0:Not Recurred                Living   \n",
       "474   101.266667  0:Not Recurred  Died of Other Causes   \n",
       "\n",
       "      LYMPH_NODES_EXAMINED_POSITIVE    NPI  AGE_AT_DIAGNOSIS  TUMOR_SIZE  \\\n",
       "342                               1  4.042             79.52        21.0   \n",
       "325                               0  2.040             61.79        20.0   \n",
       "946                               0  3.032             43.63        16.0   \n",
       "1011                              1  4.030             74.46        15.0   \n",
       "1168                              3  4.090             60.41        45.0   \n",
       "...                             ...    ...               ...         ...   \n",
       "1385                              6  5.070             58.92        35.0   \n",
       "965                               1  4.040             69.72        20.0   \n",
       "282                               4  4.052             60.00        26.0   \n",
       "607                               0  2.030             49.37        15.0   \n",
       "474                               1  5.034             70.19        17.0   \n",
       "\n",
       "      TMB_NONSYNONYMOUS CELLULARITY CHEMOTHERAPY  ...  FXYD4_mrna  \\\n",
       "342            9.152624        High           NO  ...    0.992813   \n",
       "325           11.767659    Moderate           NO  ...    0.009157   \n",
       "946           10.460142    Moderate           NO  ...   -0.491750   \n",
       "1011           6.537589        High           NO  ...   -0.023318   \n",
       "1168           3.922553        High           NO  ...    0.113393   \n",
       "...                 ...         ...          ...  ...         ...   \n",
       "1385           9.152624         Low           NO  ...    0.118282   \n",
       "965           10.460142        High           NO  ...   -0.252079   \n",
       "282            2.615035        High          YES  ...    0.101426   \n",
       "607            5.230071    Moderate           NO  ...   -0.654629   \n",
       "474            5.230071        High           NO  ...   -0.737820   \n",
       "\n",
       "     TPD52L1_mrna ENC1_mrna DNAH11_mrna_y ALDOC_mrna GRB2_mrna GPR156_mrna  \\\n",
       "342      1.416946 -1.223732     -1.054836  -1.081067 -0.300710    0.919122   \n",
       "325      0.942511  0.392749      0.240805  -0.491881  0.271729   -0.483658   \n",
       "946      0.745315  0.274270      0.850794  -0.957869 -0.096607    1.658370   \n",
       "1011     0.906088 -0.305917     -0.715573  -0.390119 -0.838757    0.300666   \n",
       "1168     0.353704  0.089972      0.233243   0.291860  0.632116    0.926125   \n",
       "...           ...       ...           ...        ...       ...         ...   \n",
       "1385     0.365060  0.822153      0.158290   1.741177 -1.079096   -0.040037   \n",
       "965      0.657425 -0.533618      0.059456  -0.672819  0.489529    0.075651   \n",
       "282      1.261980 -0.096220     -0.623950   1.074558 -0.049146   -0.609271   \n",
       "607     -0.214460  0.339306      0.374215   0.642533 -0.306414   -0.694937   \n",
       "474     -0.181638  0.534758      0.357674   2.884460 -0.691514   -0.037620   \n",
       "\n",
       "     VANGL1_mrna NKX2-2_mrna UMODL1_mrna  \n",
       "342     0.161028   -0.428187    0.224989  \n",
       "325     0.484974   -0.176472   -0.728284  \n",
       "946    -0.290612   -0.415236   -0.821329  \n",
       "1011   -1.049251   -0.455549   -0.755907  \n",
       "1168   -0.171070    2.668258   -0.582256  \n",
       "...          ...         ...         ...  \n",
       "1385   -0.530432    3.234289   -1.201279  \n",
       "965    -1.201659   -0.632220   -0.407730  \n",
       "282     0.196321   -0.558113    1.014398  \n",
       "607    -0.365147   -0.654322   -0.322543  \n",
       "474     0.060212   -0.632401   -0.277618  \n",
       "\n",
       "[426 rows x 250 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>RFS_MONTHS</th>\n",
       "      <th>RFS_STATUS</th>\n",
       "      <th>VITAL_STATUS</th>\n",
       "      <th>LYMPH_NODES_EXAMINED_POSITIVE</th>\n",
       "      <th>NPI</th>\n",
       "      <th>AGE_AT_DIAGNOSIS</th>\n",
       "      <th>TUMOR_SIZE</th>\n",
       "      <th>TMB_NONSYNONYMOUS</th>\n",
       "      <th>CELLULARITY</th>\n",
       "      <th>CHEMOTHERAPY</th>\n",
       "      <th>...</th>\n",
       "      <th>FXYD4_mrna</th>\n",
       "      <th>TPD52L1_mrna</th>\n",
       "      <th>ENC1_mrna</th>\n",
       "      <th>DNAH11_mrna_y</th>\n",
       "      <th>ALDOC_mrna</th>\n",
       "      <th>GRB2_mrna</th>\n",
       "      <th>GPR156_mrna</th>\n",
       "      <th>VANGL1_mrna</th>\n",
       "      <th>NKX2-2_mrna</th>\n",
       "      <th>UMODL1_mrna</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>342</th>\n",
       "      <td>121.966667</td>\n",
       "      <td>0:Not Recurred</td>\n",
       "      <td>Living</td>\n",
       "      <td>1</td>\n",
       "      <td>4.042</td>\n",
       "      <td>79.52</td>\n",
       "      <td>21.0</td>\n",
       "      <td>9.152624</td>\n",
       "      <td>High</td>\n",
       "      <td>NO</td>\n",
       "      <td>...</td>\n",
       "      <td>0.992813</td>\n",
       "      <td>1.416946</td>\n",
       "      <td>-1.223732</td>\n",
       "      <td>-1.054836</td>\n",
       "      <td>-1.081067</td>\n",
       "      <td>-0.300710</td>\n",
       "      <td>0.919122</td>\n",
       "      <td>0.161028</td>\n",
       "      <td>-0.428187</td>\n",
       "      <td>0.224989</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>325</th>\n",
       "      <td>149.866667</td>\n",
       "      <td>0:Not Recurred</td>\n",
       "      <td>Living</td>\n",
       "      <td>0</td>\n",
       "      <td>2.040</td>\n",
       "      <td>61.79</td>\n",
       "      <td>20.0</td>\n",
       "      <td>11.767659</td>\n",
       "      <td>Moderate</td>\n",
       "      <td>NO</td>\n",
       "      <td>...</td>\n",
       "      <td>0.009157</td>\n",
       "      <td>0.942511</td>\n",
       "      <td>0.392749</td>\n",
       "      <td>0.240805</td>\n",
       "      <td>-0.491881</td>\n",
       "      <td>0.271729</td>\n",
       "      <td>-0.483658</td>\n",
       "      <td>0.484974</td>\n",
       "      <td>-0.176472</td>\n",
       "      <td>-0.728284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>946</th>\n",
       "      <td>45.533333</td>\n",
       "      <td>1:Recurred</td>\n",
       "      <td>Died of Disease</td>\n",
       "      <td>0</td>\n",
       "      <td>3.032</td>\n",
       "      <td>43.63</td>\n",
       "      <td>16.0</td>\n",
       "      <td>10.460142</td>\n",
       "      <td>Moderate</td>\n",
       "      <td>NO</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.491750</td>\n",
       "      <td>0.745315</td>\n",
       "      <td>0.274270</td>\n",
       "      <td>0.850794</td>\n",
       "      <td>-0.957869</td>\n",
       "      <td>-0.096607</td>\n",
       "      <td>1.658370</td>\n",
       "      <td>-0.290612</td>\n",
       "      <td>-0.415236</td>\n",
       "      <td>-0.821329</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1011</th>\n",
       "      <td>187.033333</td>\n",
       "      <td>0:Not Recurred</td>\n",
       "      <td>Died of Other Causes</td>\n",
       "      <td>1</td>\n",
       "      <td>4.030</td>\n",
       "      <td>74.46</td>\n",
       "      <td>15.0</td>\n",
       "      <td>6.537589</td>\n",
       "      <td>High</td>\n",
       "      <td>NO</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.023318</td>\n",
       "      <td>0.906088</td>\n",
       "      <td>-0.305917</td>\n",
       "      <td>-0.715573</td>\n",
       "      <td>-0.390119</td>\n",
       "      <td>-0.838757</td>\n",
       "      <td>0.300666</td>\n",
       "      <td>-1.049251</td>\n",
       "      <td>-0.455549</td>\n",
       "      <td>-0.755907</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1168</th>\n",
       "      <td>21.066667</td>\n",
       "      <td>1:Recurred</td>\n",
       "      <td>Died of Disease</td>\n",
       "      <td>3</td>\n",
       "      <td>4.090</td>\n",
       "      <td>60.41</td>\n",
       "      <td>45.0</td>\n",
       "      <td>3.922553</td>\n",
       "      <td>High</td>\n",
       "      <td>NO</td>\n",
       "      <td>...</td>\n",
       "      <td>0.113393</td>\n",
       "      <td>0.353704</td>\n",
       "      <td>0.089972</td>\n",
       "      <td>0.233243</td>\n",
       "      <td>0.291860</td>\n",
       "      <td>0.632116</td>\n",
       "      <td>0.926125</td>\n",
       "      <td>-0.171070</td>\n",
       "      <td>2.668258</td>\n",
       "      <td>-0.582256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1385</th>\n",
       "      <td>239.166667</td>\n",
       "      <td>0:Not Recurred</td>\n",
       "      <td>Living</td>\n",
       "      <td>6</td>\n",
       "      <td>5.070</td>\n",
       "      <td>58.92</td>\n",
       "      <td>35.0</td>\n",
       "      <td>9.152624</td>\n",
       "      <td>Low</td>\n",
       "      <td>NO</td>\n",
       "      <td>...</td>\n",
       "      <td>0.118282</td>\n",
       "      <td>0.365060</td>\n",
       "      <td>0.822153</td>\n",
       "      <td>0.158290</td>\n",
       "      <td>1.741177</td>\n",
       "      <td>-1.079096</td>\n",
       "      <td>-0.040037</td>\n",
       "      <td>-0.530432</td>\n",
       "      <td>3.234289</td>\n",
       "      <td>-1.201279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>965</th>\n",
       "      <td>62.633333</td>\n",
       "      <td>1:Recurred</td>\n",
       "      <td>Died of Disease</td>\n",
       "      <td>1</td>\n",
       "      <td>4.040</td>\n",
       "      <td>69.72</td>\n",
       "      <td>20.0</td>\n",
       "      <td>10.460142</td>\n",
       "      <td>High</td>\n",
       "      <td>NO</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.252079</td>\n",
       "      <td>0.657425</td>\n",
       "      <td>-0.533618</td>\n",
       "      <td>0.059456</td>\n",
       "      <td>-0.672819</td>\n",
       "      <td>0.489529</td>\n",
       "      <td>0.075651</td>\n",
       "      <td>-1.201659</td>\n",
       "      <td>-0.632220</td>\n",
       "      <td>-0.407730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>282</th>\n",
       "      <td>130.900000</td>\n",
       "      <td>0:Not Recurred</td>\n",
       "      <td>Living</td>\n",
       "      <td>4</td>\n",
       "      <td>4.052</td>\n",
       "      <td>60.00</td>\n",
       "      <td>26.0</td>\n",
       "      <td>2.615035</td>\n",
       "      <td>High</td>\n",
       "      <td>YES</td>\n",
       "      <td>...</td>\n",
       "      <td>0.101426</td>\n",
       "      <td>1.261980</td>\n",
       "      <td>-0.096220</td>\n",
       "      <td>-0.623950</td>\n",
       "      <td>1.074558</td>\n",
       "      <td>-0.049146</td>\n",
       "      <td>-0.609271</td>\n",
       "      <td>0.196321</td>\n",
       "      <td>-0.558113</td>\n",
       "      <td>1.014398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>607</th>\n",
       "      <td>199.966667</td>\n",
       "      <td>0:Not Recurred</td>\n",
       "      <td>Living</td>\n",
       "      <td>0</td>\n",
       "      <td>2.030</td>\n",
       "      <td>49.37</td>\n",
       "      <td>15.0</td>\n",
       "      <td>5.230071</td>\n",
       "      <td>Moderate</td>\n",
       "      <td>NO</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.654629</td>\n",
       "      <td>-0.214460</td>\n",
       "      <td>0.339306</td>\n",
       "      <td>0.374215</td>\n",
       "      <td>0.642533</td>\n",
       "      <td>-0.306414</td>\n",
       "      <td>-0.694937</td>\n",
       "      <td>-0.365147</td>\n",
       "      <td>-0.654322</td>\n",
       "      <td>-0.322543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>474</th>\n",
       "      <td>101.266667</td>\n",
       "      <td>0:Not Recurred</td>\n",
       "      <td>Died of Other Causes</td>\n",
       "      <td>1</td>\n",
       "      <td>5.034</td>\n",
       "      <td>70.19</td>\n",
       "      <td>17.0</td>\n",
       "      <td>5.230071</td>\n",
       "      <td>High</td>\n",
       "      <td>NO</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.737820</td>\n",
       "      <td>-0.181638</td>\n",
       "      <td>0.534758</td>\n",
       "      <td>0.357674</td>\n",
       "      <td>2.884460</td>\n",
       "      <td>-0.691514</td>\n",
       "      <td>-0.037620</td>\n",
       "      <td>0.060212</td>\n",
       "      <td>-0.632401</td>\n",
       "      <td>-0.277618</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>426 rows × 250 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 79
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-19T23:11:46.119623Z",
     "start_time": "2026-01-19T23:11:44.379441Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##output all dataset ready for prediction models\n",
    "X_train_integrate_1_imp.to_csv(\"/Users/mac1/Desktop/X_train_integrate_1_imp.csv\", index=False)\n",
    "X_test_integrate_1_imp.to_csv(\"/Users/mac1/Desktop/X_test_integrate_1_imp.csv\", index=False)\n",
    "y_train_integrate_1.to_csv(\"/Users/mac1/Desktop/y_train_integrate_1.csv\", index=False)\n",
    "y_test_integrate_1.to_csv(\"/Users/mac1/Desktop/y_test_integrate_1.csv\", index=False)\n",
    "X_train_integrate_2_imp.to_csv(\"/Users/mac1/Desktop/X_train_integrate_2_imp.csv\", index=False)\n",
    "X_test_integrate_2_imp.to_csv(\"/Users/mac1/Desktop/X_test_integrate_2_imp.csv\", index=False)\n",
    "y_train_integrate_2.to_csv(\"/Users/mac1/Desktop/y_train_integrate_2.csv\", index=False)\n",
    "y_test_integrate_2.to_csv(\"/Users/mac1/Desktop/y_test_integrate_2.csv\", index=False)\n",
    "X_train_overall_imp.to_csv(\"/Users/mac1/Desktop/X_train_overall_imp.csv\", index=False)\n",
    "X_test_overall_imp.to_csv(\"/Users/mac1/Desktop/X_test_overall_imp.csv\", index=False)\n",
    "y_train_overall.to_csv(\"/Users/mac1/Desktop/y_train_overall.csv\", index=False)\n",
    "y_test_overall.to_csv(\"/Users/mac1/Desktop/y_test_overall.csv\", index=False)"
   ],
   "id": "8dcf44f40d22a512",
   "outputs": [],
   "execution_count": 82
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2026-01-19T23:09:16.525249Z",
     "start_time": "2026-01-19T23:09:16.452141Z"
    }
   },
   "cell_type": "code",
   "source": [
    "##ID list and feature set\n",
    "keep_ids_overall.to_frame(name=\"PATIENT_ID\").to_csv(\n",
    "    \"/Users/mac1/Desktop/keep_ids_overall.csv\", index=False\n",
    ")\n",
    "keep_ids_integrate.to_frame(name=\"PATIENT_ID\").to_csv(\n",
    "    \"/Users/mac1/Desktop/keep_ids_integrate.csv\", index=False\n",
    ")"
   ],
   "id": "cf0ef553b39e5d76",
   "outputs": [],
   "execution_count": 81
  }
 ],
 "metadata": {
  "kernelspec": {
   "name": "python3",
   "language": "python",
   "display_name": "Python 3 (ipykernel)"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
