{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "f295d4cf-b508-495a-9436-a1e1d0a73c37",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c6e92ae0-8aba-471f-8d56-7139f0030f83",
   "metadata": {},
   "outputs": [],
   "source": [
    "# The following file is returned from \"Ensembl\"\n",
    "filepath = \"T2KEssJatkFJFqoq.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "519c8a74-13f6-4a8c-b332-71db1c1379e2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>›#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>...</th>\n",
       "      <th>AF</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1_1986896_A/G</td>\n",
       "      <td>1:1986896-1986896</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000378567.3</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1_1986896_A/G</td>\n",
       "      <td>1:1986896-1986896</td>\n",
       "      <td>G</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000466651.1</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1_1986896_A/G</td>\n",
       "      <td>1:1986896-1986896</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000468310.1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1_1986896_A/G</td>\n",
       "      <td>1:1986896-1986896</td>\n",
       "      <td>G</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000479566.1</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1_1986896_A/G</td>\n",
       "      <td>1:1986896-1986896</td>\n",
       "      <td>G</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000484419.1</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101496</th>\n",
       "      <td>X_153138145_C/G</td>\n",
       "      <td>X:153138145-153138145</td>\n",
       "      <td>G</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>ENSG00000198910</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000464967.1</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101497</th>\n",
       "      <td>X_153138145_C/G</td>\n",
       "      <td>X:153138145-153138145</td>\n",
       "      <td>G</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>ENSG00000198910</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000484652.1</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101498</th>\n",
       "      <td>X_153138145_C/G</td>\n",
       "      <td>X:153138145-153138145</td>\n",
       "      <td>G</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>ENSG00000198910</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000496122.1</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101499</th>\n",
       "      <td>X_153138145_C/G</td>\n",
       "      <td>X:153138145-153138145</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>ENSG00000198910</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000538883.1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101500</th>\n",
       "      <td>X_153138145_C/G</td>\n",
       "      <td>X:153138145-153138145</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>ENSG00000198910</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000543994.1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>101501 rows × 44 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       ›#Uploaded_variation               Location Allele  \\\n",
       "0             1_1986896_A/G      1:1986896-1986896      G   \n",
       "1             1_1986896_A/G      1:1986896-1986896      G   \n",
       "2             1_1986896_A/G      1:1986896-1986896      G   \n",
       "3             1_1986896_A/G      1:1986896-1986896      G   \n",
       "4             1_1986896_A/G      1:1986896-1986896      G   \n",
       "...                     ...                    ...    ...   \n",
       "101496      X_153138145_C/G  X:153138145-153138145      G   \n",
       "101497      X_153138145_C/G  X:153138145-153138145      G   \n",
       "101498      X_153138145_C/G  X:153138145-153138145      G   \n",
       "101499      X_153138145_C/G  X:153138145-153138145      G   \n",
       "101500      X_153138145_C/G  X:153138145-153138145      G   \n",
       "\n",
       "                               Consequence    IMPACT SYMBOL             Gene  \\\n",
       "0                         missense_variant  MODERATE  PRKCZ  ENSG00000067606   \n",
       "1                    upstream_gene_variant  MODIFIER  PRKCZ  ENSG00000067606   \n",
       "2                         missense_variant  MODERATE  PRKCZ  ENSG00000067606   \n",
       "3       non_coding_transcript_exon_variant  MODIFIER  PRKCZ  ENSG00000067606   \n",
       "4       non_coding_transcript_exon_variant  MODIFIER  PRKCZ  ENSG00000067606   \n",
       "...                                    ...       ...    ...              ...   \n",
       "101496  non_coding_transcript_exon_variant  MODIFIER  L1CAM  ENSG00000198910   \n",
       "101497               upstream_gene_variant  MODIFIER  L1CAM  ENSG00000198910   \n",
       "101498               upstream_gene_variant  MODIFIER  L1CAM  ENSG00000198910   \n",
       "101499                    missense_variant  MODERATE  L1CAM  ENSG00000198910   \n",
       "101500                    missense_variant  MODERATE  L1CAM  ENSG00000198910   \n",
       "\n",
       "       Feature_type            Feature               BIOTYPE  ... AF  \\\n",
       "0        Transcript  ENST00000378567.3        protein_coding  ...  -   \n",
       "1        Transcript  ENST00000466651.1  processed_transcript  ...  -   \n",
       "2        Transcript  ENST00000468310.1        protein_coding  ...  -   \n",
       "3        Transcript  ENST00000479566.1  processed_transcript  ...  -   \n",
       "4        Transcript  ENST00000484419.1       retained_intron  ...  -   \n",
       "...             ...                ...                   ...  ... ..   \n",
       "101496   Transcript  ENST00000464967.1       retained_intron  ...  -   \n",
       "101497   Transcript  ENST00000484652.1       retained_intron  ...  -   \n",
       "101498   Transcript  ENST00000496122.1       retained_intron  ...  -   \n",
       "101499   Transcript  ENST00000538883.1        protein_coding  ...  -   \n",
       "101500   Transcript  ENST00000543994.1        protein_coding  ...  -   \n",
       "\n",
       "                    CLIN_SIG SOMATIC PHENO PUBMED MOTIF_NAME MOTIF_POS  \\\n",
       "0                          -       -     -      -          -         -   \n",
       "1                          -       -     -      -          -         -   \n",
       "2                          -       -     -      -          -         -   \n",
       "3                          -       -     -      -          -         -   \n",
       "4                          -       -     -      -          -         -   \n",
       "...                      ...     ...   ...    ...        ...       ...   \n",
       "101496  benign,likely_benign       -     1      -          -         -   \n",
       "101497  benign,likely_benign       -     1      -          -         -   \n",
       "101498  benign,likely_benign       -     1      -          -         -   \n",
       "101499  benign,likely_benign       -     1      -          -         -   \n",
       "101500  benign,likely_benign       -     1      -          -         -   \n",
       "\n",
       "       HIGH_INF_POS MOTIF_SCORE_CHANGE TRANSCRIPTION_FACTORS  \n",
       "0                 -                  -                     -  \n",
       "1                 -                  -                     -  \n",
       "2                 -                  -                     -  \n",
       "3                 -                  -                     -  \n",
       "4                 -                  -                     -  \n",
       "...             ...                ...                   ...  \n",
       "101496            -                  -                     -  \n",
       "101497            -                  -                     -  \n",
       "101498            -                  -                     -  \n",
       "101499            -                  -                     -  \n",
       "101500            -                  -                     -  \n",
       "\n",
       "[101501 rows x 44 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(\n",
    "   filepath,\n",
    "    sep=\"\\t\"\n",
    ")\n",
    "\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "99d773dd-58ed-4171-9444-6e1bc292c34c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['›#Uploaded_variation', 'Location', 'Allele', 'Consequence', 'IMPACT', 'SYMBOL', 'Gene', 'Feature_type', 'Feature', 'BIOTYPE', 'EXON', 'INTRON', 'HGVSc', 'HGVSp', 'cDNA_position', 'CDS_position', 'Protein_position', 'Amino_acids', 'Codons', 'Existing_variation', 'REF_ALLELE', 'UPLOADED_ALLELE', 'DISTANCE', 'STRAND', 'FLAGS', 'SYMBOL_SOURCE', 'HGNC_ID', 'MANE', 'MANE_SELECT', 'MANE_PLUS_CLINICAL', 'TSL', 'APPRIS', 'SIFT', 'PolyPhen', 'AF', 'CLIN_SIG', 'SOMATIC', 'PHENO', 'PUBMED', 'MOTIF_NAME', 'MOTIF_POS', 'HIGH_INF_POS', 'MOTIF_SCORE_CHANGE', 'TRANSCRIPTION_FACTORS']\n"
     ]
    }
   ],
   "source": [
    "print(df.columns.tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "eae84e36-34a2-4046-95b1-31bfca7eb469",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SYMBOL\n",
       "TP53             14376\n",
       "SYNE1             4636\n",
       "AHNAK2            3033\n",
       "PDE4DIP           2995\n",
       "KMT2C             2139\n",
       "                 ...  \n",
       "RP11-199F11.2        1\n",
       "AC087239.1           1\n",
       "RP11-461L13.5        1\n",
       "AC005884.1           1\n",
       "RP11-848P1.5         1\n",
       "Name: count, Length: 405, dtype: int64"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"SYMBOL\"].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "310d9543-6e05-41ad-b10b-7011191fd550",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['›#Uploaded_variation', 'Location', 'Allele', 'Consequence', 'IMPACT', 'SYMBOL', 'Gene', 'Feature_type', 'Feature', 'BIOTYPE', 'EXON', 'INTRON', 'HGVSc', 'HGVSp', 'cDNA_position', 'CDS_position', 'Protein_position', 'Amino_acids', 'Codons', 'Existing_variation', 'REF_ALLELE', 'UPLOADED_ALLELE', 'DISTANCE', 'STRAND', 'FLAGS', 'SYMBOL_SOURCE', 'HGNC_ID', 'MANE', 'MANE_SELECT', 'MANE_PLUS_CLINICAL', 'TSL', 'APPRIS', 'SIFT', 'PolyPhen', 'AF', 'CLIN_SIG', 'SOMATIC', 'PHENO', 'PUBMED', 'MOTIF_NAME', 'MOTIF_POS', 'HIGH_INF_POS', 'MOTIF_SCORE_CHANGE', 'TRANSCRIPTION_FACTORS']\n"
     ]
    }
   ],
   "source": [
    "print(df.columns.tolist())"
   ]
  },
  {
   "cell_type": "raw",
   "id": "b2eda6ae-b822-4282-88e6-68baedd5df2b",
   "metadata": {},
   "source": [
    "cols_to_keep = [\n",
    "    \"#Uploaded_variation\",\n",
    "    \"Location\",\n",
    "    \"Existing_variation\",\n",
    "    \"SYMBOL\",\n",
    "    \"Gene\",\n",
    "    \"Consequence\",\n",
    "    \"IMPACT\",\n",
    "    \"CLIN_SIG\",\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "9655a667-6b37-414e-b6a8-95f4cc2f4404",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Existing_variation\n",
       "-                                                                                   17391\n",
       "rs121434592,CM1611160,COSV62571334,COSV62576849                                       948\n",
       "rs121913279,CM153086,COSV55873195,COSV55873401,COSV55888015                           932\n",
       "rs28934578,CM062017,CM951224,COSV52661038,COSV52677601,COSV52731306,COSV53801286      663\n",
       "rs121912651,CM010465,CM900211,COSV52662035,COSV52797251,COSV53097881                  527\n",
       "                                                                                    ...  \n",
       "rs397517199,COSV104381248,COSV55880235,COSV55880477                                     1\n",
       "rs2108399415                                                                            1\n",
       "COSV56004160                                                                            1\n",
       "rs202013300,COSV55889280                                                                1\n",
       "rs375468997                                                                             1\n",
       "Name: count, Length: 9945, dtype: int64"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Consequence\n",
    "#IMPACT\n",
    "\n",
    "#Protein_position\n",
    "#Amino_acids\n",
    "#Codons\n",
    "\n",
    "df[\"Existing_variation\"].value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "1ae06884-0095-42da-b4be-b6051bffe206",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "9945"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df[\"Existing_variation\"] != \".\", \"Existing_variation\"].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "9bfe3cd1-ba87-460d-8c48-7786d01ecf66",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_new = df[[\"›#Uploaded_variation\", \"Location\", \"Existing_variation\", \"SYMBOL\",\n",
    "    \"Gene\",\n",
    "    \"SYMBOL\",\n",
    "    \"Consequence\",\n",
    "    \"IMPACT\",\n",
    "    \"CLIN_SIG\",]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "bbdfe34f-0957-4c15-a8c1-9826290b4d68",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/hn/_7rb9zk51r13c_fprldgz1zc0000gn/T/ipykernel_15627/600224594.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_new[\"rsid\"] = df[\"Existing_variation\"].str.extract(r\"(rs\\d+)\")\n",
      "/var/folders/hn/_7rb9zk51r13c_fprldgz1zc0000gn/T/ipykernel_15627/600224594.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_new[\"clinvar\"] = df[\"Existing_variation\"].str.extract(r\"(CM\\d+)\")\n"
     ]
    }
   ],
   "source": [
    "df_new[\"rsid\"] = df[\"Existing_variation\"].str.extract(r\"(rs\\d+)\")\n",
    "df_new[\"clinvar\"] = df[\"Existing_variation\"].str.extract(r\"(CM\\d+)\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "227302c8-b991-43c2-a0bf-c807b4c2fdb2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>›#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>rsid</th>\n",
       "      <th>clinvar</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1_1986896_A/G</td>\n",
       "      <td>1:1986896-1986896</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1_1986896_A/G</td>\n",
       "      <td>1:1986896-1986896</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1_1986896_A/G</td>\n",
       "      <td>1:1986896-1986896</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1_1986896_A/G</td>\n",
       "      <td>1:1986896-1986896</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1_1986896_A/G</td>\n",
       "      <td>1:1986896-1986896</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101496</th>\n",
       "      <td>X_153138145_C/G</td>\n",
       "      <td>X:153138145-153138145</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>ENSG00000198910</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101497</th>\n",
       "      <td>X_153138145_C/G</td>\n",
       "      <td>X:153138145-153138145</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>ENSG00000198910</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101498</th>\n",
       "      <td>X_153138145_C/G</td>\n",
       "      <td>X:153138145-153138145</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>ENSG00000198910</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101499</th>\n",
       "      <td>X_153138145_C/G</td>\n",
       "      <td>X:153138145-153138145</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>ENSG00000198910</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101500</th>\n",
       "      <td>X_153138145_C/G</td>\n",
       "      <td>X:153138145-153138145</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>ENSG00000198910</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>101501 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       ›#Uploaded_variation               Location Existing_variation SYMBOL  \\\n",
       "0             1_1986896_A/G      1:1986896-1986896        rs765901754  PRKCZ   \n",
       "1             1_1986896_A/G      1:1986896-1986896        rs765901754  PRKCZ   \n",
       "2             1_1986896_A/G      1:1986896-1986896        rs765901754  PRKCZ   \n",
       "3             1_1986896_A/G      1:1986896-1986896        rs765901754  PRKCZ   \n",
       "4             1_1986896_A/G      1:1986896-1986896        rs765901754  PRKCZ   \n",
       "...                     ...                    ...                ...    ...   \n",
       "101496      X_153138145_C/G  X:153138145-153138145        rs201990980  L1CAM   \n",
       "101497      X_153138145_C/G  X:153138145-153138145        rs201990980  L1CAM   \n",
       "101498      X_153138145_C/G  X:153138145-153138145        rs201990980  L1CAM   \n",
       "101499      X_153138145_C/G  X:153138145-153138145        rs201990980  L1CAM   \n",
       "101500      X_153138145_C/G  X:153138145-153138145        rs201990980  L1CAM   \n",
       "\n",
       "                   Gene SYMBOL                         Consequence    IMPACT  \\\n",
       "0       ENSG00000067606  PRKCZ                    missense_variant  MODERATE   \n",
       "1       ENSG00000067606  PRKCZ               upstream_gene_variant  MODIFIER   \n",
       "2       ENSG00000067606  PRKCZ                    missense_variant  MODERATE   \n",
       "3       ENSG00000067606  PRKCZ  non_coding_transcript_exon_variant  MODIFIER   \n",
       "4       ENSG00000067606  PRKCZ  non_coding_transcript_exon_variant  MODIFIER   \n",
       "...                 ...    ...                                 ...       ...   \n",
       "101496  ENSG00000198910  L1CAM  non_coding_transcript_exon_variant  MODIFIER   \n",
       "101497  ENSG00000198910  L1CAM               upstream_gene_variant  MODIFIER   \n",
       "101498  ENSG00000198910  L1CAM               upstream_gene_variant  MODIFIER   \n",
       "101499  ENSG00000198910  L1CAM                    missense_variant  MODERATE   \n",
       "101500  ENSG00000198910  L1CAM                    missense_variant  MODERATE   \n",
       "\n",
       "                    CLIN_SIG         rsid clinvar  \n",
       "0                          -  rs765901754     NaN  \n",
       "1                          -  rs765901754     NaN  \n",
       "2                          -  rs765901754     NaN  \n",
       "3                          -  rs765901754     NaN  \n",
       "4                          -  rs765901754     NaN  \n",
       "...                      ...          ...     ...  \n",
       "101496  benign,likely_benign  rs201990980     NaN  \n",
       "101497  benign,likely_benign  rs201990980     NaN  \n",
       "101498  benign,likely_benign  rs201990980     NaN  \n",
       "101499  benign,likely_benign  rs201990980     NaN  \n",
       "101500  benign,likely_benign  rs201990980     NaN  \n",
       "\n",
       "[101501 rows x 11 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "7c8d3f0a-65ac-49bb-964c-62aeb751fbfd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8939"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_new.loc[df_new[\"rsid\"] != \".\", \"rsid\"].nunique()"
   ]
  },
  {
   "cell_type": "raw",
   "id": "fca1a814-874e-4e1a-8b50-aaddbb001514",
   "metadata": {},
   "source": [
    "#df_new.to_csv(\n",
    "#    \"annotated_df5.csv\",\n",
    "#    sep=\"\\t\",\n",
    "#    index=False\n",
    "#)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "aead1d3e-ce1d-42ab-84ae-4dd223e3781f",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/hn/_7rb9zk51r13c_fprldgz1zc0000gn/T/ipykernel_7155/3551136093.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_new[\"Chromosome\"] = tmp[0]\n",
      "/var/folders/hn/_7rb9zk51r13c_fprldgz1zc0000gn/T/ipykernel_7155/3551136093.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_new[\"Start_Position\"] = pos[0].astype(int)\n",
      "/var/folders/hn/_7rb9zk51r13c_fprldgz1zc0000gn/T/ipykernel_7155/3551136093.py:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_new[\"End_Position\"] = pos[1].astype(int)\n",
      "/var/folders/hn/_7rb9zk51r13c_fprldgz1zc0000gn/T/ipykernel_7155/3551136093.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_new[\"Allele\"] = (\n"
     ]
    }
   ],
   "source": [
    "tmp = df_new[\"Location\"].str.split(\":\", expand=True)\n",
    "pos = tmp[1].str.split(\"-\", expand=True)\n",
    "\n",
    "df_new[\"Chromosome\"] = tmp[0]\n",
    "df_new[\"Start_Position\"] = pos[0].astype(int)\n",
    "df_new[\"End_Position\"] = pos[1].astype(int)\n",
    "\n",
    "df_new[\"Allele\"] = (\n",
    "    df_new[\"›#Uploaded_variation\"]\n",
    "    .str.split(\"_\", expand=True)[2]\n",
    ")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "fcf93e17-89e8-48b9-8c27-44fe1c1df5a2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>›#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>rsid</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1_1986896_A/G</td>\n",
       "      <td>1:1986896-1986896</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1986896</td>\n",
       "      <td>1986896</td>\n",
       "      <td>A/G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1_1986896_A/G</td>\n",
       "      <td>1:1986896-1986896</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1986896</td>\n",
       "      <td>1986896</td>\n",
       "      <td>A/G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1_1986896_A/G</td>\n",
       "      <td>1:1986896-1986896</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>-</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1986896</td>\n",
       "      <td>1986896</td>\n",
       "      <td>A/G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1_1986896_A/G</td>\n",
       "      <td>1:1986896-1986896</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1986896</td>\n",
       "      <td>1986896</td>\n",
       "      <td>A/G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1_1986896_A/G</td>\n",
       "      <td>1:1986896-1986896</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>ENSG00000067606</td>\n",
       "      <td>PRKCZ</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>rs765901754</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1986896</td>\n",
       "      <td>1986896</td>\n",
       "      <td>A/G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101496</th>\n",
       "      <td>X_153138145_C/G</td>\n",
       "      <td>X:153138145-153138145</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>ENSG00000198910</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X</td>\n",
       "      <td>153138145</td>\n",
       "      <td>153138145</td>\n",
       "      <td>C/G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101497</th>\n",
       "      <td>X_153138145_C/G</td>\n",
       "      <td>X:153138145-153138145</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>ENSG00000198910</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X</td>\n",
       "      <td>153138145</td>\n",
       "      <td>153138145</td>\n",
       "      <td>C/G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101498</th>\n",
       "      <td>X_153138145_C/G</td>\n",
       "      <td>X:153138145-153138145</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>ENSG00000198910</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X</td>\n",
       "      <td>153138145</td>\n",
       "      <td>153138145</td>\n",
       "      <td>C/G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101499</th>\n",
       "      <td>X_153138145_C/G</td>\n",
       "      <td>X:153138145-153138145</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>ENSG00000198910</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X</td>\n",
       "      <td>153138145</td>\n",
       "      <td>153138145</td>\n",
       "      <td>C/G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101500</th>\n",
       "      <td>X_153138145_C/G</td>\n",
       "      <td>X:153138145-153138145</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>ENSG00000198910</td>\n",
       "      <td>L1CAM</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>benign,likely_benign</td>\n",
       "      <td>rs201990980</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X</td>\n",
       "      <td>153138145</td>\n",
       "      <td>153138145</td>\n",
       "      <td>C/G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>101501 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       ›#Uploaded_variation               Location Existing_variation SYMBOL  \\\n",
       "0             1_1986896_A/G      1:1986896-1986896        rs765901754  PRKCZ   \n",
       "1             1_1986896_A/G      1:1986896-1986896        rs765901754  PRKCZ   \n",
       "2             1_1986896_A/G      1:1986896-1986896        rs765901754  PRKCZ   \n",
       "3             1_1986896_A/G      1:1986896-1986896        rs765901754  PRKCZ   \n",
       "4             1_1986896_A/G      1:1986896-1986896        rs765901754  PRKCZ   \n",
       "...                     ...                    ...                ...    ...   \n",
       "101496      X_153138145_C/G  X:153138145-153138145        rs201990980  L1CAM   \n",
       "101497      X_153138145_C/G  X:153138145-153138145        rs201990980  L1CAM   \n",
       "101498      X_153138145_C/G  X:153138145-153138145        rs201990980  L1CAM   \n",
       "101499      X_153138145_C/G  X:153138145-153138145        rs201990980  L1CAM   \n",
       "101500      X_153138145_C/G  X:153138145-153138145        rs201990980  L1CAM   \n",
       "\n",
       "                   Gene SYMBOL                         Consequence    IMPACT  \\\n",
       "0       ENSG00000067606  PRKCZ                    missense_variant  MODERATE   \n",
       "1       ENSG00000067606  PRKCZ               upstream_gene_variant  MODIFIER   \n",
       "2       ENSG00000067606  PRKCZ                    missense_variant  MODERATE   \n",
       "3       ENSG00000067606  PRKCZ  non_coding_transcript_exon_variant  MODIFIER   \n",
       "4       ENSG00000067606  PRKCZ  non_coding_transcript_exon_variant  MODIFIER   \n",
       "...                 ...    ...                                 ...       ...   \n",
       "101496  ENSG00000198910  L1CAM  non_coding_transcript_exon_variant  MODIFIER   \n",
       "101497  ENSG00000198910  L1CAM               upstream_gene_variant  MODIFIER   \n",
       "101498  ENSG00000198910  L1CAM               upstream_gene_variant  MODIFIER   \n",
       "101499  ENSG00000198910  L1CAM                    missense_variant  MODERATE   \n",
       "101500  ENSG00000198910  L1CAM                    missense_variant  MODERATE   \n",
       "\n",
       "                    CLIN_SIG         rsid clinvar Chromosome  Start_Position  \\\n",
       "0                          -  rs765901754     NaN          1         1986896   \n",
       "1                          -  rs765901754     NaN          1         1986896   \n",
       "2                          -  rs765901754     NaN          1         1986896   \n",
       "3                          -  rs765901754     NaN          1         1986896   \n",
       "4                          -  rs765901754     NaN          1         1986896   \n",
       "...                      ...          ...     ...        ...             ...   \n",
       "101496  benign,likely_benign  rs201990980     NaN          X       153138145   \n",
       "101497  benign,likely_benign  rs201990980     NaN          X       153138145   \n",
       "101498  benign,likely_benign  rs201990980     NaN          X       153138145   \n",
       "101499  benign,likely_benign  rs201990980     NaN          X       153138145   \n",
       "101500  benign,likely_benign  rs201990980     NaN          X       153138145   \n",
       "\n",
       "        End_Position Allele  \n",
       "0            1986896    A/G  \n",
       "1            1986896    A/G  \n",
       "2            1986896    A/G  \n",
       "3            1986896    A/G  \n",
       "4            1986896    A/G  \n",
       "...              ...    ...  \n",
       "101496     153138145    C/G  \n",
       "101497     153138145    C/G  \n",
       "101498     153138145    C/G  \n",
       "101499     153138145    C/G  \n",
       "101500     153138145    C/G  \n",
       "\n",
       "[101501 rows x 15 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "222aa605-cf13-4d33-9984-ab6fae0a9757",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Keep the cols with the following values in functional_consequences function.\n",
    "functional_consequences = {\n",
    "    \"missense_variant\",\n",
    "    \"stop_gained\",\n",
    "    \"stop_lost\",\n",
    "    \"frameshift_variant\",\n",
    "    \"splice_acceptor_variant\",\n",
    "    \"splice_donor_variant\",\n",
    "    \"start_lost\",\n",
    "    \"inframe_insertion\",\n",
    "    \"inframe_deletion\",\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "686d2bae-1a7d-4f02-9bb6-06fbeed03c05",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Keep the pathogenic variants with HIGH or at least MODERATE impact.\n",
    "df_filtered = df_new[\n",
    "    df_new[\"IMPACT\"].isin([\"HIGH\", \"MODERATE\"])\n",
    "    &\n",
    "    df_new[\"Consequence\"].apply(\n",
    "        lambda x: any(c in x for c in functional_consequences)\n",
    "        if isinstance(x, str) else False\n",
    "    )\n",
    "\n",
    "    &\n",
    "    (df_new[\"CLIN_SIG\"].str.contains(\"pathogenic\", case=False, na=False))\n",
    "].copy()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "6f482efc-7f66-4c16-b839-6475dbc27142",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>›#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>rsid</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>541</th>\n",
       "      <td>1_27022913_CCCGCCGCCGCCAGCAGCCTGGGCAA/-</td>\n",
       "      <td>1:27022912-27022938</td>\n",
       "      <td>rs797045262</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>rs797045262</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>27022912</td>\n",
       "      <td>27022938</td>\n",
       "      <td>CCCGCCGCCGCCAGCAGCCTGGGCAA/-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>542</th>\n",
       "      <td>1_27022913_CCCGCCGCCGCCAGCAGCCTGGGCAA/-</td>\n",
       "      <td>1:27022912-27022938</td>\n",
       "      <td>rs797045262</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>rs797045262</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>27022912</td>\n",
       "      <td>27022938</td>\n",
       "      <td>CCCGCCGCCGCCAGCAGCCTGGGCAA/-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1232</th>\n",
       "      <td>1_27105880_C/G</td>\n",
       "      <td>1:27105880-27105880</td>\n",
       "      <td>rs150534917,COSV99055364</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>likely_benign,conflicting_interpretations_of_p...</td>\n",
       "      <td>rs150534917</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>27105880</td>\n",
       "      <td>27105880</td>\n",
       "      <td>C/G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1233</th>\n",
       "      <td>1_27105880_C/G</td>\n",
       "      <td>1:27105880-27105880</td>\n",
       "      <td>rs150534917,COSV99055364</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>likely_benign,conflicting_interpretations_of_p...</td>\n",
       "      <td>rs150534917</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>27105880</td>\n",
       "      <td>27105880</td>\n",
       "      <td>C/G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1234</th>\n",
       "      <td>1_27105880_C/G</td>\n",
       "      <td>1:27105880-27105880</td>\n",
       "      <td>rs150534917,COSV99055364</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>likely_benign,conflicting_interpretations_of_p...</td>\n",
       "      <td>rs150534917</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>27105880</td>\n",
       "      <td>27105880</td>\n",
       "      <td>C/G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99469</th>\n",
       "      <td>X_44969369_C/T</td>\n",
       "      <td>X:44969369-44969369</td>\n",
       "      <td>rs1318649487,CM138471,COSV65038966</td>\n",
       "      <td>KDM6A</td>\n",
       "      <td>ENSG00000147050</td>\n",
       "      <td>KDM6A</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>rs1318649487</td>\n",
       "      <td>CM138471</td>\n",
       "      <td>X</td>\n",
       "      <td>44969369</td>\n",
       "      <td>44969369</td>\n",
       "      <td>C/T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99470</th>\n",
       "      <td>X_44969369_C/T</td>\n",
       "      <td>X:44969369-44969369</td>\n",
       "      <td>rs1318649487,CM138471,COSV65038966</td>\n",
       "      <td>KDM6A</td>\n",
       "      <td>ENSG00000147050</td>\n",
       "      <td>KDM6A</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>rs1318649487</td>\n",
       "      <td>CM138471</td>\n",
       "      <td>X</td>\n",
       "      <td>44969369</td>\n",
       "      <td>44969369</td>\n",
       "      <td>C/T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99471</th>\n",
       "      <td>X_44969369_C/T</td>\n",
       "      <td>X:44969369-44969369</td>\n",
       "      <td>rs1318649487,CM138471,COSV65038966</td>\n",
       "      <td>KDM6A</td>\n",
       "      <td>ENSG00000147050</td>\n",
       "      <td>KDM6A</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>rs1318649487</td>\n",
       "      <td>CM138471</td>\n",
       "      <td>X</td>\n",
       "      <td>44969369</td>\n",
       "      <td>44969369</td>\n",
       "      <td>C/T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99473</th>\n",
       "      <td>X_44969369_C/T</td>\n",
       "      <td>X:44969369-44969369</td>\n",
       "      <td>rs1318649487,CM138471,COSV65038966</td>\n",
       "      <td>KDM6A</td>\n",
       "      <td>ENSG00000147050</td>\n",
       "      <td>KDM6A</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>rs1318649487</td>\n",
       "      <td>CM138471</td>\n",
       "      <td>X</td>\n",
       "      <td>44969369</td>\n",
       "      <td>44969369</td>\n",
       "      <td>C/T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99474</th>\n",
       "      <td>X_44969369_C/T</td>\n",
       "      <td>X:44969369-44969369</td>\n",
       "      <td>rs1318649487,CM138471,COSV65038966</td>\n",
       "      <td>KDM6A</td>\n",
       "      <td>ENSG00000147050</td>\n",
       "      <td>KDM6A</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>rs1318649487</td>\n",
       "      <td>CM138471</td>\n",
       "      <td>X</td>\n",
       "      <td>44969369</td>\n",
       "      <td>44969369</td>\n",
       "      <td>C/T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9043 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          ›#Uploaded_variation             Location  \\\n",
       "541    1_27022913_CCCGCCGCCGCCAGCAGCCTGGGCAA/-  1:27022912-27022938   \n",
       "542    1_27022913_CCCGCCGCCGCCAGCAGCCTGGGCAA/-  1:27022912-27022938   \n",
       "1232                            1_27105880_C/G  1:27105880-27105880   \n",
       "1233                            1_27105880_C/G  1:27105880-27105880   \n",
       "1234                            1_27105880_C/G  1:27105880-27105880   \n",
       "...                                        ...                  ...   \n",
       "99469                           X_44969369_C/T  X:44969369-44969369   \n",
       "99470                           X_44969369_C/T  X:44969369-44969369   \n",
       "99471                           X_44969369_C/T  X:44969369-44969369   \n",
       "99473                           X_44969369_C/T  X:44969369-44969369   \n",
       "99474                           X_44969369_C/T  X:44969369-44969369   \n",
       "\n",
       "                       Existing_variation  SYMBOL             Gene  SYMBOL  \\\n",
       "541                           rs797045262  ARID1A  ENSG00000117713  ARID1A   \n",
       "542                           rs797045262  ARID1A  ENSG00000117713  ARID1A   \n",
       "1232             rs150534917,COSV99055364  ARID1A  ENSG00000117713  ARID1A   \n",
       "1233             rs150534917,COSV99055364  ARID1A  ENSG00000117713  ARID1A   \n",
       "1234             rs150534917,COSV99055364  ARID1A  ENSG00000117713  ARID1A   \n",
       "...                                   ...     ...              ...     ...   \n",
       "99469  rs1318649487,CM138471,COSV65038966   KDM6A  ENSG00000147050   KDM6A   \n",
       "99470  rs1318649487,CM138471,COSV65038966   KDM6A  ENSG00000147050   KDM6A   \n",
       "99471  rs1318649487,CM138471,COSV65038966   KDM6A  ENSG00000147050   KDM6A   \n",
       "99473  rs1318649487,CM138471,COSV65038966   KDM6A  ENSG00000147050   KDM6A   \n",
       "99474  rs1318649487,CM138471,COSV65038966   KDM6A  ENSG00000147050   KDM6A   \n",
       "\n",
       "              Consequence    IMPACT  \\\n",
       "541    frameshift_variant      HIGH   \n",
       "542    frameshift_variant      HIGH   \n",
       "1232     missense_variant  MODERATE   \n",
       "1233     missense_variant  MODERATE   \n",
       "1234     missense_variant  MODERATE   \n",
       "...                   ...       ...   \n",
       "99469         stop_gained      HIGH   \n",
       "99470         stop_gained      HIGH   \n",
       "99471         stop_gained      HIGH   \n",
       "99473         stop_gained      HIGH   \n",
       "99474         stop_gained      HIGH   \n",
       "\n",
       "                                                CLIN_SIG          rsid  \\\n",
       "541                                           pathogenic   rs797045262   \n",
       "542                                           pathogenic   rs797045262   \n",
       "1232   likely_benign,conflicting_interpretations_of_p...   rs150534917   \n",
       "1233   likely_benign,conflicting_interpretations_of_p...   rs150534917   \n",
       "1234   likely_benign,conflicting_interpretations_of_p...   rs150534917   \n",
       "...                                                  ...           ...   \n",
       "99469                                         pathogenic  rs1318649487   \n",
       "99470                                         pathogenic  rs1318649487   \n",
       "99471                                         pathogenic  rs1318649487   \n",
       "99473                                         pathogenic  rs1318649487   \n",
       "99474                                         pathogenic  rs1318649487   \n",
       "\n",
       "        clinvar Chromosome  Start_Position  End_Position  \\\n",
       "541         NaN          1        27022912      27022938   \n",
       "542         NaN          1        27022912      27022938   \n",
       "1232        NaN          1        27105880      27105880   \n",
       "1233        NaN          1        27105880      27105880   \n",
       "1234        NaN          1        27105880      27105880   \n",
       "...         ...        ...             ...           ...   \n",
       "99469  CM138471          X        44969369      44969369   \n",
       "99470  CM138471          X        44969369      44969369   \n",
       "99471  CM138471          X        44969369      44969369   \n",
       "99473  CM138471          X        44969369      44969369   \n",
       "99474  CM138471          X        44969369      44969369   \n",
       "\n",
       "                             Allele  \n",
       "541    CCCGCCGCCGCCAGCAGCCTGGGCAA/-  \n",
       "542    CCCGCCGCCGCCAGCAGCCTGGGCAA/-  \n",
       "1232                            C/G  \n",
       "1233                            C/G  \n",
       "1234                            C/G  \n",
       "...                             ...  \n",
       "99469                           C/T  \n",
       "99470                           C/T  \n",
       "99471                           C/T  \n",
       "99473                           C/T  \n",
       "99474                           C/T  \n",
       "\n",
       "[9043 rows x 15 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "b01fe659-fb37-47c9-967b-ffcbc1c7f099",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "686"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_filtered.loc[df_filtered[\"rsid\"] != \".\", \"rsid\"].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "d7e4fd31-7ffd-48ae-88db-43ccf47bf4d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Remove Duplicates\n",
    "df_filtered_unique = df_filtered.drop_duplicates(\n",
    "    subset=[\"Chromosome\", \"Start_Position\", \"End_Position\", \"Allele\", \"rsid\", \"CLIN_SIG\", \"IMPACT\", \"Consequence\"]\n",
    ")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "164f8e2f-db2e-49ea-b923-d973eda26ce1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>›#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>rsid</th>\n",
       "      <th>clinvar</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>541</th>\n",
       "      <td>1_27022913_CCCGCCGCCGCCAGCAGCCTGGGCAA/-</td>\n",
       "      <td>1:27022912-27022938</td>\n",
       "      <td>rs797045262</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>rs797045262</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>27022912</td>\n",
       "      <td>27022938</td>\n",
       "      <td>CCCGCCGCCGCCAGCAGCCTGGGCAA/-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1232</th>\n",
       "      <td>1_27105880_C/G</td>\n",
       "      <td>1:27105880-27105880</td>\n",
       "      <td>rs150534917,COSV99055364</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>ENSG00000117713</td>\n",
       "      <td>ARID1A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>likely_benign,conflicting_interpretations_of_p...</td>\n",
       "      <td>rs150534917</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>27105880</td>\n",
       "      <td>27105880</td>\n",
       "      <td>C/G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1596</th>\n",
       "      <td>1_65312347_C/A</td>\n",
       "      <td>1:65312347-65312347</td>\n",
       "      <td>rs1057519753,COSV61086927,COSV61091213</td>\n",
       "      <td>JAK1</td>\n",
       "      <td>ENSG00000162434</td>\n",
       "      <td>JAK1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>likely_pathogenic</td>\n",
       "      <td>rs1057519753</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>65312347</td>\n",
       "      <td>65312347</td>\n",
       "      <td>C/A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1939</th>\n",
       "      <td>1_115258748_C/T</td>\n",
       "      <td>1:115258748-115258748</td>\n",
       "      <td>rs121913250,CM136798,CM177295,COSV54736487,COS...</td>\n",
       "      <td>NRAS</td>\n",
       "      <td>ENSG00000213281</td>\n",
       "      <td>NRAS</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>likely_pathogenic,pathogenic</td>\n",
       "      <td>rs121913250</td>\n",
       "      <td>CM136798</td>\n",
       "      <td>1</td>\n",
       "      <td>115258748</td>\n",
       "      <td>115258748</td>\n",
       "      <td>C/T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5894</th>\n",
       "      <td>1_209791389_G/A</td>\n",
       "      <td>1:209791389-209791389</td>\n",
       "      <td>rs747916314,CM970866</td>\n",
       "      <td>LAMB3</td>\n",
       "      <td>ENSG00000196878</td>\n",
       "      <td>LAMB3</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>rs747916314</td>\n",
       "      <td>CM970866</td>\n",
       "      <td>1</td>\n",
       "      <td>209791389</td>\n",
       "      <td>209791389</td>\n",
       "      <td>G/A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98206</th>\n",
       "      <td>9_139412607_C/T</td>\n",
       "      <td>9:139412607-139412607</td>\n",
       "      <td>rs373770404,COSV53069197</td>\n",
       "      <td>NOTCH1</td>\n",
       "      <td>ENSG00000148400</td>\n",
       "      <td>NOTCH1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>uncertain_significance,conflicting_interpretat...</td>\n",
       "      <td>rs373770404</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>139412607</td>\n",
       "      <td>139412607</td>\n",
       "      <td>C/T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98215</th>\n",
       "      <td>9_139413049_G/A</td>\n",
       "      <td>9:139413049-139413049</td>\n",
       "      <td>rs1064795070,COSV53025487,COSV53045576</td>\n",
       "      <td>NOTCH1</td>\n",
       "      <td>ENSG00000148400</td>\n",
       "      <td>NOTCH1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>uncertain_significance,likely_pathogenic</td>\n",
       "      <td>rs1064795070</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>139413049</td>\n",
       "      <td>139413049</td>\n",
       "      <td>G/A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98242</th>\n",
       "      <td>9_139413921_T/C</td>\n",
       "      <td>9:139413921-139413921</td>\n",
       "      <td>rs367825691,CM1610439,COSV53052040</td>\n",
       "      <td>NOTCH1</td>\n",
       "      <td>ENSG00000148400</td>\n",
       "      <td>NOTCH1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>uncertain_significance,conflicting_interpretat...</td>\n",
       "      <td>rs367825691</td>\n",
       "      <td>CM1610439</td>\n",
       "      <td>9</td>\n",
       "      <td>139413921</td>\n",
       "      <td>139413921</td>\n",
       "      <td>T/C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98266</th>\n",
       "      <td>9_139417517_C/T</td>\n",
       "      <td>9:139417517-139417517</td>\n",
       "      <td>rs375065108,COSV53058648</td>\n",
       "      <td>NOTCH1</td>\n",
       "      <td>ENSG00000148400</td>\n",
       "      <td>NOTCH1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>uncertain_significance,conflicting_interpretat...</td>\n",
       "      <td>rs375065108</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>139417517</td>\n",
       "      <td>139417517</td>\n",
       "      <td>C/T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99467</th>\n",
       "      <td>X_44969369_C/T</td>\n",
       "      <td>X:44969369-44969369</td>\n",
       "      <td>rs1318649487,CM138471,COSV65038966</td>\n",
       "      <td>KDM6A</td>\n",
       "      <td>ENSG00000147050</td>\n",
       "      <td>KDM6A</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>rs1318649487</td>\n",
       "      <td>CM138471</td>\n",
       "      <td>X</td>\n",
       "      <td>44969369</td>\n",
       "      <td>44969369</td>\n",
       "      <td>C/T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>896 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          ›#Uploaded_variation               Location  \\\n",
       "541    1_27022913_CCCGCCGCCGCCAGCAGCCTGGGCAA/-    1:27022912-27022938   \n",
       "1232                            1_27105880_C/G    1:27105880-27105880   \n",
       "1596                            1_65312347_C/A    1:65312347-65312347   \n",
       "1939                           1_115258748_C/T  1:115258748-115258748   \n",
       "5894                           1_209791389_G/A  1:209791389-209791389   \n",
       "...                                        ...                    ...   \n",
       "98206                          9_139412607_C/T  9:139412607-139412607   \n",
       "98215                          9_139413049_G/A  9:139413049-139413049   \n",
       "98242                          9_139413921_T/C  9:139413921-139413921   \n",
       "98266                          9_139417517_C/T  9:139417517-139417517   \n",
       "99467                           X_44969369_C/T    X:44969369-44969369   \n",
       "\n",
       "                                      Existing_variation  SYMBOL  \\\n",
       "541                                          rs797045262  ARID1A   \n",
       "1232                            rs150534917,COSV99055364  ARID1A   \n",
       "1596              rs1057519753,COSV61086927,COSV61091213    JAK1   \n",
       "1939   rs121913250,CM136798,CM177295,COSV54736487,COS...    NRAS   \n",
       "5894                                rs747916314,CM970866   LAMB3   \n",
       "...                                                  ...     ...   \n",
       "98206                           rs373770404,COSV53069197  NOTCH1   \n",
       "98215             rs1064795070,COSV53025487,COSV53045576  NOTCH1   \n",
       "98242                 rs367825691,CM1610439,COSV53052040  NOTCH1   \n",
       "98266                           rs375065108,COSV53058648  NOTCH1   \n",
       "99467                 rs1318649487,CM138471,COSV65038966   KDM6A   \n",
       "\n",
       "                  Gene  SYMBOL         Consequence    IMPACT  \\\n",
       "541    ENSG00000117713  ARID1A  frameshift_variant      HIGH   \n",
       "1232   ENSG00000117713  ARID1A    missense_variant  MODERATE   \n",
       "1596   ENSG00000162434    JAK1    missense_variant  MODERATE   \n",
       "1939   ENSG00000213281    NRAS    missense_variant  MODERATE   \n",
       "5894   ENSG00000196878   LAMB3         stop_gained      HIGH   \n",
       "...                ...     ...                 ...       ...   \n",
       "98206  ENSG00000148400  NOTCH1    missense_variant  MODERATE   \n",
       "98215  ENSG00000148400  NOTCH1    missense_variant  MODERATE   \n",
       "98242  ENSG00000148400  NOTCH1    missense_variant  MODERATE   \n",
       "98266  ENSG00000148400  NOTCH1    missense_variant  MODERATE   \n",
       "99467  ENSG00000147050   KDM6A         stop_gained      HIGH   \n",
       "\n",
       "                                                CLIN_SIG          rsid  \\\n",
       "541                                           pathogenic   rs797045262   \n",
       "1232   likely_benign,conflicting_interpretations_of_p...   rs150534917   \n",
       "1596                                   likely_pathogenic  rs1057519753   \n",
       "1939                        likely_pathogenic,pathogenic   rs121913250   \n",
       "5894                                          pathogenic   rs747916314   \n",
       "...                                                  ...           ...   \n",
       "98206  uncertain_significance,conflicting_interpretat...   rs373770404   \n",
       "98215           uncertain_significance,likely_pathogenic  rs1064795070   \n",
       "98242  uncertain_significance,conflicting_interpretat...   rs367825691   \n",
       "98266  uncertain_significance,conflicting_interpretat...   rs375065108   \n",
       "99467                                         pathogenic  rs1318649487   \n",
       "\n",
       "         clinvar Chromosome  Start_Position  End_Position  \\\n",
       "541          NaN          1        27022912      27022938   \n",
       "1232         NaN          1        27105880      27105880   \n",
       "1596         NaN          1        65312347      65312347   \n",
       "1939    CM136798          1       115258748     115258748   \n",
       "5894    CM970866          1       209791389     209791389   \n",
       "...          ...        ...             ...           ...   \n",
       "98206        NaN          9       139412607     139412607   \n",
       "98215        NaN          9       139413049     139413049   \n",
       "98242  CM1610439          9       139413921     139413921   \n",
       "98266        NaN          9       139417517     139417517   \n",
       "99467   CM138471          X        44969369      44969369   \n",
       "\n",
       "                             Allele  \n",
       "541    CCCGCCGCCGCCAGCAGCCTGGGCAA/-  \n",
       "1232                            C/G  \n",
       "1596                            C/A  \n",
       "1939                            C/T  \n",
       "5894                            G/A  \n",
       "...                             ...  \n",
       "98206                           C/T  \n",
       "98215                           G/A  \n",
       "98242                           T/C  \n",
       "98266                           C/T  \n",
       "99467                           C/T  \n",
       "\n",
       "[896 rows x 15 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_filtered_unique"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9c287ee4-3d5d-469f-9967-edd4f459937d",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_filtered_unique.to_csv(\n",
    "    \"rsid_unique_variants2.csv\",\n",
    "    sep=\"\\t\",\n",
    "    index=False\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "38ee4077-d028-4591-9031-ceac873b7075",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
